// @admiral-ds/react-ui/dist/components/Breadcrumbs/BreadCrumb.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),styled=require("styled-components"),typography=require("../Typography/typography.cjs.js");require("../Typography/defaultTypographyMixin.cjs.js");var index=require("../TooltipHOC/index.cjs.js"),style=require("./style.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const getTypography=styled.css`
  ${({$dimension})=>{switch($dimension){case"s":return typography.typography["Caption/Caption 1"];case"m":return typography.typography["Body/Body 2 Short"];default:return typography.typography["Body/Body 1 Short"]}}}
`,Crumb=styled__default.default.li`
  display: flex;
  flex-shrink: 0;
  align-items: center;
  white-space: nowrap;
  ${getTypography}
`,IconContainer=styled__default.default.div`
  width: ${({$dimension})=>"l"==$dimension?20:16}px;
  height: ${({$dimension})=>"l"==$dimension?20:16}px;
  flex-shrink: 0;
  margin-right: 8px;

  & *[fill^='#'] {
    fill: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }
`,Content=styled__default.default.span`
  width: 100%;
  height: 100%;
  padding: 4px;
  display: flex;
  align-items: center;
`,TextWrapper=styled__default.default.div`
  position: relative;
  display: inline-block;
  cursor: pointer;
`,TextWithTooltip=index.TooltipHoc(TextWrapper),activeCrumbMixin=styled.css`
  &:hover {
    color: var(--admiral-color-Primary_Primary70, ${p=>p.theme.color["Primary/Primary 70"]});
    ${IconContainer} *[fill^='#'] {
      fill: var(--admiral-color-Primary_Primary70, ${p=>p.theme.color["Primary/Primary 70"]});
    }
  }
  &:active {
    color: var(--admiral-color-Primary_Primary80, ${p=>p.theme.color["Primary/Primary 80"]});
    ${IconContainer} *[fill^='#'] {
      fill: var(--admiral-color-Primary_Primary80, ${p=>p.theme.color["Primary/Primary 80"]});
    }
  }
`,CrumbAnchor=styled__default.default.a`
  display: flex;
  align-items: center;
  text-decoration: none;
  position: relative;
  color: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});

  ${({$active})=>$active&&activeCrumbMixin}

  &:focus {
    &:before {
      position: absolute;
      content: '';
      border: 2px solid var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
      border-radius: 4px;
      top: 0;
      left: 0;
      bottom: 0;
      right: 0;
    }
  }
  &:focus,
  & > ${Content} {
    outline: none;
  }
`,Breadcrumb=e.forwardRef(({text,iconStart,url="#",linkAs,linkProps,tabIndex,dimension="l",active=!0,displaySeparator=!0,containerPropsConfig=props=>props,...props},ref)=>{const tooltip=text.length>40,iconSize="l"===dimension?20:16,crumbProps={ref,$dimension:dimension,...props};return jsxRuntime.jsxs(Crumb,{...crumbProps,...containerPropsConfig(crumbProps),children:[jsxRuntime.jsx(CrumbAnchor,{...active?{href:url}:{},as:active?linkAs:"span",tabIndex,$active:active,...linkProps,children:jsxRuntime.jsxs(Content,{tabIndex:-1,role:"link",children:[iconStart&&jsxRuntime.jsx(IconContainer,{$dimension:dimension,children:iconStart}),tooltip?jsxRuntime.jsx(TextWithTooltip,{renderContent:()=>text,children:text.slice(0,37)+"..."}):text]})}),displaySeparator&&jsxRuntime.jsx(style.Separator,{width:iconSize,height:iconSize,"aria-hidden":!0})]})});exports.Breadcrumb=Breadcrumb,exports.Content=Content,exports.Crumb=Crumb,exports.CrumbAnchor=CrumbAnchor;
