// @admiral-ds/react-ui/dist/components/Breadcrumbs/Menu.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),styled=require("styled-components"),index$1=require("../OverflowMenu/index.cjs.js"),index=require("../TooltipHOC/index.cjs.js"),MenuItem=require("../Menu/MenuItem.cjs.js"),uid=require("../common/uid.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const Option=styled__default.default.a`
  position: relative;
  display: inline-flex;
  align-items: center;
  width: 100%;
  height: 100%;
  text-decoration: none;

  &,
  &:visited,
  &:hover,
  &:active,
  &:focus {
    color: inherit;
  }
`,IconContainer=styled__default.default.div`
  width: ${({$dimension})=>"m"==$dimension?20:16}px;
  height: ${({$dimension})=>"m"==$dimension?20:16}px;
  flex-shrink: 0;
  margin-right: 8px;

  & *[fill^='#'] {
    fill: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }
`,MenuItemWithTooltip=index.TooltipHoc(MenuItem.MenuItem),OverflowMenuStyled=styled__default.default(index$1.OverflowMenu)`
  margin: 0 4px 0 4px;
`;exports.MenuButton=({dimension,options,dropContainerCssMixin,dropContainerClassName,dropContainerStyle})=>{const model=e.useMemo(()=>options.map(item=>{const id=uid.uid();return{id,render:options=>item.text.length>40?e.createElement(MenuItemWithTooltip,{renderContent:()=>item.text,dimension,...options,key:id,role:"option"},jsxRuntime.jsxs(Option,{href:item.url,as:item.linkAs,...item.linkProps,children:[item.iconStart&&jsxRuntime.jsx(IconContainer,{$dimension:dimension,children:item.iconStart}),item.text.slice(0,37)+"..."]})):e.createElement(MenuItem.MenuItem,{dimension,...options,key:id,role:"option"},jsxRuntime.jsxs(Option,{href:item.url,as:item.linkAs,...item.linkProps,children:[item.iconStart&&jsxRuntime.jsx(IconContainer,{$dimension:dimension,children:item.iconStart}),item.text]}))}}),[options]);return jsxRuntime.jsx(OverflowMenuStyled,{dimension,items:model,dropContainerCssMixin,alignSelf:"auto",dropContainerClassName,dropContainerStyle})};
