// @admiral-ds/react-ui/dist/components/Breadcrumbs/style.cjs.js
"use strict";var styled=require("styled-components"),ChevronRightOutline=require("../../node_modules/@admiral-ds/icons/build/system/ChevronRightOutline.svg.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const Separator=styled__default.default(ChevronRightOutline.ReactComponent)`
  & *[fill^='#'] {
    fill: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }
  flex-shrink: 0;
  margin-left: 4px;

  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
`,Navigation=styled__default.default.nav`
  position: relative;
  box-sizing: border-box;
  display: flex;
  width: 100%;
  max-width: 800px;
`,Wrapper=styled__default.default.ol`
  position: relative;
  box-sizing: border-box;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-start;
  width: 100%;
  list-style: none;
  padding: 0;
  margin: 0;
  opacity: ${({$opacity})=>$opacity};
  flex-wrap: nowrap;
  overflow-x: ${({$mobile})=>$mobile?"scroll":"visible"};
  overflow-y: visible;
  &::-webkit-scrollbar {
    width: 0 !important;
    height: 0 !important;
  }
  overflow: -moz-scrollbars-none;
  -ms-overflow-style: none;
  scrollbar-width: none;

  & > li {
    margin-left: 4px;
  }
  & > li:first-child,
  & > li[data-overflow] {
    margin-left: 0;
  }
`,OverflowContentWrapper=styled__default.default.li`
  display: flex;
  overflow: hidden;
  height: ${({$dimension})=>"l"===$dimension?28:24}px;
  max-height: ${({$dimension})=>"l"===$dimension?28:24}px;
  width: 100%;
`,OverflowContent=styled__default.default.ol`
  display: flex;
  flex-direction: row-reverse;
  flex-wrap: wrap;
  box-sizing: border-box;
  overflow: hidden;
  height: 100%;
  width: 100%;
  padding: 0;
  margin: 0;
  list-style: none;

  & > li {
    margin-left: 4px;
  }
  & > li[data-number='0'],
  & > li:last-child {
    margin-left: 0;
  }
`,Compensator=styled__default.default.li`
  display: flex;
  flex: 1;
  order: 0;
`,OverflowItem=styled__default.default.li`
  display: flex;
  flex-shrink: 0;
  align-items: center;
`;exports.Compensator=Compensator,exports.Navigation=Navigation,exports.OverflowContent=OverflowContent,exports.OverflowContentWrapper=OverflowContentWrapper,exports.OverflowItem=OverflowItem,exports.Separator=Separator,exports.Wrapper=Wrapper;
