// @admiral-ds/react-ui/dist/components/ProgressHeader/index.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(require("styled-components"));const Container=styled__default.default.div`
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
`,Wrapper=styled__default.default.div`
  background: var(--admiral-color-Neutral_Neutral20, ${p=>p.theme.color["Neutral/Neutral 20"]});
  min-width: 140px;
  min-height: 4px;
  display: flex;
`,Progress=styled__default.default.div`
  background: ${({theme,$appearance})=>"error"===$appearance?`var(--admiral-color-Error_Error60Main, ${theme.color["Error/Error 60 Main"]})`:`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`};
  transition: all 0.3s linear;
`,ProgressHeader=({percent=0,appearance="primary",...props})=>jsxRuntime.jsx(Container,{...props,children:jsxRuntime.jsx(Wrapper,{children:jsxRuntime.jsx(Progress,{$appearance:appearance,$percent:percent,style:{width:`${percent}%`}})})});ProgressHeader.displayName="ProgressHeader",exports.ProgressHeader=ProgressHeader;
