// @admiral-ds/react-ui/dist/components/GroupActionsPane/SearchInput.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),styled=require("styled-components"),e=require("react"),typography=require("../Typography/typography.cjs.js");require("../Typography/defaultTypographyMixin.cjs.js");var refSetter=require("../common/utils/refSetter.cjs.js"),index=require("../themes/light/index.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}function _interopNamespace(e){if(e&&e.__esModule)return e;var n=Object.create(null);return e&&Object.keys(e).forEach(function(k){if("default"!==k){var d=Object.getOwnPropertyDescriptor(e,k);Object.defineProperty(n,k,d.get?d:{enumerable:!0,get:function(){return e[k]}})}}),n.default=e,Object.freeze(n)}require("../themes/common/locales.cjs.js");var styled__default=_interopDefault(styled),e__namespace=_interopNamespace(e);const collapsedMixin=styled.css`
  max-width: ${({$dimension})=>"s"===$dimension?"240px":"276px"};
  overflow: hidden;
  white-space: nowrap;
  z-index: 1;
  &::after {
    background-color: var(--admiral-color-Neutral_Neutral20, ${p=>p.theme.color["Neutral/Neutral 20"]});
  }
`,InputWrapper=styled__default.default.div`
  display: flex;
  flex: 1 0 auto;
  flex: ${({$visibleInput})=>$visibleInput?"1 0 auto":"0 0 auto"};
  height: 100%;
  position: relative;
  background-color: transparent;
  align-items: center;
  padding-right: 4px;
  &::after {
    content: '';
    position: absolute;
    bottom: 0;
    left: 0;
    width: calc(100% - 4px);
    height: 2px;
    background-color: ${({theme,$visibleInput})=>$visibleInput?`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`:"none"};
  }

  ${({$collapsed})=>$collapsed&&collapsedMixin}
`,StyledInput=styled__default.default.input`
  flex-grow: 1;
  display: ${({$visible})=>$visible?"block":"none"};
  outline: none;
  appearance: none;
  border: none;
  background-color: transparent;
  padding: ${({$dimension})=>"s"===$dimension?"0 12px":"0 16px"};
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});
  box-sizing: border-box;
  width: 100%;
  text-overflow: ellipsis;
  ${props=>"s"===props.$dimension?typography.typography["Body/Body 2 Long"]:typography.typography["Body/Body 1 Long"]};

  &::placeholder {
    color: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }
`,SearchInput=e__namespace.forwardRef(({dimension="l",opened,children,value,locale,collapsed,...props},ref)=>{const theme=styled.useTheme()||index.LIGHT_THEME,placeholder=locale?.inputPlaceholder||theme.locales[theme.currentLocale].groupActionsPane.inputPlaceholder,inputRef=e.useRef(null),[collapsedState,setCollapsedState]=e.useState(!1);e.useEffect(()=>{setCollapsedState(!opened&&!!value)},[opened,value]),e.useEffect(()=>{opened&&inputRef?.current?.focus()},[opened]);const visible=opened||!!value;return jsxRuntime.jsxs(InputWrapper,{$collapsed:collapsedState,$visibleInput:visible,$dimension:dimension,children:[jsxRuntime.jsx(StyledInput,{ref:refSetter.refSetter(ref,inputRef),...props,$visible:visible,value,$dimension:dimension,placeholder}),children]})});exports.SearchInput=SearchInput;
