// @admiral-ds/react-ui/dist/components/StatusIndicator/index.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),styled=require("styled-components"),typography=require("../Typography/typography.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}require("../Typography/defaultTypographyMixin.cjs.js");var styled__default=_interopDefault(styled);const Text=styled__default.default.span``,Circle=styled__default.default.div``,cssDefault=styled.css`
  > ${Text} {
    color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});
  }
  > ${Circle} {
    svg {
      path {
        fill: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
      }
    }
  }
`,StatusContainer=styled__default.default.div`
  display: inline-flex;
  align-items: center;
  flex-direction: ${p=>p.$displayRight?"row":"row-reverse"};
  height: ${p=>"m"===p.$dimension?"24px":"20px"};
  ${p=>"m"===p.$dimension?typography.typography["Body/Body 1 Long"]:typography.typography["Body/Body 2 Long"]}
  > ${Circle} {
    width: ${p=>"m"===p.$dimension?"20px":"16px"};
    height: ${p=>"m"===p.$dimension?"20px":"16px"};
    margin: ${p=>p.$displayRight?"0 10px 0 0":"0 0 0 10px"};
  }
  ${p=>p.$cssMixin||cssDefault}
`,StatusIndicator=({dimension="m",displayRight=!0,icon,cssMixin,text})=>jsxRuntime.jsxs(StatusContainer,{$dimension:dimension,$displayRight:displayRight,$cssMixin:cssMixin,children:[icon&&jsxRuntime.jsx(Circle,{children:icon}),jsxRuntime.jsx(Text,{children:text})]});StatusIndicator.displayName="StatusIndicator",exports.StatusIndicator=StatusIndicator;
