// @admiral-ds/react-ui/dist/components/input/FileUploader/FileInfo.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),styled=require("styled-components"),ErrorBlock=require("./ErrorBlock.cjs.js"),index=require("../../Spinner/index.cjs.js"),index$1=require("../../Tooltip/index.cjs.js"),utils=require("./utils.cjs.js"),FilePDFSolid=require("../../../node_modules/@admiral-ds/icons/build/documents/FilePDFSolid.svg.cjs.js"),FilePPTSolid=require("../../../node_modules/@admiral-ds/icons/build/documents/FilePPTSolid.svg.cjs.js"),FileWordSolid=require("../../../node_modules/@admiral-ds/icons/build/documents/FileWordSolid.svg.cjs.js"),FileXLSSolid=require("../../../node_modules/@admiral-ds/icons/build/documents/FileXLSSolid.svg.cjs.js"),DocsSolid=require("../../../node_modules/@admiral-ds/icons/build/documents/DocsSolid.svg.cjs.js"),JpgSolid=require("../../../node_modules/@admiral-ds/icons/build/documents/JpgSolid.svg.cjs.js"),CloseOutline=require("../../../node_modules/@admiral-ds/icons/build/service/CloseOutline.svg.cjs.js"),borderRadius=require("../../themes/borderRadius.cjs.js"),checkOverflow=require("../../common/utils/checkOverflow.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}function _interopNamespace(e){if(e&&e.__esModule)return e;var n=Object.create(null);return e&&Object.keys(e).forEach(function(k){if("default"!==k){var d=Object.getOwnPropertyDescriptor(e,k);Object.defineProperty(n,k,d.get?d:{enumerable:!0,get:function(){return e[k]}})}}),n.default=e,Object.freeze(n)}var e__namespace=_interopNamespace(e),styled__default=_interopDefault(styled);const Container=styled__default.default.div`
  display: flex;
  flex-direction: column;
  flex: ${p=>"xl"===p.$dimension?"1 1 36%":"1 1 auto"};
  margin-top: 16px;
  overflow: hidden;

  &:nth-of-type(even) {
    margin-left: ${p=>"xl"===p.$dimension?"16px":"0"};
  }
`,statusMixin=styled.css`
  border-color: ${p=>"Error"===p.$status?`var(--admiral-color-Error_Error60Main, ${p.theme.color["Error/Error 60 Main"]})`:`var(--admiral-color-Neutral_Neutral40, ${p.theme.color["Neutral/Neutral 40"]})`};
  color: ${p=>"Queue"===p.$status?`var(--admiral-color-Neutral_Neutral30, ${p.theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral90, ${p.theme.color["Neutral/Neutral 90"]})`};
  background: var(--admiral-color-Special_StaticWhite, ${p=>p.theme.color["Special/Static White"]});
`,PreviewWrapper=styled__default.default.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  border-radius: var(--admiral-border-radius-Medium, ${p=>borderRadius.mediumGroupBorderRadius(p.theme.shape)});
  border-width: 1px;
  border-style: solid;
  padding: ${p=>"xl"===p.$fileDimension?"7px 3px 7px 8px":"7px 15px"};
  ${statusMixin};
`,InfoBlock=styled__default.default.div`
  display: ${p=>"m"===p.$dimension?"block":"flex"};
  align-items: center;
  overflow: hidden;
`,ImagePreview=styled__default.default.div`
  min-width: 40px;
  width: 40px;
  height: 40px;
  border-radius: 4px;
  margin-right: 8px;
  overflow: hidden;

  > img {
    height: 100%;
    width: 100%;
  }
`,IconWrapper=styled__default.default.div`
  margin-right: 8px;
  border-radius: 4px;

  & svg {
    width: 40px;
    height: 40px;
    fill: ${p=>"Queue"===p.$status?`var(--admiral-color-Neutral_Neutral30, ${p.theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral50, ${p.theme.color["Neutral/Neutral 50"]})`};
  }
`,sizeMixin=styled.css`
  align-items: center;
  justify-content: space-between;
`,Content=styled__default.default.div`
  ${p=>"m"===p.$fileDimension&&sizeMixin};
  display: flex;
  flex-direction: ${p=>"m"===p.$fileDimension?"row":"column"};
  min-width: 0;
`,Title=styled__default.default.div`
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
`,Size=styled__default.default.span`
  color: ${p=>"Queue"===p.$status?`var(--admiral-color-Neutral_Neutral30, ${p.theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral50, ${p.theme.color["Neutral/Neutral 50"]})`};
  margin-left: ${p=>"xl"===p.$fileDimension?"0":"4px"};
  white-space: nowrap;
`,FunctionalWrapper=styled__default.default.div`
  display: flex;
  flex: 0 0 auto;
  margin-left: 8px;
`,StyledSpinner=styled__default.default(index.Spinner)`
  margin-right: 8px;
`,Close=styled__default.default.div`
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 32px;
  height: 32px;
  &:hover {
    border-radius: 50%;
    background-color: var(--admiral-color-Opacity_Hover, ${p=>p.theme.color["Opacity/Hover"]});
  }
  & svg {
    width: 24px;
    height: 24px;
    fill: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }
`;exports.FileInfo=({file,onCloseClick=()=>{},dimension,fileDimension,children,...props})=>{const{file:{type,name,size},status,error,showPreview}=file,PreviewIcon=(type=>{switch(type){case"image/jpeg":case"image/png":case"image/tiff":return JpgSolid.ReactComponent;case"application/pdf":return FilePDFSolid.ReactComponent;case"application/vnd.ms-powerpoint":case"application/vnd.openxmlformats-officedocument.presentationml.presentation":return FilePPTSolid.ReactComponent;case"application/vnd.ms-excel":case"application/vnd.openxmlformats-officedocument.spreadsheetml.sheet":return FileXLSSolid.ReactComponent;case"application/msword":case"application/vnd.openxmlformats-officedocument.wordprocessingml.document":return FileWordSolid.ReactComponent;default:return DocsSolid.ReactComponent}})(type),fileFormat=(type=>{switch(type){case"image/jpeg":return"JPEG";case"image/png":return"PNG";case"image/tiff":return"TIFF";case"application/pdf":return"PDF";case"application/vnd.ms-powerpoint":case"application/vnd.openxmlformats-officedocument.presentationml.presentation":return"PPT";case"application/vnd.ms-excel":case"application/vnd.openxmlformats-officedocument.spreadsheetml.sheet":return"XLS";case"application/msword":case"application/vnd.openxmlformats-officedocument.wordprocessingml.document":return"Word";default:return"Docs"}})(type),fileSize=utils.formatBytes(size),fileName=name.substring(0,name.lastIndexOf(".")),[imageSrc,setImageSrc]=e__namespace.useState(""),imageFile=showPreview&&type.startsWith("image"),fileInfo=`${fileFormat}・${fileSize} Mb`,titleRef=e__namespace.useRef(null),[tooltipVisible,setTooltipVisible]=e__namespace.useState(!1),[titleOverflow,setTitleOverflow]=e__namespace.useState(!1);return e.useEffect(()=>{const element=titleRef.current;element&&checkOverflow.checkOverflow(element)!==titleOverflow&&setTitleOverflow(checkOverflow.checkOverflow(element))},[tooltipVisible,setTitleOverflow]),e.useEffect(()=>{function show(){setTooltipVisible(!0)}function hide(){setTooltipVisible(!1)}const title=titleRef.current;if(title)return title.addEventListener("mouseenter",show),title.addEventListener("mouseleave",hide),()=>{title.removeEventListener("mouseenter",show),title.removeEventListener("mouseleave",hide)}},[setTooltipVisible]),e.useEffect(()=>{file&&imageFile&&!children&&(file=>{const reader=new FileReader;reader.onloadend=function(){"string"==typeof reader.result&&setImageSrc(reader.result)},file?reader.readAsDataURL(file.file):setImageSrc("")})(file)},[file]),jsxRuntime.jsxs(Container,{$dimension:dimension,children:[jsxRuntime.jsxs(PreviewWrapper,{...props,$dimension:dimension,$fileDimension:fileDimension,$status:status,children:[children||jsxRuntime.jsxs(InfoBlock,{$dimension:dimension,children:["xl"===fileDimension&&(imageFile?jsxRuntime.jsx(ImagePreview,{children:jsxRuntime.jsx("img",{src:imageSrc,alt:""})}):jsxRuntime.jsx(IconWrapper,{$status:status,children:jsxRuntime.jsx(PreviewIcon,{})})),jsxRuntime.jsxs(Content,{$fileDimension:fileDimension,children:[jsxRuntime.jsx(Title,{ref:titleRef,children:fileName}),tooltipVisible&&titleOverflow&&jsxRuntime.jsx(index$1.Tooltip,{targetElement:titleRef.current,renderContent:()=>`${fileName}`}),jsxRuntime.jsx(Size,{$fileDimension:fileDimension,$status:status,children:fileInfo})]})]}),jsxRuntime.jsxs(FunctionalWrapper,{children:["Loading"===status&&jsxRuntime.jsx(StyledSpinner,{}),utils.dataTransferConstructorSupported()&&jsxRuntime.jsx(Close,{onClick:onCloseClick,children:jsxRuntime.jsx(CloseOutline.ReactComponent,{})})]})]}),error&&"Error"===status&&jsxRuntime.jsx(ErrorBlock.ErrorBlock,{$status:status,children:error})]})};
