// @admiral-ds/react-ui/dist/components/input/FileUploader/index.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),refSetter=require("../../common/utils/refSetter.cjs.js"),FileInfo=require("./FileInfo.cjs.js"),typography=require("../../Typography/typography.cjs.js");require("../../Typography/defaultTypographyMixin.cjs.js");var AttachFileOutline=require("../../../node_modules/@admiral-ds/icons/build/system/AttachFileOutline.svg.cjs.js"),styled=require("styled-components"),utils=require("./utils.cjs.js"),borderRadius=require("../../themes/borderRadius.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}function _interopNamespace(e){if(e&&e.__esModule)return e;var n=Object.create(null);return e&&Object.keys(e).forEach(function(k){if("default"!==k){var d=Object.getOwnPropertyDescriptor(e,k);Object.defineProperty(n,k,d.get?d:{enumerable:!0,get:function(){return e[k]}})}}),n.default=e,Object.freeze(n)}var e__namespace=_interopNamespace(e),styled__default=_interopDefault(styled);const Icon=styled__default.default(AttachFileOutline.ReactComponent)`
  height: ${p=>"xl"===p.$dimension?"40px":"24px"};
  width: ${p=>"xl"===p.$dimension?"40px":"24px"};
  margin-right: ${p=>"m"===p.$dimension?"14px":""};
  margin-bottom: ${p=>"xl"===p.$dimension?"14px":""};

  > * {
    fill: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
  }
`,disabledStyles=styled.css`
  border: 1px dashed var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});

  & svg {
    > * {
      fill: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
    }
  }

  cursor: not-allowed;
  & > * {
    pointer-events: none;
  }
`,hoverStyles=styled.css`
  &:hover {
    cursor: pointer;
    background: var(--admiral-color-Opacity_Hover, ${p=>p.theme.color["Opacity/Hover"]});
  }
  &:active {
    background: var(--admiral-color-Opacity_Press, ${p=>p.theme.color["Opacity/Press"]});
  }
`,CustomWrapper=styled__default.default.div`
  position: relative;
`,UploaderWrapperXL=styled__default.default.div`
  position: relative;
  display: flex;
  box-sizing: border-box;
  align-items: center;
  flex-direction: column;
  padding: 24px 0;
  border: 1px dashed var(--admiral-color-Neutral_Neutral40, ${p=>p.theme.color["Neutral/Neutral 40"]});
  border-radius: var(--admiral-border-radius-Medium, ${p=>borderRadius.mediumGroupBorderRadius(p.theme.shape)});
  ${p=>p.disabled?disabledStyles:hoverStyles};
`,UploaderWrapperM=styled__default.default.div`
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: row;
  padding: 8px 0;
  border: 1px dashed var(--admiral-color-Neutral_Neutral40, ${p=>p.theme.color["Neutral/Neutral 40"]});
  border-radius: var(--admiral-border-radius-Medium, ${p=>borderRadius.mediumGroupBorderRadius(p.theme.shape)});
  ${p=>p.disabled?disabledStyles:hoverStyles};
`,TitleText=styled__default.default.div`
  text-align: ${p=>"xl"===p.$dimension?"center":"start"};
  margin: 0 ${p=>"xl"===p.$dimension?"24px":""};
  margin-bottom: ${p=>"m"===p.$dimension?"16px":"0px"};
  max-width: 100%;
  color: ${p=>p.disabled?`var(--admiral-color-Neutral_Neutral30, ${p.theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral90, ${p.theme.color["Neutral/Neutral 90"]})`};
  ${typography.typography["Body/Body 2 Long"]};
`,Desc=styled__default.default.div`
  color: ${p=>p.disabled?`var(--admiral-color-Neutral_Neutral30, ${p.theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral90, ${p.theme.color["Neutral/Neutral 90"]})`};
  ${typography.typography["Body/Body 1 Long"]};
`,FileInput=styled__default.default.input`
  display: block;
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  opacity: 0;
  cursor: pointer;
  &:disabled {
    cursor: not-allowed;
  }

  &:focus {
    outline: none;
  }
`,Wrapper=styled__default.default.div`
  ${typography.typography["Body/Body 2 Long"]};
`,FileWrapper=styled__default.default.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
`,FileUploader=e__namespace.forwardRef(({multiple=!0,dimension="xl",fileDimension="xl",renderFileInfoList:r,disabled,title,description,children,uploadedFiles,style,className,onRemoveFile,...props},ref)=>{const wrapperRef=e__namespace.useRef(null),inputRef=e__namespace.useRef(null),[files,setFiles]=e__namespace.useState(uploadedFiles),previewProps={dimension,fileDimension};e__namespace.useEffect(()=>{void 0!==uploadedFiles&&setFiles(uploadedFiles)},[uploadedFiles]),e__namespace.useEffect(()=>{function onChangeEventHandler(){const filesToAdd=Array.from(this.files||[]).filter(file=>!props.accept||utils.acceptFile(file,props.accept)).map(file=>({file})),dt=new DataTransfer;filesToAdd.forEach(({file})=>dt.items.add(file)),this.files=dt.files,setFiles(filesToAdd)}const input=inputRef.current;if(input)return input.addEventListener("change",onChangeEventHandler,!0),()=>input.removeEventListener("change",onChangeEventHandler,!0)},[inputRef.current,files,props.accept]);const handleRemoveFile=index=>{if(inputRef.current&&inputRef.current.files){const files=inputRef.current.files,dt=new DataTransfer;for(let i=0;i<files.length;i++){const file=files.item(i);index!==i&&file&&dt.items.add(file)}let event;inputRef.current.files=dt.files,"function"==typeof Event?event=new Event("change",{bubbles:!0}):(event=document.createEvent("Event"),event.initEvent("change",!0,!0)),inputRef.current.dispatchEvent(event)}onRemoveFile?.(index)},renderFileInfoList=r??(({files,onRemoveFile})=>files.map((file,index)=>e.createElement(FileInfo.FileInfo,{...previewProps,key:index,file,onCloseClick:()=>onRemoveFile(index)})));return jsxRuntime.jsx(Wrapper,{ref:wrapperRef,style,className,children:"xl"===dimension?jsxRuntime.jsxs(jsxRuntime.Fragment,{children:[e__namespace.Children.count(children)?jsxRuntime.jsxs(CustomWrapper,{children:[children,jsxRuntime.jsx(FileInput,{...props,ref:refSetter.refSetter(ref,inputRef),type:"file",multiple,disabled})]}):jsxRuntime.jsxs(UploaderWrapperXL,{disabled,children:[jsxRuntime.jsx(Icon,{$dimension:dimension}),title&&jsxRuntime.jsx(TitleText,{$dimension:dimension,disabled,children:title}),jsxRuntime.jsx(FileInput,{...props,ref:refSetter.refSetter(ref,inputRef),type:"file",multiple,disabled})]}),files&&jsxRuntime.jsx(FileWrapper,{children:renderFileInfoList({files,onRemoveFile:handleRemoveFile})})]}):jsxRuntime.jsxs(jsxRuntime.Fragment,{children:[e__namespace.Children.count(children)?jsxRuntime.jsxs(CustomWrapper,{children:[children,jsxRuntime.jsx(FileInput,{...props,ref:refSetter.refSetter(ref,inputRef),type:"file",multiple,disabled})]}):jsxRuntime.jsxs(jsxRuntime.Fragment,{children:[title&&jsxRuntime.jsx(TitleText,{$dimension:dimension,disabled,children:title}),jsxRuntime.jsxs(UploaderWrapperM,{disabled,children:[jsxRuntime.jsx(Icon,{$dimension:dimension}),description&&jsxRuntime.jsx(Desc,{disabled,children:description}),jsxRuntime.jsx(FileInput,{...props,ref:refSetter.refSetter(ref,inputRef),type:"file",multiple,disabled})]})]}),files&&jsxRuntime.jsx(FileWrapper,{children:renderFileInfoList({files,onRemoveFile:handleRemoveFile})})]})})});FileUploader.displayName="FileUploader",exports.FileInfo=FileInfo.FileInfo,exports.FileUploader=FileUploader;
