// @admiral-ds/react-ui/dist/components/form/FieldSet/index.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),styled=require("styled-components"),index=require("../../Field/index.cjs.js"),typography=require("../../Typography/typography.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}require("../../Typography/defaultTypographyMixin.cjs.js");var styled__default=_interopDefault(styled);const Legend=styled__default.default.legend`
  ${props=>"s"===props.$dimension?typography.typography["Body/Body 2 Short"]:typography.typography["Body/Body 1 Short"]}
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});
  fieldset:disabled & {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
    cursor: default;
  }

  [aria-required='true'] &:after {
    content: ' *';
    color: var(--admiral-color-Error_Error60Main, ${p=>p.theme.color["Error/Error 60 Main"]});
  }

  [data-status='error'] & {
    color: var(--admiral-color-Error_Error60Main, ${p=>p.theme.color["Error/Error 60 Main"]});
  }
`,FieldSetExtrasContainer=styled__default.default.div`
  flex: 1 1 100%;
  display: flex;
  margin: 0;
`,FieldSetContainer=styled__default.default.fieldset`
  border: none;
  padding: 0;

  display: flex;
  flex-wrap: wrap;
  flex-direction: ${({$flexDirection})=>$flexDirection||"column"};

  && > *:not(${FieldSetExtrasContainer}) {
    margin-top: ${props=>"s"===props.$dimension?12:16}px;
    margin-right: 16px;
  }

  ${Legend} {
    margin-top: 0;
  }
`,FieldSet=props=>{const{legend,children,extraText,dimension,flexDirection,...fieldsetProps}=props;return jsxRuntime.jsxs(FieldSetContainer,{...fieldsetProps,$dimension:dimension,$flexDirection:flexDirection,"data-dimension":dimension,"aria-required":props.required,"data-status":props.status,"data-required-within":props.required?"":void 0,"data-disabled":props.disabled?"":void 0,children:[legend&&jsxRuntime.jsx(Legend,{$dimension:dimension,children:legend}),children,extraText&&jsxRuntime.jsx(FieldSetExtrasContainer,{children:jsxRuntime.jsx(index.ExtraTextContainer,{children:extraText})})]})};FieldSet.displayName="FieldSet",exports.ALL_FIELDSET_DIMENSIONS_VALUES=["m","s"],exports.FieldSet=FieldSet;
