// @admiral-ds/react-ui/dist/components/Table/mixins.cjs.js
"use strict";var styled=require("styled-components"),typography=require("../Typography/typography.cjs.js");require("../Typography/defaultTypographyMixin.cjs.js");const cellStyle=styled.css`
  padding: ${({$dimension})=>{switch($dimension){case"s":return"6px 12px 5px 12px";case"l":return"12px 16px 11px 16px";case"xl":return"16px 16px 15px 16px";default:return"10px 12px 9px 12px"}}};
`,rowStyle=styled.css`
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});
  ${({$dimension})=>"l"===$dimension||"xl"===$dimension?typography.typography["Body/Body 1 Short"]:typography.typography["Body/Body 2 Short"]}
`,groupRowStyle=styled.css`
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});
  ${({$dimension})=>"l"===$dimension||"xl"===$dimension?typography.typography["Subtitle/Subtitle 2"]:typography.typography["Subtitle/Subtitle 3"]}
`,headerStyle=styled.css`
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});
  ${({$dimension})=>"l"===$dimension||"xl"===$dimension?typography.typography["Subtitle/Subtitle 2"]:typography.typography["Subtitle/Subtitle 3"]}
`,extraTextStyle=styled.css`
  color: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  ${({$dimension})=>"l"===$dimension||"xl"===$dimension?typography.typography["Body/Body 2 Long"]:typography.typography["Caption/Caption 1"]}
`,singleLineTitle=styled.css`
  display: inline-block;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
`,multiLineTitle=styled.css`
  display: -webkit-inline-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: ${({$lineClamp})=>$lineClamp};
  overflow: hidden;
`,disabledRow=styled.css`
  color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  cursor: not-allowed;
  & > * {
    pointer-events: none;
  }
`,underlineRow=styled.css`
  border-bottom: 1px solid var(--admiral-color-Neutral_Neutral20, ${p=>p.theme.color["Neutral/Neutral 20"]});
`,rowBackground=styled.css`
  ${({theme,selected,disabled,$grey,$status,$rowStatusMap})=>disabled?`var(--admiral-color-Neutral_Neutral00, ${theme.color["Neutral/Neutral 00"]})`:selected?`var(--admiral-color-Primary_Primary20, ${theme.color["Primary/Primary 20"]})`:$status&&$rowStatusMap?.[$status]?"string"==typeof $rowStatusMap[$status]?$rowStatusMap[$status]:$rowStatusMap[$status](theme.color):$grey?`var(--admiral-color-Neutral_Neutral05, ${theme.color["Neutral/Neutral 05"]})`:`var(--admiral-color-Neutral_Neutral00, ${theme.color["Neutral/Neutral 00"]})`}
`,borderStyle=styled.css`
  border-right: 1px solid transparent;
  [data-borders='true'] & {
    border-color: ${p=>p.$resizer?`var(--admiral-color-Neutral_Neutral20, ${p.theme.color["Neutral/Neutral 20"]})`:"transparent"};
  }
`;exports.borderStyle=borderStyle,exports.cellStyle=cellStyle,exports.disabledRow=disabledRow,exports.extraTextStyle=extraTextStyle,exports.groupRowStyle=groupRowStyle,exports.headerStyle=headerStyle,exports.multiLineTitle=multiLineTitle,exports.rowBackground=rowBackground,exports.rowStyle=rowStyle,exports.singleLineTitle=singleLineTitle,exports.underlineRow=underlineRow;
