// @admiral-ds/react-ui/dist/components/Table/style.cjs.js
"use strict";var styled=require("styled-components"),ArrowUpOutline=require("../../node_modules/@admiral-ds/icons/build/system/ArrowUpOutline.svg.cjs.js"),ChevronDownOutline=require("../../node_modules/@admiral-ds/icons/build/system/ChevronDownOutline.svg.cjs.js"),dragIcon=require("./icons/dragIcon.svg.cjs.js"),mixins=require("./mixins.cjs.js"),index=require("../IconPlacement/index.cjs.js"),parseShadowFromTheme=require("../common/utils/parseShadowFromTheme.cjs.js"),typography=require("../Typography/typography.cjs.js");require("../Typography/defaultTypographyMixin.cjs.js");var RowWidthResizer=require("./RowWidthResizer.cjs.js"),OverflowMenu=require("./OverflowMenu.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const TableContainer=styled__default.default.div`
  position: relative;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  background: var(--admiral-color-Neutral_Neutral00, ${p=>p.theme.color["Neutral/Neutral 00"]});
  overflow: auto;

  &[data-dragging='true'] ${RowWidthResizer.ResizerWrapper} {
    pointer-events: none;
  }

  &[data-borders='true'] {
    border: 1px solid var(--admiral-color-Neutral_Neutral20, ${p=>p.theme.color["Neutral/Neutral 20"]});
  }
`,StickyWrapper=styled__default.default.div`
  display: flex;
  position: sticky;
  left: 0;
  z-index: 5;
  background: ${({theme,$greyHeader})=>$greyHeader?`var(--admiral-color-Neutral_Neutral05, ${theme.color["Neutral/Neutral 05"]})`:`var(--admiral-color-Neutral_Neutral00, ${theme.color["Neutral/Neutral 00"]})`};
  transition: box-shadow 0.3s;
  ${TableContainer}[data-shadow-left='true'] & {
    box-shadow: 4px 0 12px rgba(0, 0, 0, 0.12);
  }
`,NormalWrapper=styled__default.default.div`
  display: flex;
  &:has(+ div[data-empty='true']) {
    overflow-x: hidden;
  }
`,Filler=styled__default.default.div`
  display: flex;
  flex: 1 1 auto;
  width: unset;
`,HeaderWrapper=styled__default.default.div`
  box-sizing: border-box;
  width: 100%;
  position: sticky;
  top: 0;
  z-index: 6;
`,Header=styled__default.default.div`
  box-sizing: border-box;
  display: flex;
  flex: 1 0 auto;
  min-width: fit-content;
  ${mixins.headerStyle}
  background: ${p=>p.$greyHeader?`var(--admiral-color-Neutral_Neutral05, ${p.theme.color["Neutral/Neutral 05"]})`:`var(--admiral-color-Neutral_Neutral00, ${p.theme.color["Neutral/Neutral 00"]})`};
  border-bottom: 1px solid var(--admiral-color-Neutral_Neutral20, ${p=>p.theme.color["Neutral/Neutral 20"]});
`,Body=styled__default.default.div`
  display: flex;
  flex-direction: column;
  flex: 1 1 auto;
`,ExpandIcon=styled__default.default(ChevronDownOutline.ReactComponent)`
  transition: transform 0.3s ease-in-out;
  transform: rotate(${p=>p.$isOpened?180:0}deg);
`,ExpandIconPlacement=styled__default.default(index.IconPlacement)`
  margin: 0;
  flex-shrink: 0;
`,SortIconWrapper=styled__default.default.div`
  position: relative;
`,SortIcon=styled__default.default(ArrowUpOutline.ReactComponent)`
  display: flex;
  flex-shrink: 0;
  transition: transform 0.3s ease-in-out;
  transform: rotate(0deg);
  margin: 2px 0;

  & *[fill^='#'] {
    fill: ${({theme,$sort})=>"initial"===$sort?"transparent":`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`};
  }
  ${({$sort})=>"desc"===$sort?"transform: rotate(180deg);":""}
`,SortOrder=styled__default.default.div`
  position: absolute;
  top: 1px;
  right: 0;
  font-family: var(--admiral-font-family, ${p=>p.theme.fontFamily});
  font-style: normal;
  font-weight: 500;
  font-size: 8px;
  line-height: 9px;
  font-feature-settings:
    'tnum' on,
    'lnum' on;
  color: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
`,DragIcon=styled__default.default(dragIcon.ReactComponent)`
  display: flex;
  flex-shrink: 0;
  cursor: ${p=>p.$disabled?"not-allowed":"pointer"};
  & *[fill^='#'] {
    fill: ${({theme})=>`var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]})`};
  }
`,Cell=styled__default.default.div`
  display: flex;
  align-items: flex-start;
  flex: 0 0 auto;
  box-sizing: border-box;
  ${mixins.cellStyle};
  overflow: hidden;
  ${mixins.borderStyle}
`,CellTextContent=styled__default.default.div`
  display: block;
  align-items: center;
  width: 100%;
  margin: 2px 0;
  ${({$cellAlign})=>"right"===$cellAlign&&"text-align: right;"}
  overflow: hidden;
`,GroupTextContent=styled__default.default.div`
  display: block;
  align-items: center;
  width: 100%;
  overflow: hidden;
`,CheckboxCell=styled__default.default(Cell)`
  width: ${({$dimension})=>"s"===$dimension||"m"===$dimension?44:56}px;
  padding: ${({$dimension})=>{switch($dimension){case"s":return"6px 12px 5px 12px";case"l":return"12px 16px 11px 16px";case"xl":return"16px 16px 15px 16px";default:return"10px 12px 9px 12px"}}};
  border: none;
`,ExpandCell=styled__default.default(Cell)`
  width: ${({$dimension})=>"s"===$dimension||"m"===$dimension?44:56}px;
  padding: ${({$dimension})=>{switch($dimension){case"s":return"6px 12px 5px 12px";case"l":return"12px 16px 11px 16px";case"xl":return"16px 16px 15px 16px";default:return"10px 12px 9px 12px"}}};
  border: none;
`,DragCell=styled__default.default(Cell)`
  width: ${({$dimension})=>"s"===$dimension||"m"===$dimension?36:48}px;
  padding: ${({$dimension})=>{switch($dimension){case"s":return"6px 8px 5px 8px";case"l":return"12px 12px 11px 12px";case"xl":return"16px 12px 15px 12px";default:return"10px 8px 9px 8px"}}};
  border: none;
`,HeaderCell=styled__default.default.div`
  position: relative;
  display: inline-flex;
  box-sizing: border-box;
  flex: 0 0 auto;
  ${mixins.cellStyle}
  ${mixins.borderStyle}
  cursor: default;
  &[data-draggable='true'] {
    cursor: pointer;
  }
  align-items: flex-start;
`,HeaderCellContent=styled__default.default.div`
  box-sizing: border-box;
  display: flex;
  align-items: flex-start;
  width: 100%;
  ${({$cellAlign})=>"right"===$cellAlign&&styled.css`
      flex-direction: row-reverse;
      & > ${HeaderCellTitle} {
        text-align: right;
        flex-direction: row-reverse;
      }
    `}
`,HeaderCellSpacer=styled__default.default.div`
  display: flex;
  align-self: stretch;
  width: ${p=>p.width?p.width:"12px"};
  flex-shrink: 0;
`,HeaderCellTitle=styled__default.default.div`
  display: inline-flex;
  align-items: flex-start;
  width: 100%;
  overflow: hidden;
  ${({$sortable})=>$sortable&&"cursor: pointer;"}
  &:hover {
    ${SortIcon} *[fill^='#'] {
      fill: ${({theme,$sort})=>"initial"===$sort?`var(--admiral-color-Neutral_Neutral50, ${theme.color["Neutral/Neutral 50"]})`:`var(--admiral-color-Primary_Primary70, ${theme.color["Primary/Primary 70"]})`};
    }

    ${SortOrder} {
      color: ${({theme,$sort})=>"initial"===$sort?`var(--admiral-color-Neutral_Neutral50, ${theme.color["Neutral/Neutral 50"]})`:`var(--admiral-color-Primary_Primary70, ${theme.color["Primary/Primary 70"]})`};
    }
  }
`,TitleContent=styled__default.default.div`
  display: flex;
  flex-direction: column;

  // leave 20px/16px for SortIcon
  max-width: ${({$sortable,$dimension})=>$sortable?`calc(100% - ${"s"===$dimension||"m"===$dimension?16:20}px)`:"100%"};
`,Title=styled__default.default.div`
  position: relative;
  width: 100%;
  ${({$lineClamp})=>1===$lineClamp?mixins.singleLineTitle:mixins.multiLineTitle}
  ${({$withTooltip})=>$withTooltip&&"cursor: pointer;"}
`,ExtraText=styled__default.default(Title)`
  margin: 2px 0;
  ${mixins.extraTextStyle}
`,rowWidthStyle=styled.css`
  width: ${p=>p.$rowWidth};
`,rowHoverMixin=styled.css`
  cursor: pointer;
  & > .tr-simple > * {
    background: var(--admiral-color-Primary_Primary10, ${p=>p.theme.color["Primary/Primary 10"]});
  }
`,groupRowHoverMixin=styled.css`
  &[data-groupover='true'] > .tr-simple > * {
    background: var(--admiral-color-Opacity_Hover, ${p=>p.theme.color["Opacity/Hover"]});
  }
`,Row=styled__default.default.div`
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: fit-content;
  background: var(--admiral-color-Neutral_Neutral00, ${p=>p.theme.color["Neutral/Neutral 00"]});
  ${p=>p.$isGroup?mixins.groupRowStyle:mixins.rowStyle}
  ${({disabled})=>disabled&&mixins.disabledRow}
  ${({$underline})=>$underline&&mixins.underlineRow}
  ${({$rowWidth})=>$rowWidth&&rowWidthStyle}
   &:hover:is(.hoverable) {
    ${({$hover,disabled})=>$hover&&!disabled&&rowHoverMixin}
  }

  &[data-dragover='true'] > * {
    opacity: 0.4;
  }
  transition: opacity 0.3s ease;

  ${groupRowHoverMixin}
`,SimpleRow=styled__default.default.div`
  display: inline-flex;
  min-width: max-content;

  & > * {
    background: ${mixins.rowBackground};
  }

  ${({$showRowsActions})=>!$showRowsActions&&styled.css`
      &:hover {
        & div[data-overflowmenu] {
          visibility: visible;
        }
      }
      & div[data-overflowmenu][data-opened='true'] {
        visibility: visible;
      }
    `}
`,ExpandedRowWrapper=styled__default.default.div`
  display: inline-flex;
  overflow: hidden;
  transition: height 250ms cubic-bezier(0.4, 0, 0.2, 1);
`,ExpandedRowContent=styled__default.default.div`
  display: flex;
  flex: 1 0 auto;
  height: fit-content;
  box-sizing: border-box;
  padding: 0 12px 11px 12px;
`,EmptyMessage=styled__default.default(Cell)`
  margin: 2px 0;
  color: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  border: none;
`,HeaderCellsWrapper=styled__default.default.div`
  display: flex;
  flex: 0 0 auto;
  width: calc(100% - ${p=>(p=>(p.$selectionColumn?"s"===p.$dimension||"m"===p.$dimension?44:56:0)+(p.$expansionColumn?"s"===p.$dimension||"m"===p.$dimension?44:56:0)+(p.$overflowMenuColumn?OverflowMenu.getActionSize(p.$dimension):0))(p)+"px"});
`,HiddenHeader=styled__default.default.div`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  visibility: hidden;
  display: flex;
  overflow: hidden;
  ${mixins.headerStyle}
`,MirrorColumn=styled__default.default(HeaderCell)`
  position: fixed;
  z-index: 6;
  visibility: hidden;
  display: flex;
  box-sizing: border-box;
  max-width: 200px;
  box-shadow: var(--admiral-box-shadow-Shadow08, ${p=>parseShadowFromTheme.parseShadow(p.theme.shadow["Shadow 08"])});
  background: var(--admiral-color-Neutral_Neutral00, ${p=>p.theme.color["Neutral/Neutral 00"]});
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});
  ${({$dimension})=>"s"===$dimension||"m"===$dimension?typography.typography["Subtitle/Subtitle 3"]:typography.typography["Subtitle/Subtitle 2"]}
  padding-left: ${({$dimension})=>"s"===$dimension||"m"===$dimension?36:40}px;
  border: none;

  &[data-cursor='normal'] {
    cursor: grabbing;
  }
  &[data-cursor='error'] {
    cursor: not-allowed;
  }
  & > [data-title] {
    ${mixins.singleLineTitle}
  }
  ${p=>p.$cssMixin}
`,MirrorRow=styled__default.default.div`
  position: fixed;
  z-index: 6;
  visibility: hidden;
  display: flex;
  align-items: center;
  box-sizing: border-box;
  max-width: 288px;
  box-shadow: var(--admiral-box-shadow-Shadow08, ${p=>parseShadowFromTheme.parseShadow(p.theme.shadow["Shadow 08"])});
  background: var(--admiral-color-Neutral_Neutral00, ${p=>p.theme.color["Neutral/Neutral 00"]});
  padding-left: ${({$dimension})=>"s"===$dimension||"m"===$dimension?36:48}px;
  ${mixins.rowStyle}

  &[data-cursor='normal'] {
    cursor: grabbing;
  }
  &[data-cursor='error'] {
    cursor: not-allowed;
  }
  & > .td {
    width: 100%;
  }
  ${p=>p.$cssMixin}
`,Spacer=styled__default.default.div`
  display: flex;
  flex: 0 0 auto;
  will-change: min-height;
  transform: translate3d(0px, 0px, 0px);
`,Edge=styled__default.default.div`
  display: flex;
  width: 0;
  height: auto;
`,ActionMock=styled__default.default.div`
  display: flex;
  position: sticky;
  right: 0;
  z-index: 5;
  .table[data-shadow-right='true'] & {
    box-shadow: -4px 0 12px rgba(0, 0, 0, 0.12);
  }

  min-height: ${({$dimension})=>OverflowMenu.getActionSize($dimension)-1}px;
  width: ${({$dimension})=>OverflowMenu.getActionSize($dimension)}px;
  background-color: inherit;
`;exports.ActionMock=ActionMock,exports.Body=Body,exports.Cell=Cell,exports.CellTextContent=CellTextContent,exports.CheckboxCell=CheckboxCell,exports.DragCell=DragCell,exports.DragIcon=DragIcon,exports.Edge=Edge,exports.EmptyMessage=EmptyMessage,exports.ExpandCell=ExpandCell,exports.ExpandIcon=ExpandIcon,exports.ExpandIconPlacement=ExpandIconPlacement,exports.ExpandedRowContent=ExpandedRowContent,exports.ExpandedRowWrapper=ExpandedRowWrapper,exports.ExtraText=ExtraText,exports.Filler=Filler,exports.GroupTextContent=GroupTextContent,exports.Header=Header,exports.HeaderCell=HeaderCell,exports.HeaderCellContent=HeaderCellContent,exports.HeaderCellSpacer=HeaderCellSpacer,exports.HeaderCellTitle=HeaderCellTitle,exports.HeaderCellsWrapper=HeaderCellsWrapper,exports.HeaderWrapper=HeaderWrapper,exports.HiddenHeader=HiddenHeader,exports.MirrorColumn=MirrorColumn,exports.MirrorRow=MirrorRow,exports.NormalWrapper=NormalWrapper,exports.Row=Row,exports.SimpleRow=SimpleRow,exports.SortIcon=SortIcon,exports.SortIconWrapper=SortIconWrapper,exports.SortOrder=SortOrder,exports.Spacer=Spacer,exports.StickyWrapper=StickyWrapper,exports.TableContainer=TableContainer,exports.Title=Title,exports.TitleContent=TitleContent;
