// @admiral-ds/react-ui/dist/components/input/MultiInput/styled.cjs.js
"use strict";var styled=require("styled-components"),Container$1=require("../Container.cjs.js");require("../../themes/common/locales.cjs.js");var typography=require("../../Typography/typography.cjs.js");require("../../Typography/defaultTypographyMixin.cjs.js");var borderRadius=require("../../themes/borderRadius.cjs.js"),index=require("../TextInput/index.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const containerPadding=styled.css`
  padding: ${({$dimension})=>{switch($dimension){case"xl":return"16px 16px";case"s":return"6px 12px";default:return"8px 16px"}}};
`,ieFixes=styled.css`
  ::-ms-clear,
  ::-ms-reveal {
    display: none;
  }
`,WrapperOptions=styled__default.default.div`
  display: flex;
  flex-wrap: wrap;
  gap: 4px;
  margin-right: 4px;
  flex-grow: 1;
`,IconPanel=styled__default.default.div`
  display: flex;
  align-items: flex-start;

  & > svg {
    border-radius: var(--admiral-border-radius-Medium, ${p=>borderRadius.mediumGroupBorderRadius(p.theme.shape)});
    display: block;
    width: ${props=>{switch(props.$dimension){case"xl":default:return 24;case"s":return 20}}}px;

    &:focus {
      outline: none;
    }

    &:focus-visible {
      outline-offset: 2px;
      outline: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]}) solid 2px;
    }
  }
`,Container=styled__default.default.div`
  position: relative;
  display: flex;
  column-gap: 8px;
  align-items: stretch;
  overflow: hidden;
  border: none;
  border-radius: ${p=>p.$skeleton?0:`var(--admiral-border-radius-Medium, ${borderRadius.mediumGroupBorderRadius(p.theme.shape)})`};

  background-color: var(--admiral-color-Neutral_Neutral00, ${p=>p.theme.color["Neutral/Neutral 00"]});

  &[data-read-only],
  &[data-disabled] {
    background-color: var(--admiral-color-Neutral_Neutral10, ${p=>p.theme.color["Neutral/Neutral 10"]});
  }

  ${({disabled})=>disabled?"cursor: not-allowed;":""}
  ${({$skeleton})=>$skeleton&&Container$1.skeletonMixin};
  ${containerPadding};
  ${index.BorderedDivStyles}
`,Input=styled__default.default.input`
  outline: none;
  appearance: none;
  border-radius: inherit;
  box-sizing: border-box;
  flex: 1 1 48px;
  min-width: 10px;
  border: none;
  background: transparent;
  text-overflow: ellipsis;
  padding: 0;

  ${props=>"s"===props.$dimension?typography.typography["Body/Body 2 Long"]:typography.typography["Body/Body 1 Long"]}
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});

  &&&:disabled {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
    cursor: not-allowed;
  }

  &::placeholder {
    color: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }

  &:disabled::placeholder {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }

  [data-disable-copying] & {
    user-select: none;
    pointer-events: none;
  }

  &&&:user-invalid + ${index.InputBorderedDiv} {
    border: 1px solid var(--admiral-color-Error_Error60Main, ${p=>p.theme.color["Error/Error 60 Main"]});
  }

  &&&:user-invalid:hover:not(:disabled) + ${index.InputBorderedDiv} {
    border: 1px solid var(--admiral-color-Error_Error70, ${p=>p.theme.color["Error/Error 70"]});
  }

  &&&:user-invalid:focus:not(:disabled) + ${index.InputBorderedDiv} {
    border: 2px solid var(--admiral-color-Error_Error60Main, ${p=>p.theme.color["Error/Error 60 Main"]});
  }

  ${ieFixes}
`,IconPanelBefore=styled__default.default(IconPanel)`
  height: fit-content;
  & > *:not(:last-child) {
    margin-right: 8px;
  }
`,IconPanelAfter=styled__default.default(IconPanel)`
  height: fit-content;
  & > *:not(:first-child) {
    margin-left: 8px;
  }
`;exports.Container=Container,exports.IconPanelAfter=IconPanelAfter,exports.IconPanelBefore=IconPanelBefore,exports.Input=Input,exports.WrapperOptions=WrapperOptions;
