// @admiral-ds/react-ui/dist/components/TabMenu/style.cjs.js
"use strict";var styled=require("styled-components"),typography=require("../Typography/typography.cjs.js");require("../Typography/defaultTypographyMixin.cjs.js");var index=require("../OverflowMenu/index.cjs.js"),constants=require("./constants.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const IconWrapper=styled__default.default.div`
  ${({$dimension})=>`\n    width: ${"l"===$dimension?constants.ICON_SIZE_L:constants.ICON_SIZE_M}px;\n    height: ${"l"===$dimension?constants.ICON_SIZE_L:constants.ICON_SIZE_M}px;`}
  margin-right: 8px;
`,BadgeWrapper=styled__default.default.div`
  margin-left: 8px;
  display: flex;
  align-items: center;
`,MenuItemWrapper=styled__default.default.div`
  display: flex;
  overflow: hidden;
  text-overflow: ellipsis;
  align-items: center;
`,Wrapper=styled__default.default.div`
  position: relative;
  display: flex;
  flex: 1 1 auto;
  flex-wrap: ${({$mobile})=>$mobile?"nowrap":"wrap"};
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  box-shadow: inset 0 -${constants.LINE_HEIGHT} 0 0
    ${p=>p.$underline?`var(--admiral-color-Neutral_Neutral20, ${p.theme.color["Neutral/Neutral 20"]})`:"transparent"};
  overflow: ${({$mobile})=>$mobile?"scroll":"hidden"};

  height: ${({$dimension})=>"l"===$dimension?constants.TAB_HEIGHT_L:constants.TAB_HEIGHT_M}px;
  max-height: ${({$dimension})=>"l"===$dimension?constants.TAB_HEIGHT_L:constants.TAB_HEIGHT_M}px;

  &::-webkit-scrollbar {
    width: 0 !important;
    height: 0 !important;
  }

  overflow: -moz-scrollbars-none;
  -ms-overflow-style: none;
  scrollbar-width: none;
`,Underline=styled__default.default.div`
  position: absolute;
  bottom: 0;
  display: flex;
  background-color: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
  height: ${constants.LINE_HEIGHT};
  transition: all 0.2s ease-out;
`,TabContent=styled__default.default.div`
  height: 100%;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  white-space: nowrap;
`,TabContentWrapper=styled__default.default.span`
  position: relative;
  display: inline-flex;
  align-items: center;
  width: 100%;
  height: 100%;
  padding: ${({$dimension})=>"l"===$dimension?constants.TAB_PADDING_L:constants.TAB_PADDING_M};
`,Tab=styled__default.default.button`
  position: relative;
  box-sizing: border-box;
  display: inline-flex;
  align-items: center;
  height: ${({$dimension})=>"m"===$dimension?constants.TAB_HEIGHT_M:constants.TAB_HEIGHT_L}px;
  padding: 0;
  background: transparent;
  appearance: none;
  border: none;
  color: ${p=>p.$selected?`var(--admiral-color-Neutral_Neutral90, ${p.theme.color["Neutral/Neutral 90"]})`:`var(--admiral-color-Neutral_Neutral50, ${p.theme.color["Neutral/Neutral 50"]})`};
  ${({$dimension})=>"m"===$dimension?typography.typography["Body/Body 2 Long"]:typography.typography["Body/Body 1 Long"]}
  user-select: none;
  cursor: pointer;
  -webkit-tap-highlight-color: transparent;

  &:focus:not(:active) {
    &:before {
      position: absolute;
      content: '';
      border: ${constants.LINE_HEIGHT} solid
        var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
      top: 0;
      left: 0;
      bottom: 0;
      right: 0;
    }
  }
  &:focus,
  & > ${TabContentWrapper} {
    outline: none;
  }

  & svg {
    display: inline-block;
    flex: 1 0 auto;
    & *[fill^='#'] {
      fill: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
    }
    width: ${({$dimension})=>"m"===$dimension?constants.ICON_SIZE_M:constants.ICON_SIZE_L}px;
    height: ${({$dimension})=>"m"===$dimension?constants.ICON_SIZE_M:constants.ICON_SIZE_L}px;
    margin-right: ${constants.ICON_MARGIN};
  }
  & [data-badge] {
    margin-left: ${constants.BADGE_MARGIN};
  }

  &:hover:not(:disabled) {
    background: var(--admiral-color-Opacity_Hover, ${p=>p.theme.color["Opacity/Hover"]});
  }
  &:active:not(:disabled) {
    background: var(--admiral-color-Opacity_Press, ${p=>p.theme.color["Opacity/Press"]});
  }
  &:disabled {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
    cursor: not-allowed;
    & *[fill^='#'] {
      fill: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
    }
  }
`,TabWrapper=styled__default.default.div`
  display: flex;
  align-items: center;

  margin-left: ${({$needsOffset,$dimension})=>$needsOffset?"l"===$dimension?constants.TAB_LEFT_OFFSET_L:constants.TAB_LEFT_OFFSET_M:0}px;
`,OverflowMenuContainer=styled__default.default.div`
  visibility: ${({$isHidden})=>$isHidden?"hidden":"visible"};
  ${({$dimension})=>`\n    width: ${"l"===$dimension?constants.OVERFLOW_MENU_CONTAINER_SIZE_L:constants.OVERFLOW_MENU_CONTAINER_SIZE_M};\n    height: ${"l"===$dimension?constants.OVERFLOW_MENU_CONTAINER_SIZE_L:constants.OVERFLOW_MENU_CONTAINER_SIZE_M};\n  `}
`,StyledOverflowMenu=styled__default.default(index.OverflowMenu)`
  margin: ${p=>"l"===p.dimension?constants.OVERFLOW_MENU_MARGIN_L:constants.OVERFLOW_MENU_MARGIN_M};
  &:focus-visible {
    outline-offset: ${p=>"l"===p.dimension?constants.FOCUS_BORDER_OFFSET_L:constants.FOCUS_BORDER_OFFSET_M};
    border-radius: 0;
  }

  & svg {
    & *[fill^='#'] {
      fill: ${({theme,$isActive})=>$isActive?`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`:`var(--admiral-color-Neutral_Neutral50, ${theme.color["Neutral/Neutral 50"]})`};
    }
  }
`;exports.BadgeWrapper=BadgeWrapper,exports.IconWrapper=IconWrapper,exports.MenuItemWrapper=MenuItemWrapper,exports.OverflowMenuContainer=OverflowMenuContainer,exports.StyledOverflowMenu=StyledOverflowMenu,exports.Tab=Tab,exports.TabContent=TabContent,exports.TabContentWrapper=TabContentWrapper,exports.TabWrapper=TabWrapper,exports.Underline=Underline,exports.Wrapper=Wrapper;
