// @admiral-ds/react-ui/dist/components/input/InputLine/index.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),styled=require("styled-components"),e=require("react"),refSetter=require("../../common/utils/refSetter.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const StyledInputLine=styled__default.default.input`
  border: none;
  outline: none;
  appearance: none;
  background-image: none;
  background-color: transparent;
  -webkit-box-shadow: none;
  -moz-box-shadow: none;
  box-shadow: none;
  padding: 0;
  box-sizing: border-box;
  width: 100%;
  color: ${p=>p.$isTmpValue?`var(--admiral-color-Neutral_Neutral30, ${p.theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral90, ${p.theme.color["Neutral/Neutral 90"]})`};

  ::placeholder {
    color: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }
`,PlaceholderValue=styled__default.default.span`
  color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  transition: color 0.3s ease-in-out;
`,AdditionalText=styled__default.default(PlaceholderValue)`
  user-select: none;
  -webkit-user-select: none;
  pointer-events: none;
`,InvisibleInputValue=styled__default.default.span`
  visibility: hidden;
`,InputLineContainer=styled__default.default.div`
  text-align: start;
  width: 100%;
  display: flex;
  overflow: hidden;
  position: relative;

  &:focus-within ${PlaceholderValue} {
    color: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }
`,MaskBox=styled__default.default.div`
  cursor: pointer;
  position: absolute;
  pointer-events: none;
  display: flex;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
`,InputContainer=styled__default.default.div`
  position: relative;
  width: 100%;
`,AdditionalTextWrapper=styled__default.default.div`
  white-space: pre;
`,InputLine=e.forwardRef(({className,dataPlaceholder,tmpValue,prefix,suffix,...inputProps},ref)=>{const inputRef=e.useRef(null),placeholderRef=e.useRef(null),placeholder=inputProps.placeholder,isTmpValue="string"==typeof tmpValue;e.useEffect(()=>{const inputNode=inputRef.current,placeholderNode=placeholderRef.current;function oninput(){const{value}=this,len=value.length;placeholderNode&&("string"==typeof placeholder&&placeholder.length>0&&0==len?placeholderNode.innerHTML="":placeholderNode.innerHTML=dataPlaceholder?.substring(len)||"")}if("string"==typeof dataPlaceholder&&inputNode&&placeholderNode)return inputNode.addEventListener("input",oninput),()=>inputNode.removeEventListener("input",oninput)},[dataPlaceholder,placeholder,inputProps.value,inputProps.defaultValue,tmpValue]);const containerProps={className,onFocus:e=>{e.preventDefault(),"input"!==e.target.tagName.toLowerCase()&&inputRef.current&&inputRef.current.focus()},tabIndex:-1};return jsxRuntime.jsxs(InputLineContainer,{...containerProps,children:[prefix&&(""!==inputProps.value||tmpValue)&&jsxRuntime.jsxs(AdditionalTextWrapper,{children:[jsxRuntime.jsx(AdditionalText,{children:prefix}),jsxRuntime.jsx(AdditionalText,{children:" "})]}),jsxRuntime.jsxs(InputContainer,{children:[jsxRuntime.jsxs(MaskBox,{children:[jsxRuntime.jsx(InvisibleInputValue,{children:isTmpValue?tmpValue:inputProps.value}),jsxRuntime.jsx(PlaceholderValue,{ref:placeholderRef}),suffix&&(""!==inputProps.value||tmpValue)&&jsxRuntime.jsxs(AdditionalTextWrapper,{children:[jsxRuntime.jsx(AdditionalText,{children:" "}),jsxRuntime.jsx(AdditionalText,{children:suffix})]})]}),jsxRuntime.jsx(StyledInputLine,{ref:refSetter.refSetter(ref,inputRef),...inputProps,$isTmpValue:isTmpValue,value:isTmpValue?tmpValue:inputProps.value})]})]})});exports.InputLine=InputLine,exports.StyledInputLine=StyledInputLine;
