// @admiral-ds/react-ui/dist/components/Tooltip/style.cjs.js
"use strict";var styled=require("styled-components"),PositionInPortal=require("../PositionInPortal.cjs.js"),typography=require("../Typography/typography.cjs.js");require("../Typography/defaultTypographyMixin.cjs.js");var borderRadius=require("../themes/borderRadius.cjs.js"),parseShadowFromTheme=require("../common/utils/parseShadowFromTheme.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const TooltipWrapper=styled__default.default.div`
  box-sizing: border-box;
  opacity: 0;
  transition-delay: 200ms;
  transition-property: opacity;
  align-self: center;
  width: max-content;
  min-width: max-content;
  pointer-events: initial;
`,TooltipContainer=styled__default.default.div`
  box-sizing: border-box;
  background-color: var(--admiral-color-Neutral_Neutral80, ${p=>p.theme.color["Neutral/Neutral 80"]});
  ${p=>"m"===p.$dimension?typography.typography["Body/Body 2 Short"]:typography.typography["Caption/Caption 1"]}
  color: var(--admiral-color-Neutral_Neutral00, ${p=>p.theme.color["Neutral/Neutral 00"]});
  border-radius: var(--admiral-border-radius-Small, ${p=>borderRadius.smallGroupBorderRadius(p.theme.shape)});
  box-shadow: var(--admiral-box-shadow-Shadow04, ${p=>parseShadowFromTheme.parseShadow(p.theme.shadow["Shadow 04"])});
  padding: ${p=>"m"===p.$dimension?"4px 8px":"2px 6px"};
  max-width: min(488px, calc(100vw - 16px));
  overflow-wrap: break-word;
`,FakeTarget=styled__default.default.div`
  pointer-events: none;
  height: 100%;
  width: 100%;
  flex: 0 0 auto;
`,Portal=styled__default.default(PositionInPortal.PositionInPortal)`
  display: flex;
  flex-wrap: nowrap;
  ${({$flexDirection})=>$flexDirection?`flex-direction: ${$flexDirection};`:""}
  z-index: var(--admiral-z-index-tooltip, ${({theme})=>theme.zIndex.tooltip});
`;exports.FakeTarget=FakeTarget,exports.Portal=Portal,exports.TooltipContainer=TooltipContainer,exports.TooltipWrapper=TooltipWrapper;
