// @admiral-ds/react-ui/dist/components/ImageViewer/ImageMiniature.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),styled=require("styled-components"),constants=require("./constants.cjs.js"),GalleryOutline=require("../../node_modules/@admiral-ds/icons/build/category/GalleryOutline.svg.cjs.js"),EyeOutline=require("../../node_modules/@admiral-ds/icons/build/service/EyeOutline.svg.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);function getImageMiniatureSize(dimension){switch(dimension){case"xxs":return constants.IMAGE_MINIATURE_SIZE_XXS;case"xs":return constants.IMAGE_MINIATURE_SIZE_XS;case"s":return constants.IMAGE_MINIATURE_SIZE_S;case"m":return constants.IMAGE_MINIATURE_SIZE_M;case"l":return constants.IMAGE_MINIATURE_SIZE_L;case"xl":return constants.IMAGE_MINIATURE_SIZE_XL}}function getImageMiniatureIconSize(dimension){switch(dimension){case"xxs":case"xs":return constants.IMAGE_MINIATURE_ICON_SIZE_XXS_XS;case"s":case"m":return constants.IMAGE_MINIATURE_ICON_SIZE_S_M;case"l":case"xl":return constants.IMAGE_MINIATURE_ICON_SIZE_L_XL}}const StyledCategoryGalleryOutline=styled__default.default(GalleryOutline.ReactComponent)`
  & *[fill^='#'] {
    fill: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }
`,StyledServiceEyeOutline=styled__default.default(EyeOutline.ReactComponent)`
  & *[fill^='#'] {
    fill: var(--admiral-color-Special_StaticWhite, ${p=>p.theme.color["Special/Static White"]});
  }
`,blockPositionCss=styled.css`
  display: flex;
  align-items: center;
  justify-content: center;

  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
`,ErrorOnLoadBlock=styled__default.default.div`
  display: flex;
  align-items: center;
  justify-content: center;

  width: ${p=>getImageMiniatureSize(p.$dimension)}px;
  height: ${p=>getImageMiniatureSize(p.$dimension)}px;
  border-radius: 4px;

  background-color: var(--admiral-color-Neutral_Neutral10, ${p=>p.theme.color["Neutral/Neutral 10"]});

  & svg {
    width: ${p=>getImageMiniatureIconSize(p.$dimension)}px;
    height: ${p=>getImageMiniatureIconSize(p.$dimension)}px;
  }
`,EmptyMiniature=({dimension="m",...props})=>jsxRuntime.jsx(ErrorOnLoadBlock,{...props,$dimension:dimension,children:jsxRuntime.jsx(StyledCategoryGalleryOutline,{})}),StyledEmptyMiniature=styled__default.default(EmptyMiniature)`
  ${blockPositionCss};
`,HoverEffectBlock=styled__default.default.div`
  ${blockPositionCss};
  visibility: hidden;
  background-color: var(--admiral-color-Opacity_Modal, ${p=>p.theme.color["Opacity/Modal"]});
`,hoverStyle=styled.css`
  &:hover {
    & ${HoverEffectBlock} {
      visibility: visible;
    }
  }
`,Wrapper=styled__default.default.div`
  width: ${p=>getImageMiniatureSize(p.$dimension)}px;
  height: ${p=>getImageMiniatureSize(p.$dimension)}px;
  border-radius: 4px;
  position: relative;
  cursor: pointer;
  overflow: hidden;

  flex-shrink: 0;
  display: flex;
  align-items: center;
  justify-content: center;

  & svg {
    width: ${p=>getImageMiniatureIconSize(p.$dimension)}px;
    height: ${p=>getImageMiniatureIconSize(p.$dimension)}px;
  }

  ${hoverStyle}
`,StyledImg=styled__default.default.img`
  object-fit: cover;
  height: 100%;
  width: 100%;
`;exports.EmptyMiniature=EmptyMiniature,exports.ImageMiniature=({item,dimension="m",onError,onMouseDown,...props})=>{const itemSrc="string"==typeof item?item:item.src,itemProps="string"==typeof item?void 0:item,imgRef=e.useRef(null),[errorOnLoadImg,setErrorOnLoadImg]=e.useState(!1);return jsxRuntime.jsxs(Wrapper,{...props,$dimension:dimension,$errorOnLoadImg:errorOnLoadImg,onMouseDown,children:[jsxRuntime.jsx(StyledImg,{...itemProps,ref:imgRef,src:itemSrc,onError:e=>{setErrorOnLoadImg(!0),onError?.(e)}}),errorOnLoadImg?jsxRuntime.jsx(StyledEmptyMiniature,{dimension}):jsxRuntime.jsx(HoverEffectBlock,{children:jsxRuntime.jsx(StyledServiceEyeOutline,{})})]})};
