// @admiral-ds/react-ui/dist/components/ImageViewer/ImageViewerToolbarButton.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),styled=require("styled-components"),constants=require("./constants.cjs.js"),index=require("../TooltipHOC/index.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}const ToolbarButton=_interopDefault(styled).default.button`
  all: unset;
  appearance: none;
  cursor: pointer;
  -webkit-tap-highlight-color: transparent;

  box-sizing: border-box;
  width: ${constants.IMAGE_VIEWER_TOOLBAR_BUTTON_SIZE}px;
  height: ${constants.IMAGE_VIEWER_TOOLBAR_BUTTON_SIZE}px;

  display: flex;
  align-items: center;
  justify-content: center;

  & svg {
    width: ${constants.IMAGE_VIEWER_TOOLBAR_ICON_SIZE}px;
    height: ${constants.IMAGE_VIEWER_TOOLBAR_ICON_SIZE}px;
    & *[fill^='#'] {
      fill: var(--admiral-color-Special_DarkStaticNeutral70, ${p=>p.theme.color["Special/Dark Static Neutral 70"]});
    }
  }

  &:hover {
    & *[fill^='#'] {
      fill: var(--admiral-color-Special_DarkStaticNeutral80, ${p=>p.theme.color["Special/Dark Static Neutral 80"]});
    }
  }

  &:active {
    & *[fill^='#'] {
      fill: var(--admiral-color-Special_DarkStaticNeutral90, ${p=>p.theme.color["Special/Dark Static Neutral 90"]});
    }
  }

  &:disabled {
    cursor: not-allowed;
    & *[fill^='#'] {
      fill: var(--admiral-color-Special_DarkStaticNeutral50, ${p=>p.theme.color["Special/Dark Static Neutral 50"]});
    }
  }
`,TooltipedToolbarButton=index.TooltipHoc(ToolbarButton);exports.ImageViewerToolbarButton=({showTooltip=!0,text="",icon,...props})=>showTooltip?jsxRuntime.jsx(TooltipedToolbarButton,{...props,tooltipDimension:"s",renderContent:()=>text,children:icon}):jsxRuntime.jsx(ToolbarButton,{...props,children:icon});
