// @admiral-ds/react-ui/dist/components/Checkbox/index.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),styled=require("styled-components"),Success=require("./Success.svg.cjs.js"),Minus=require("./Minus.svg.cjs.js"),borderRadius=require("../themes/borderRadius.cjs.js"),keyboardKey=require("../common/keyboardKey.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const width=styled.css`
  width: ${({$dimension})=>{switch($dimension){case"m":default:return"20px";case"s":return"16px"}}};

  fieldset[data-dimension='s'] & {
    width: 16px;
  }
`,height=styled.css`
  height: ${({$dimension})=>{switch($dimension){case"m":default:return"20px";case"s":return"16px"}}};
  fieldset[data-dimension='s'] & {
    height: 16px;
  }
`,Check=styled__default.default(Success.ReactComponent)`
  pointer-events: none;
`,Indeterminate=styled__default.default(Minus.ReactComponent)`
  pointer-events: none;
  & *[fill^='#'] {
    fill: var(--admiral-color-Special_StaticWhite, ${p=>p.theme.color["Special/Static White"]});
  }
`,Container=styled__default.default.div`
  position: relative;
  box-sizing: border-box;
  cursor: ${props=>props.$disabled?"not-allowed":props.$readOnly?"default":"pointer"};

  overflow: visible;
  ${width};
  ${height};
  flex: 0 0 auto; // при вставке во flex контейнер не должны изменяться размеры
`,Background=styled__default.default.div`
  pointer-events: none;
  display: inline-block;
  position: absolute;
  margin: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: var(--admiral-border-radius-Small, ${p=>borderRadius.smallGroupBorderRadius(p.theme.shape)});
  outline: 0;
  transition: background-color 0.1s ease-in;

  /* disable inheritance from parent elements */
  line-height: initial;
  background-color: var(--admiral-color-Neutral_Neutral00, ${p=>p.theme.color["Neutral/Neutral 00"]});
  border: 1px solid
    ${({$error,theme})=>$error?`var(--admiral-color-Error_Error60Main, ${theme.color["Error/Error 60 Main"]})`:`var(--admiral-color-Neutral_Neutral50, ${theme.color["Neutral/Neutral 50"]})`};
  & *[fill^='#'] {
    fill: var(--admiral-color-Special_StaticWhite, ${p=>p.theme.color["Special/Static White"]});
  }
`,actionCss=styled.css`
  content: '';
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  border-radius: 50%;
  width: calc(100% + 16px);
  height: calc(100% + 16px);
`,hoveredCss=styled.css`
  &:not(:disabled) {
    &::after {
      ${actionCss};
      background-color: var(--admiral-color-Opacity_Hover, ${p=>p.theme.color["Opacity/Hover"]});
    }
  }
`,activeCss=styled.css`
  &:not(:disabled) {
    &::after {
      ${actionCss};
      background-color: var(--admiral-color-Opacity_Press, ${p=>p.theme.color["Opacity/Press"]});
    }
  }
`,disabledCss=styled.css`
  pointer-events: none;
  & + ${Background} {
    border: 1px solid var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
    background-color: var(--admiral-color-Neutral_Neutral10, ${p=>p.theme.color["Neutral/Neutral 10"]});
  }
`,disabledCheckedBackgroundCss=styled.css`
  background-color: var(--admiral-color-Primary_Primary30, ${p=>p.theme.color["Primary/Primary 30"]});
  border: none;
  & *[fill^='#'] {
    fill: var(--admiral-color-Neutral_Neutral00, ${p=>p.theme.color["Neutral/Neutral 00"]});
  }
`,checkedBackgroundCss=styled.css`
  background-color: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
  border: none;
`,indeterminateBackgroundCss=styled.css`
  & + ${Background} {
    ${checkedBackgroundCss}
  }
`,disabledIndeterminateBackgroundCss=styled.css`
  & + ${Background} {
    ${disabledCheckedBackgroundCss}
  }
`,indeterminateCss=styled.css`
  ${p=>p.readOnly?disabledIndeterminateBackgroundCss:indeterminateBackgroundCss}
`,Input=styled__default.default.input`
  appearance: none;
  ::-ms-check {
    display: none;
  }
  width: 100%;
  height: 100%;

  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);

  box-sizing: border-box;
  cursor: pointer;
  border-radius: var(--admiral-border-radius-Small, ${p=>borderRadius.smallGroupBorderRadius(p.theme.shape)});
  margin: 0;
  padding: 0;

  ${props=>props.readOnly&&disabledCss}

  &:checked + ${Background} {
    ${props=>props.readOnly?disabledCheckedBackgroundCss:checkedBackgroundCss}
  }
  &:checked:disabled + ${Background} {
    ${disabledCheckedBackgroundCss}
  }

  ${p=>p.$indeterminate&&indeterminateCss}

  &:not(:checked) + ${Background} {
    > * {
      display: ${p=>p.$indeterminate?"block":"none"};
    }
  }

  &:disabled {
    cursor: not-allowed;
    ${disabledCss}
    ${p=>p.$indeterminate&&disabledIndeterminateBackgroundCss}
  }

  ${props=>!props.readOnly&&props.$hovered&&hoveredCss}

  &:hover:not(:disabled),
  &:focus:not(:disabled) + ${hoveredCss}

  &:active:not(:disabled) {
    ${activeCss}
  }

  &:focus-visible {
    outline-offset: 2px;
    outline: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]}) solid 2px;
  }
`,Checkbox=e.forwardRef(({className,dimension="m",disabled,readOnly,hovered,indeterminate,error,...props},ref)=>jsxRuntime.jsxs(Container,{$dimension:dimension,$disabled:disabled,$readOnly:readOnly,className,children:[jsxRuntime.jsx(Input,{ref,disabled,readOnly,...props,type:"checkbox",$indeterminate:indeterminate,onKeyDown:e=>{if(readOnly){keyboardKey.keyboardKey.getCode(e)===keyboardKey.keyboardKey[" "]&&e.preventDefault()}props.onKeyDown?.(e)},"data-hovered":hovered,$hovered:hovered}),jsxRuntime.jsx(Background,{$error:error,children:indeterminate?jsxRuntime.jsx(Indeterminate,{"aria-hidden":"true",focusable:"false"}):jsxRuntime.jsx(Check,{"aria-hidden":"true",focusable:"false"})})]}));Checkbox.displayName="Checkbox",exports.Background=Background,exports.Checkbox=Checkbox;
