// @admiral-ds/react-ui/dist/components/ProgressPage/index.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),styled=require("styled-components"),typography=require("../Typography/typography.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}require("../Typography/defaultTypographyMixin.cjs.js");var styled__default=_interopDefault(styled);const Wrapper=styled__default.default.div`
  background: var(--admiral-color-Neutral_Neutral20, ${p=>p.theme.color["Neutral/Neutral 20"]});
  border-radius: 2px;
  min-width: 140px;
  min-height: 4px;
  display: flex;
`,Progress=styled__default.default.div`
  background: ${({theme,$appearance})=>"error"===$appearance?`var(--admiral-color-Error_Error60Main, ${theme.color["Error/Error 60 Main"]})`:`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`};
  border-radius: 2px;
  transition: all 0.3s linear;
`,Label=styled__default.default.div`
  display: flex;
  padding-bottom: 8px;
  justify-content: space-between;
  width: 100%;
  ${typography.typography["Caption/Caption 1"]}
  color: ${({theme,$appearance})=>"error"===$appearance?`var(--admiral-color-Error_Error60Main, ${theme.color["Error/Error 60 Main"]})`:`var(--admiral-color-Neutral_Neutral90, ${theme.color["Neutral/Neutral 90"]})`};
`,Container=styled__default.default.div``,ProgressPage=({percent=0,label,appearance="primary",...props})=>jsxRuntime.jsxs(Container,{...props,children:[jsxRuntime.jsx(Label,{$appearance:appearance,children:label}),jsxRuntime.jsx(Wrapper,{children:jsxRuntime.jsx(Progress,{$appearance:appearance,style:{width:`${percent}%`}})})]});ProgressPage.displayName="ProgressPage",exports.ProgressPage=ProgressPage;
