// @admiral-ds/react-ui/dist/components/SegmentedControl/index.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),styled=require("styled-components"),SegmentedControlItem=require("./SegmentedControlItem.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}const FieldSetContainer=_interopDefault(styled).default.fieldset`
  display: inline-flex;
  flex-wrap: nowrap;
  border: none;
  padding: 0;

  &[data-appearance='outlined'] {
    & > label:not(:first-child) {
      margin-left: -1px;
    }
    label:has(input:checked) + label:has(input:checked) {
      box-shadow: -1px 0 var(--admiral-color-Neutral_Neutral00, ${p=>p.theme.color["Neutral/Neutral 00"]});
    }
  }
  &[data-appearance='filled'] {
    gap: 1px;
  }
  & > label:first-child > div {
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
  }
  & > label:last-child > div {
    border-top-left-radius: 0;
    border-bottom-left-radius: 0;
  }
  & > label:not(:first-child):not(:last-child) > div {
    border-radius: 0;
  }
`,SegmentedControl=({children,dimension="xl",appearance="outlined",...props})=>jsxRuntime.jsx(FieldSetContainer,{role:"group","data-appearance":appearance,"data-dimension":dimension,...props,children});SegmentedControl.displayName="SegmentedControl",exports.SegmentedControlItem=SegmentedControlItem.SegmentedControlItem,exports.SegmentedControl=SegmentedControl;
