// @admiral-ds/react-ui/dist/components/SegmentedControl/style.cjs.js
"use strict";var styled=require("styled-components"),borderRadius=require("../themes/borderRadius.cjs.js"),mixins=require("./mixins.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const Label=styled__default.default.label`
  position: relative;
  box-sizing: border-box;
  display: inline-block;
`,ButtonContent=styled__default.default.div`
  vertical-align: top;

  display: inline-flex;
  gap: 8px;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: flex-start;
  align-items: center;
  max-width: calc(100% - ${p=>p.$addPadding}px);

  > * {
    display: inline-block;
    flex: 0 1 auto;
    white-space: nowrap;
  }
  > ${mixins.ButtonIconContainer} {
    flex: 0 0 auto;
  }

  height: 24px;
  & > svg {
    width: 24px;
    height: 24px;
  }

  fieldset[data-dimension='s'] & {
    height: 20px;
    & > svg {
      width: 20px;
      height: 20px;
    }
  }
`,Button=styled__default.default.div`
  position: relative;
  box-sizing: border-box;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  white-space: nowrap;
  border: none;
  border-radius: ${p=>`var(--admiral-border-radius-Medium, ${borderRadius.mediumGroupBorderRadius(p.theme.shape)})`};
  vertical-align: middle;
  pointer-events: none;

  ${mixins.dimensionMixin};
  ${p=>p.$buttonCssMixin};

  ${ButtonContent} {
    ${p=>p.$loading?"visibility: hidden;":""}
  }
`,SpinnerContainer=styled__default.default(mixins.ButtonIconContainer)`
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  container-type: inline-size;
`,AdditionalPadding=styled__default.default.div`
  display: inline-block;
  width: 2px;
`,Input=styled__default.default.input`
  appearance: none;
  ::-ms-check {
    display: none;
  }
  position: absolute;
  cursor: pointer;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  padding: 0;
  margin: 0;
  opacity: 0;
  box-sizing: border-box;

  fieldset[data-appearance='outlined'] &:disabled {
    cursor: not-allowed;
    ${mixins.outlineDisabledMixin}
  }

  fieldset[data-appearance='filled'] &:disabled {
    cursor: not-allowed;
    ${mixins.filledDisabledMixin}
  }

  &:checked:not(:disabled) {
    ${mixins.checkedMixin}
    & + div > ${SpinnerContainer} {
      & path {
        fill: ${p=>`var(--admiral-color-Special_StaticWhite, ${p.theme.color["Special/Static White"]})`};
      }
    }
  }

  fieldset[data-appearance='outlined'] &:not(:checked):not(:disabled) {
    ${mixins.outlinedMixin}
  }
  fieldset[data-appearance='filled'] &:not(:checked):not(:disabled) {
    ${mixins.filledMixin}
  }

  &:not(:active):focus-visible + div {
    outline-offset: 2px;
    outline: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]}) solid 2px;
  }
`;exports.ButtonIconContainer=mixins.ButtonIconContainer,exports.AdditionalPadding=AdditionalPadding,exports.Button=Button,exports.ButtonContent=ButtonContent,exports.Input=Input,exports.Label=Label,exports.SpinnerContainer=SpinnerContainer;
