// @admiral-ds/react-ui/dist/components/Chips/style.cjs.js
"use strict";var styled=require("styled-components"),typography=require("../Typography/typography.cjs.js");require("../Typography/defaultTypographyMixin.cjs.js");var index=require("../Badge/index.cjs.js"),index$1=require("../IconPlacement/index.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const heights=styled.css`
  height: ${({$dimension})=>"m"===$dimension?"32px":"24px"};
`,widths=styled.css`
  width: ${({$dimension})=>"m"===$dimension?"32px":"24px"};
`,heightIcons=styled.css`
  height: ${({$dimension})=>"m"===$dimension?"20px":"16px"};
`,heightText=styled.css`
  height: ${({$dimension})=>"m"===$dimension?"20px":"16px"};
`,widthIcons=styled.css`
  width: ${({$dimension})=>"m"===$dimension?"20px":"16px"};
`,filledPaddings=styled.css`
  padding: ${({$dimension})=>"m"===$dimension?"6px 12px":"4px 8px"};
`,outlinedPaddings=styled.css`
  padding: ${({$dimension})=>"m"===$dimension?"5px 11px":"3px 7px"};
`,paddings=styled.css`
  ${({$appearance})=>"filled"===$appearance?filledPaddings:outlinedPaddings}
`,chipTypographyHover=styled.css`
  &:hover {
    color: ${({theme,$appearance,$selected})=>$selected?`var(--admiral-color-Special_StaticWhite, ${theme.color["Special/Static White"]})`:"filled"!==$appearance||$selected?`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`:`var(--admiral-color-Neutral_Neutral90, ${theme.color["Neutral/Neutral 90"]})`};
  }
`,chipTypography=styled.css`
  ${({$dimension})=>"s"===$dimension?typography.typography["Caption/Caption 1"]:typography.typography["Body/Body 2 Long"]}
  color: ${({theme,$appearance,$disabled,$selected})=>$disabled&&!$selected?`var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]})`:$selected||$selected&&$disabled?`var(--admiral-color-Special_StaticWhite, ${theme.color["Special/Static White"]})`:"filled"===$appearance?`var(--admiral-color-Neutral_Neutral90, ${theme.color["Neutral/Neutral 90"]})`:`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`};

  ${$disabled=>!$disabled&&chipTypographyHover}
`,actionsMixin=styled.css`
  &:hover {
    ${({theme,$appearance,$selected,$withCloseIcon})=>$selected?`background-color: var(--admiral-color-Primary_Primary70, ${theme.color["Primary/Primary 70"]});`:"filled"===$appearance?`background-color: var(--admiral-color-Opacity_Neutral12, ${theme.color["Opacity/Neutral 12"]});`:$withCloseIcon?void 0:`background-color: var(--admiral-color-Opacity_Hover, ${theme.color["Opacity/Hover"]});`};
    ${p=>p.$selected&&`\n      border-color: var(--admiral-color-Primary_Primary70, ${p.theme.color["Primary/Primary 70"]});\n    `}
  }
  &:active {
    ${({theme,$appearance,$selected,$withCloseIcon})=>$selected?`background-color: var(--admiral-color-Primary_Primary80, ${theme.color["Primary/Primary 80"]});`:"filled"===$appearance?`background-color: var(--admiral-color-Opacity_Neutral16, ${theme.color["Opacity/Neutral 16"]});`:$withCloseIcon?void 0:`background-color: var(--admiral-color-Opacity_Press, ${theme.color["Opacity/Press"]});`};
    ${p=>p.$selected&&`\n      border-color: var(--admiral-color-Primary_Primary80, ${p.theme.color["Primary/Primary 80"]});\n    `}
  }
`,colorsBorderAndBackground=styled.css`
  background-color: ${({theme,$appearance,$selected,$disabled})=>$selected&&!$disabled?`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`:$selected&&$disabled?`var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]})`:"filled"===$appearance?`var(--admiral-color-Opacity_Neutral8, ${theme.color["Opacity/Neutral 8"]})`:"transparent"};

  border: ${({theme,$appearance,$disabled})=>"filled"===$appearance?"none":$disabled?`1px solid var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]})`:`1px solid var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`};

  border-radius: 16px;

  ${p=>p.$clickable&&!p.$disabled&&actionsMixin}

  &:focus-visible {
    outline: 0;

    &:before {
      border: 2px solid var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
      border-radius: 20px;
      content: '';
      display: block;
      position: absolute;
      top: -5px;
      left: -5px;
      right: -5px;
      bottom: -5px;
      pointer-events: none;
    }
  }
`,ChipComponentStyled=styled__default.default.div`
  display: inline-flex;
  align-items: center;
  box-sizing: border-box;
  position: relative;
  max-width: 190px;
  user-select: none;
  & > * {
    pointer-events: ${({$disabled})=>$disabled?"none":"auto"};
  }
  cursor: ${({$defaultChip,$disabled,$withTooltip})=>!$defaultChip&&!$withTooltip||$disabled?$disabled?"not-allowed":"default":"pointer"};
  ${colorsBorderAndBackground}
  ${heights}
  ${p=>p.$withCloseIcon?`padding-inline-start: ${("s"===p.$dimension?8:12)-("outlined"===p.$appearance?1:0)}px;`:paddings}
  ${p=>p.$withBadge&&!p.$withCloseIcon?`padding-inline-end: ${("s"===p.$dimension?4:6)-("outlined"===p.$appearance?1:0)}px;\n         padding-inline-start: ${("s"===p.$dimension?8:12)-("outlined"===p.$appearance?1:0)}px;`:""}
  ${chipTypography}
`,closeIconWrapperStyle=styled.css`
  display: flex;
  align-items: center;
  justify-content: center;
`,ChipContentWrapperStyled=styled__default.default.div`
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  display: inline-flex;

  ${p=>p.$withCloseIcon&&closeIconWrapperStyle}
  ${p=>p.$withCloseIcon?heights:heightText}
  & svg {
    ${heightIcons}
    ${widthIcons}
    & *[fill^='#'] {
      fill: ${({theme,$appearance,$disabled,$selected})=>$selected?`var(--admiral-color-Special_StaticWhite, ${theme.color["Special/Static White"]})`:$disabled?`var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]})`:"filled"===$appearance?`var(--admiral-color-Neutral_Neutral50, ${theme.color["Neutral/Neutral 50"]})`:`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`};
    }
  }
`,ChipChildrenWrapperStyled=styled__default.default.div`
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  display: inline-block;
`,IconStartWrapperStyled=styled__default.default.div`
  display: inline-block;
  margin-inline-end: 8px;
`,IconEndWrapperStyled=styled__default.default.div`
  display: inline-block;
  margin-inline-start: ${p=>p.$withCloseIcon?"2px":"8px"};
`,IconWrapperStyled=styled__default.default.div`
  ${p=>p.$withCloseIcon&&closeIconWrapperStyle}
  ${p=>p.$withCloseIcon?heights:heightIcons}
  ${p=>p.$withCloseIcon?widths:widthIcons}
  & > svg {
    ${heightIcons}
    ${widthIcons}
  }
`,StyledBadge=styled__default.default(index.Badge)`
  margin-inline-start: ${({dimension})=>"s"===dimension?"6px":"8px"};
`,CloseIconButton=styled__default.default(index$1.CloseIconPlacementButton)`
  //дополнительный отступ в 2px, чтобы кружок ховера не стоял вплотную к элементу слева
  margin-inline-start: ${p=>"s"===p.dimension?"6px":"8px"};
  ${p=>"primary"===p.appearance?"s"===p.dimension?"margin-inline-end: 3px":"margin-inline-end: 5px":""};
`;exports.ChipChildrenWrapperStyled=ChipChildrenWrapperStyled,exports.ChipComponentStyled=ChipComponentStyled,exports.ChipContentWrapperStyled=ChipContentWrapperStyled,exports.CloseIconButton=CloseIconButton,exports.IconEndWrapperStyled=IconEndWrapperStyled,exports.IconStartWrapperStyled=IconStartWrapperStyled,exports.IconWrapperStyled=IconWrapperStyled,exports.StyledBadge=StyledBadge;
