// @admiral-ds/react-ui/dist/components/PaginationSimple/index.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),styled=require("styled-components"),uid=require("../common/uid.cjs.js"),borderRadius=require("../themes/borderRadius.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}function _interopNamespace(e){if(e&&e.__esModule)return e;var n=Object.create(null);return e&&Object.keys(e).forEach(function(k){if("default"!==k){var d=Object.getOwnPropertyDescriptor(e,k);Object.defineProperty(n,k,d.get?d:{enumerable:!0,get:function(){return e[k]}})}}),n.default=e,Object.freeze(n)}var e__namespace=_interopNamespace(e),styled__default=_interopDefault(styled);const Wrapper=styled__default.default.ul`
  position: relative;
  display: flex;
  overflow-x: scroll;
  list-style: none;

  & > li {
    margin-right: 8px;
  }

  & > li:last-child {
    margin-right: 0;
  }

  &::-webkit-scrollbar {
    width: 0 !important;
    height: 0 !important;
  }

  overflow: -moz-scrollbars-none;
  -ms-overflow-style: none;
  scrollbar-width: none;
`,ItemContentWrapper=styled__default.default.span`
  position: relative;
  display: inline-flex;
  align-items: center;
  width: 100%;
  height: 100%;
  padding: 4px;
`,ItemContent=styled__default.default.div`
  display: inline-flex;
  height: 4px;
  width: 16px;
  border-radius: 2px;
  background-color: var(--admiral-color-Neutral_Neutral20, ${p=>p.theme.color["Neutral/Neutral 20"]});

  [aria-selected='true'] & {
    background-color: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
  }
`,Item=styled__default.default.button`
  width: 24px;
  height: 12px;
  border-radius: var(--admiral-border-radius-Small, ${p=>borderRadius.smallGroupBorderRadius(p.theme.shape)});

  position: relative;
  box-sizing: border-box;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  padding: 0;
  background: transparent;
  appearance: none;
  border: 2px solid transparent;
  cursor: pointer;
  -webkit-tap-highlight-color: transparent;

  &:focus {
    border-color: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
  }
  &:focus,
  & > ${ItemContentWrapper} {
    outline: none;
  }

  &:hover ${ItemContent} {
    background-color: var(--admiral-color-Primary_Primary70, ${p=>p.theme.color["Primary/Primary 70"]});
  }
  &:active ${ItemContent} {
    background-color: var(--admiral-color-Primary_Primary80, ${p=>p.theme.color["Primary/Primary 80"]});
  }
`,PaginationSimple=({currentItem,onChange,children,...props})=>jsxRuntime.jsx(Wrapper,{...props,children:children&&e__namespace.Children.toArray(children).map((child,index)=>{const id=uid.uid();return jsxRuntime.jsx("li",{"aria-selected":currentItem===index,onClick:event=>onChange(event,index),children:child},id)})});PaginationSimple.displayName="PaginationSimple",exports.PaginationSimple=PaginationSimple,exports.PaginationSimpleItem=props=>jsxRuntime.jsx(Item,{...props,children:jsxRuntime.jsx(ItemContentWrapper,{tabIndex:-1,children:jsxRuntime.jsx(ItemContent,{})})});
