// @admiral-ds/react-ui/dist/components/Button/dimensionMixin.cjs.js
"use strict";var styled=require("styled-components"),typography=require("../Typography/typography.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}require("../Typography/defaultTypographyMixin.cjs.js");const ButtonIconContainer=_interopDefault(styled).default.div``;function getHorizontalIconPadding(dimension){switch(dimension){case"s":return 13;case"m":return 17;case"l":return 21;default:return 29}}const PaddingForButtonWithIconLeft=styled.css`
  .button-group[data-dimension='xl'] &&,
  &[data-dimension='xl'] {
    ${props=>!props.$displayAsSquare&&`padding-left: ${getHorizontalIconPadding("xl")}px;`}
  }

  .button-group[data-dimension='l'] &&,
  &[data-dimension='l'] {
    ${props=>!props.$displayAsSquare&&`padding-left: ${getHorizontalIconPadding("l")}px;`}
  }

  .button-group[data-dimension='m'] &&,
  &[data-dimension='m'] {
    ${props=>!props.$displayAsSquare&&`padding-left: ${getHorizontalIconPadding("m")}px;`}
  }

  .button-group[data-dimension='s'] &&,
  &[data-dimension='s'] {
    ${props=>!props.$displayAsSquare&&`padding-left: ${getHorizontalIconPadding("s")}px;`}
  }
`,PaddingForButtonWithIconRight=styled.css`
  .button-group[data-dimension='xl'] &&,
  &[data-dimension='xl'] {
    ${props=>!props.$displayAsSquare&&`padding-right: ${getHorizontalIconPadding("xl")}px;`}
  }

  .button-group[data-dimension='l'] &&,
  &[data-dimension='l'] {
    ${props=>!props.$displayAsSquare&&`padding-right: ${getHorizontalIconPadding("l")}px;`}
  }

  .button-group[data-dimension='m'] &&,
  &[data-dimension='m'] {
    ${props=>!props.$displayAsSquare&&`padding-right: ${getHorizontalIconPadding("m")}px;`}
  }

  .button-group[data-dimension='s'] &&,
  &[data-dimension='s'] {
    ${props=>!props.$displayAsSquare&&`padding-right: ${getHorizontalIconPadding("s")}px;`}
  }
`,defaultDimensionMixin=styled.css`
  padding: 0;
  height: 56px;
  ${props=>props.$displayAsSquare?"width: 56px;":"padding: 0 29px;"}
  ${ButtonIconContainer} {
    width: 24px;
    height: 24px;
  }

  ${typography.typography["Button/Button 1"]}
`,dimensionMixin=styled.css`
  ${defaultDimensionMixin}

  .button-group[data-dimension='xl'] &&,
  &[data-dimension='xl'] {
    ${defaultDimensionMixin}
  }

  .button-group[data-dimension='l'] &&,
  &[data-dimension='l'] {
    height: 48px;
    ${props=>props.$displayAsSquare?"width: 48px;":"padding: 0 21px;"}
  }

  .button-group[data-dimension='m'] &&,
  &[data-dimension='m'] {
    height: 40px;
    ${props=>props.$displayAsSquare?"width: 40px;":"padding: 0 17px;"}
  }

  .button-group[data-dimension='s'] &&,
  &[data-dimension='s'] {
    height: 32px;
    ${props=>props.$displayAsSquare?"width: 32px;":"padding: 0 13px;"}
    ${ButtonIconContainer} {
      width: 20px;
      height: 20px;
    }
    ${typography.typography["Button/Button 2"]}
  }
`;exports.ButtonIconContainer=ButtonIconContainer,exports.PaddingForButtonWithIconLeft=PaddingForButtonWithIconLeft,exports.PaddingForButtonWithIconRight=PaddingForButtonWithIconRight,exports.dimensionMixin=dimensionMixin;
