// @admiral-ds/react-ui/dist/components/Button/index.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),styled=require("styled-components"),SpinnerIcon=require("../Spinner/SpinnerIcon.cjs.js"),index=require("./appearanceMixin/index.cjs.js"),dimensionMixin=require("./dimensionMixin.cjs.js"),borderRadius=require("../themes/borderRadius.cjs.js"),animation=require("../skeleton/animation.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const ButtonContent=styled__default.default.div`
  vertical-align: top;

  display: inline-flex;
  gap: 8px;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: flex-start;
  align-items: center;
  max-width: calc(100% - ${p=>p.$addPadding}px);

  > * {
    display: inline-block;
    flex: 0 1 auto;
    white-space: nowrap;
  }
  > ${dimensionMixin.ButtonIconContainer} {
    flex: 0 0 auto;
  }

  height: 24px;
  & > svg {
    width: 24px;
    height: 24px;
  }

  .button-group[data-dimension='m'] &&,
  .button-group[data-dimension='l'] &&,
  .button-group[data-dimension='xl'] && {
    height: 24px;
    & > svg {
      width: 24px;
      height: 24px;
    }
  }
  .button-group[data-dimension='s'] &&,
  [data-dimension='s'] & {
    height: 20px;
    & > svg {
      width: 20px;
      height: 20px;
    }
  }
`,SpinnerContainer=styled__default.default(dimensionMixin.ButtonIconContainer)`
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  container-type: inline-size;
  .button-group[data-appearance='primary'] & {
    & path {
      fill: ${p=>`var(--admiral-color-Special_StaticWhite, ${p.theme.color["Special/Static White"]})`};
    }
  }
  .button-group[data-appearance='secondary'] &,
  .button-group[data-appearance='tertiary'] & {
    & path {
      fill: ${p=>`var(--admiral-color-Primary_Primary60Main, ${p.theme.color["Primary/Primary 60 Main"]})`};
    }
  }
`,AdditionalPadding=styled__default.default.div`
  display: inline-block;
  width: 2px;
`,Button=e.forwardRef(({appearance="primary",dimension="xl",type="button",loading=!1,skeleton=!1,iconStart,iconEnd,icon,iconPlace="left",children,buttonCssMixin,displayAsDisabled,displayAsSquare,...props},ref)=>{const spinnerInverse="secondary"!==appearance&&"tertiary"!==appearance&&"ghost"!==appearance,hasIconStart=!!iconStart||!!icon&&"left"===iconPlace,hasIconEnd=!!iconEnd||!!icon&&"right"===iconPlace,additionalPadding=(displayAsSquare||hasIconStart?0:2)+(displayAsSquare||hasIconEnd?0:2);return jsxRuntime.jsxs(StyledButton,{ref,$appearance:appearance,$dimension:dimension,type,$loading:loading,$skeleton:skeleton,$buttonCssMixin:buttonCssMixin,$displayAsSquare:displayAsSquare,$displayAsDisabled:displayAsDisabled,...props,children:[loading&&jsxRuntime.jsx(SpinnerContainer,{children:jsxRuntime.jsx(SpinnerIcon.SpinnerIcon,{$inverse:spinnerInverse})}),!displayAsSquare&&!hasIconStart&&jsxRuntime.jsx(AdditionalPadding,{}),jsxRuntime.jsxs(ButtonContent,{$addPadding:additionalPadding,children:[hasIconStart?jsxRuntime.jsx(dimensionMixin.ButtonIconContainer,{children:iconStart||icon}):null,e.Children.toArray(children).map((child,index)=>"string"==typeof child?jsxRuntime.jsx("div",{children:child},child+index):child),hasIconEnd?jsxRuntime.jsx(dimensionMixin.ButtonIconContainer,{children:iconEnd||icon}):null]}),!displayAsSquare&&!hasIconEnd&&jsxRuntime.jsx(AdditionalPadding,{})]})}),StyledButton=styled__default.default.button.attrs(props=>({"data-dimension":props.$dimension,"data-appearance":[props.$appearance,props.$displayAsDisabled?"disabled":void 0].filter(val=>void 0!==val).join(" ")}))`
  position: relative;
  box-sizing: border-box;
  display: inline-block;
  white-space: nowrap;
  border: none;
  border-radius: ${p=>p.$skeleton?0:`var(--admiral-border-radius-Medium, ${borderRadius.mediumGroupBorderRadius(p.theme.shape)})`};
  appearance: none;
  vertical-align: middle;
  cursor: ${p=>p.disabled?"not-allowed":"pointer"};
  ${p=>(p.$loading||p.$skeleton)&&"pointer-events: none"};

  ${index.appearanceMixin};
  ${dimensionMixin.dimensionMixin};
  ${p=>p.$buttonCssMixin};
  ${({$skeleton})=>$skeleton&&animation.skeletonAnimationMixin};

  ${ButtonContent} {
    ${p=>p.$loading||p.$skeleton?"visibility: hidden;":""}
  }
`;Button.displayName="Button",exports.ButtonIconContainer=dimensionMixin.ButtonIconContainer,exports.PaddingForButtonWithIconLeft=dimensionMixin.PaddingForButtonWithIconLeft,exports.PaddingForButtonWithIconRight=dimensionMixin.PaddingForButtonWithIconRight,exports.dimensionMixin=dimensionMixin.dimensionMixin,exports.Button=Button;
