// @admiral-ds/react-ui/dist/components/input/Select/MultipleSelectChips/styled.cjs.js
"use strict";var styled=require("styled-components"),constants=require("../constants.cjs.js"),index=require("../../../Chips/index.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const ShadowCounterChip=styled__default.default.div`
  width: ${constants.COUNTER_WIDTH}px;
  height: 24px;
`,CounterChipWrap=styled__default.default.div`
  display: flex;
  width: ${constants.COUNTER_WIDTH}px;
  visibility: hidden;
  [data-show-count] & {
    visibility: visible;
  }
  > * {
    width: ${constants.COUNTER_WIDTH}px;
    border-radius: 16px;
    // Убирает действие inline-block
    display: flex;
  }
`,StyledCounterChip=styled__default.default(index.Chips)`
  flex-shrink: 0;
  flex-grow: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 35px;
  padding: 4px 0;
  cursor: pointer;
`,ContentTooltip=styled__default.default.div`
  max-width: 488px;

  @media (max-width: 768px) {
    max-width: 250px;
  }
`,disabledChipStyle=styled.css`
  // оставлено, чтоб обрабатывался onClick
  pointer-events: auto;
  &:hover {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }
`,StyledChip=styled__default.default(index.Chips)`
  margin-right: 4px;
  // Убирает пробел от inline-flex
  display: flex;
  ${({disabled})=>disabled&&disabledChipStyle};

  min-width: ${constants.COUNTER_WIDTH}px;
  @media (max-width: 768px) {
    max-width: 140px;
  }

  // По дизайну при наведении цвет самого чипса меняться не должен, но т.к. есть обработчик onClick для stopPropagation,
  // то крестик чипса получается clickable и при ховере меняет цвет
  // disabled и readOnly чипсы не меняют цвет при ховере (в readOnly в целом отсутствует кнопка крестика)
  &:hover {
    ${({theme,disabled})=>{if(!disabled)return`background-color: var(--admiral-color-Opacity_Neutral8, ${theme.color["Opacity/Neutral 8"]})`}}
  }
`;exports.ContentTooltip=ContentTooltip,exports.CounterChipWrap=CounterChipWrap,exports.ShadowCounterChip=ShadowCounterChip,exports.StyledChip=StyledChip,exports.StyledCounterChip=StyledCounterChip;
