// @admiral-ds/react-ui/dist/components/FloatingButton/style.cjs.js
"use strict";var styled=require("styled-components"),index$1=require("../Badge/index.cjs.js"),index$2=require("../BadgeDot/index.cjs.js"),index=require("../TooltipHOC/index.cjs.js"),parseShadowFromTheme=require("../common/utils/parseShadowFromTheme.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const focusVisibleStyle=styled.css`
  &:focus-visible {
    outline-offset: 2px;
    outline: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]}) solid 2px;
  }
`,primaryAppearanceMixin=styled.css`
  background-color: ${({theme,$disabled})=>$disabled?`var(--admiral-color-Neutral_Neutral10, ${theme.color["Neutral/Neutral 10"]})`:`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`};
  & *[fill^='#'] {
    fill: ${({theme,$disabled})=>$disabled?`var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Special_StaticWhite, ${theme.color["Special/Static White"]})`};
  }

  ${({theme,$disabled})=>$disabled?"":styled.css`
          &:hover {
            background-color: var(--admiral-color-Primary_Primary70, ${theme.color["Primary/Primary 70"]});
          }

          &:active {
            background-color: var(--admiral-color-Primary_Primary80, ${theme.color["Primary/Primary 80"]});
          }
        `}
`,secondaryAppearanceMixin=styled.css`
  background-color: var(--admiral-color-Special_ElevatedBG, ${p=>p.theme.color["Special/Elevated BG"]});
  & *[fill^='#'] {
    fill: ${({theme,$disabled})=>$disabled?`var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`};
  }

  ${({theme,$disabled})=>$disabled?"":styled.css`
          &:hover {
            background-color: var(--admiral-color-Opacity_Hover, ${theme.color["Opacity/Hover"]});
          }

          &:active {
            background-color: var(--admiral-color-Opacity_Press, ${theme.color["Opacity/Press"]});
          }
        `}
`,dimensionMixin=styled.css`
  width: ${p=>"xl"==p.$dimension?56:40}px;
  height: ${p=>"xl"==p.$dimension?56:40}px;

  .floating_button_menu[data-dimension='xl'] && {
    width: 56px;
    height: 56px;
    margin-bottom: 16px;
  }
  .floating_button_menu[data-dimension='m'] && {
    width: 40px;
    height: 40px;
    margin-bottom: 12px;
  }
`,FloatingButtonWrapper=styled__default.default.button`
  position: fixed;
  inset-inline-end: ${p=>p.$mobile?16:28}px;
  inset-block-end: ${p=>p.$mobile?16:28}px;
  display: block;
  border: none;
  border-radius: 50%;
  appearance: none;
  -webkit-tap-highlight-color: transparent;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  ${dimensionMixin};
  cursor: ${p=>p.disabled?"not-allowed":"pointer"};
  box-shadow: var(--admiral-box-shadow-Shadow08, ${p=>parseShadowFromTheme.parseShadow(p.theme.shadow["Shadow 08"])});
  ${focusVisibleStyle}
`,FloatingButtonWrapperWithTooltip=index.TooltipHoc(FloatingButtonWrapper),FloatingButtonContent=styled__default.default.div`
  display: flex;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: ${p=>"m"==p.$dimension?8:16}px;
  border-radius: 50%;
  ${p=>"primary"==p.$appearance?primaryAppearanceMixin:secondaryAppearanceMixin}
  .floating_button_menu && {
    ${secondaryAppearanceMixin}
  }

  & > svg {
    flex: 0 0 auto;
    height: 24px;
    width: 24px;
  }
`,badgeMixin=styled.css`
  position: absolute;
  border: 1px solid var(--admiral-color-Neutral_Neutral00, ${p=>p.theme.color["Neutral/Neutral 00"]});
`,Badge=styled__default.default(index$1.Badge)`
  top: -3px;
  inset-inline-end: -3px;
  ${badgeMixin}
  padding: ${({dimension})=>"s"===dimension?"0 4px":"0 5px"};
`,BadgeDot=styled__default.default(index$2.BadgeDot)`
  top: ${p=>"m"==p.dimension?1:3}px;
  inset-inline-end: ${p=>"m"==p.dimension?1:3}px;
  ${badgeMixin}
`,GroupWrapper=styled__default.default.div`
  position: fixed;
  inset-inline-end: ${p=>p.$mobile?16:28}px;
  inset-block-end: ${p=>p.$mobile?16:28}px;
  box-sizing: border-box;
  width: ${p=>"m"==p.$dimension?40:56}px;
  height: auto;

  & > div[data-visible='false'] {
    margin-bottom: ${p=>"m"==p.$dimension?-20:-28}px;
    opacity: 0;
    transition:
      margin-bottom ${"\n  200ms cubic-bezier(0.4, 0, 1, 1) 0ms\n"},
      opacity ${"\n  200ms cubic-bezier(0.4, 0, 1, 1) 0ms\n"};
  }
  & > div[data-visible='true'] {
    margin-bottom: ${p=>"m"==p.$dimension?12:16}px;
    opacity: 1;
    transition:
      margin-bottom ${"\n  200ms cubic-bezier(0, 0, 0.2, 1) 0ms\n"},
      opacity ${"\n  200ms cubic-bezier(0, 0, 0.2, 1) 0ms\n"};
  }

  & button {
    position: relative;
    inset-block-end: auto;
    inset-inline-end: auto;
  }

  ${p=>p.$dropContainerCssMixin||""}
`,MenuWrapper=styled__default.default.div`
  position: relative;
  display: block;
  box-sizing: border-box;

  && > button:last-child {
    margin: 0px;
  }
`;exports.Badge=Badge,exports.BadgeDot=BadgeDot,exports.FloatingButtonContent=FloatingButtonContent,exports.FloatingButtonWrapper=FloatingButtonWrapper,exports.FloatingButtonWrapperWithTooltip=FloatingButtonWrapperWithTooltip,exports.GroupWrapper=GroupWrapper,exports.MenuWrapper=MenuWrapper,exports.TRANSITION_DURATION=200;
