// @admiral-ds/react-ui/dist/components/Hint/style.cjs.js
"use strict";var styled=require("styled-components"),typography=require("../Typography/typography.cjs.js");require("../Typography/defaultTypographyMixin.cjs.js");var PositionInPortal=require("../PositionInPortal.cjs.js"),borderRadius=require("../themes/borderRadius.cjs.js"),index=require("../IconPlacement/index.cjs.js"),parseShadowFromTheme=require("../common/utils/parseShadowFromTheme.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const AnchorWrapper=styled__default.default.div`
  display: inline-block;
  position: relative;
  cursor: pointer;
  ${p=>p.$anchorCssMixin?p.$anchorCssMixin:""}
`,CloseButton=styled__default.default(index.CloseIconPlacementButton)`
  position: absolute;
  top: 20px;
  right: 20px;
`,sizes=styled.css`
  width: ${({$dimension,$isMobile})=>$isMobile?"calc(100vw - 32px)":"s"===$dimension?"280px":"m"===$dimension?"384px":"488px"};
`,HintWrapper=styled__default.default.div`
  display: flex;
  justify-content: space-between;
  padding: 8px;
  box-sizing: border-box;
  margin: 0 0 0 -8px;
  width: max-content;
  min-width: max-content;
  opacity: 0;
  transition-delay: 200ms;
  transition-property: opacity;
  pointer-events: initial;
  position: relative;
`,HintDialog=styled__default.default.div`
  display: flex;
  padding: 4px 0;
  background-color: var(--admiral-color-Special_ElevatedBG, ${p=>p.theme.color["Special/Elevated BG"]});
  ${typography.typography["Body/Body 2 Long"]}
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});
  box-shadow: var(--admiral-box-shadow-Shadow08, ${p=>parseShadowFromTheme.parseShadow(p.theme.shadow["Shadow 08"])});
  border-radius: var(--admiral-border-radius-Medium, ${p=>borderRadius.mediumGroupBorderRadius(p.theme.shape)});
  box-sizing: border-box;
  ${sizes}
`,HintContent=styled__default.default.div`
  display: flex;
  flex: 1 1 auto;
  box-sizing: border-box;
  max-height: 300px;
  overflow: auto;
  padding: 8px 16px;
  [data-trigger='click'] & {
    padding-right: ${52}px;
  }
`,Portal=styled__default.default(PositionInPortal.PositionInPortal)`
  display: flex;
  flex-wrap: nowrap;
  ${({$flexDirection})=>$flexDirection?`flex-direction: ${$flexDirection};`:"flex-direction: column;"}
  z-index: var(--admiral-z-index-hint, ${({theme})=>theme.zIndex.hint});
`,FakeTarget=styled__default.default.div`
  pointer-events: none;
  height: 100%;
  width: 100%;
  flex: 0 0 auto;
`;exports.AnchorWrapper=AnchorWrapper,exports.CloseButton=CloseButton,exports.FakeTarget=FakeTarget,exports.HintContent=HintContent,exports.HintDialog=HintDialog,exports.HintWrapper=HintWrapper,exports.Portal=Portal;
