// @admiral-ds/react-ui/dist/components/input/NumberInput/AutoSizeInput.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),changeInputData=require("../../common/dom/changeInputData.cjs.js"),typography=require("../../Typography/typography.cjs.js");require("../../Typography/defaultTypographyMixin.cjs.js");var styled=require("styled-components"),refSetter=require("../../common/utils/refSetter.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);function defaultHandleInput(newInputData){return newInputData||{}}const Prefix=styled__default.default.div`
  display: flex;
  flex-shrink: 0;
  align-items: center;
  user-select: none;
  color: ${({theme,disabled})=>disabled?`var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral50, ${theme.color["Neutral/Neutral 50"]})`};
  ${({$align})=>"right"===$align&&"margin-left: auto;"}
`,Suffix=styled__default.default(Prefix)`
  min-width: 0;
  ${({$align})=>"right"===$align&&styled.css`
      flex: 0 1 auto;
      margin-left: 0;
    `}
`,Sizer=styled__default.default.div`
  display: flex;
  flex-shrink: 0;
  visibility: hidden;
  white-space: pre;
  box-sizing: border-box;
  ${({$hasPrefix,$align})=>!$hasPrefix&&"right"===$align&&"margin-left: auto;"}
`,BorderedDiv=styled__default.default.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  margin: 0;
  pointer-events: none;
  overflow: hidden;
  min-width: 0;

  background: none;
  border: 1px solid
    ${({theme,$status})=>"error"===$status?`var(--admiral-color-Error_Error60Main, ${theme.color["Error/Error 60 Main"]})`:"success"===$status?`var(--admiral-color-Success_Success50Main, ${theme.color["Success/Success 50 Main"]})`:`var(--admiral-color-Neutral_Neutral40, ${theme.color["Neutral/Neutral 40"]})`};
  border-radius: inherit;

  [data-read-only] & {
    border-color: transparent;
  }
`,colorsBorderAndBackground=styled.css`
  &:focus + ${BorderedDiv} {
    border: 2px solid var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
  }

  &:disabled + ${BorderedDiv} {
    border-color: transparent;
  }

  &[data-status='error']:focus + ${BorderedDiv}, &:user-invalid:focus + ${BorderedDiv} {
    border: 2px solid var(--admiral-color-Error_Error60Main, ${p=>p.theme.color["Error/Error 60 Main"]});
  }

  &:user-invalid + ${BorderedDiv} {
    border: 1px solid var(--admiral-color-Error_Error60Main, ${p=>p.theme.color["Error/Error 60 Main"]});
  }

  &[data-status='success']:focus + ${BorderedDiv} {
    border: 2px solid var(--admiral-color-Success_Success50Main, ${p=>p.theme.color["Success/Success 50 Main"]});
  }

  &:disabled {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }

  [data-read-only] &:focus + ${BorderedDiv} {
    border-color: transparent;
  }
`,ieFixes=styled.css`
  ::-ms-clear,
  ::-ms-reveal {
    display: none;
  }
`,Input=styled__default.default.input`
  outline: none;
  appearance: none;
  border: none;
  padding: 0;
  position: relative;
  box-sizing: border-box;
  display: flex;
  flex: 1 0 auto;
  max-width: 100%;

  background: transparent;
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});
  text-align: ${({$align})=>"left"===$align?"left":"right"};

  ${props=>"s"===props.$dimension?typography.typography["Body/Body 2 Long"]:typography.typography["Body/Body 1 Long"]}
  &::placeholder {
    color: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }

  &:disabled::placeholder {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }

  [data-disable-copying] & {
    user-select: none;
    pointer-events: none;
  }

  &:disabled {
    cursor: not-allowed;
  }

  ${colorsBorderAndBackground}
  ${ieFixes}
`,horizontalPaddingValue=props=>{switch(props.$dimension){case"xl":default:return 16;case"s":return 12}},iconSizeValue=props=>{switch(props.$dimension){case"xl":default:return 24;case"s":return 20}},HiddenContent=styled__default.default.div`
  position: absolute;
  top: 0;
  bottom: 0;
  overflow: hidden;
  pointer-events: none;
  box-sizing: border-box;
  display: flex;
  align-items: center;
  left: ${horizontalPaddingValue}px;
  right: ${props=>horizontalPaddingValue(props)+(iconSizeValue(props)+8)*(props.$iconCount??0)}px;
`,AutoSizeInput=e.forwardRef(({dimension,placeholder,type,prefix,suffix,status,iconCount,align,innerValue,hideSpaceAfterPrefix=!1,hideSpaceBeforeSuffix=!1,handleInput=defaultHandleInput,...props},ref)=>{const[showPrefixSuffix,setPrefixSuffix]=e.useState(!1),sizerRef=e.useRef(null),inputRef=e.useRef(null),prefixRef=e.useRef(null),suffixRef=e.useRef(null),updateInputLeftPadding=()=>{inputRef.current&&(showPrefixSuffix&&prefix?prefixRef.current&&showPrefixSuffix&&(inputRef.current.style.paddingLeft=`${prefixRef.current.getBoundingClientRect().width}px`):inputRef.current.style.paddingLeft="0px")},updateInputRightPadding=()=>{inputRef.current&&(showPrefixSuffix&&suffix&&"left"!==align?suffixRef.current&&showPrefixSuffix&&"right"===align&&(inputRef.current.style.paddingRight=`${suffixRef.current.getBoundingClientRect().width}px`):inputRef.current.style.paddingRight="0px")};return e.useLayoutEffect(()=>{const nullHandledValue=handleInput(null);function oninput(){const{value,selectionStart,selectionEnd}=this,inputData=handleInput({value,selectionStart,selectionEnd});placeholder&&!changeInputData.isInputDataDifferent(nullHandledValue,inputData)?changeInputData.changeInputData(this,{...inputData,value:""}):changeInputData.changeInputData(this,inputData)}if("date"!==type&&inputRef.current){const node=inputRef.current;node.addEventListener("input",oninput);const{value,selectionStart,selectionEnd}=node,inputData=handleInput({value,selectionStart,selectionEnd});return placeholder&&!changeInputData.isInputDataDifferent(nullHandledValue,inputData)?changeInputData.changeInputData(node,{...inputData,value:""}):changeInputData.changeInputData(node,inputData),()=>{node.removeEventListener("input",oninput)}}},[handleInput,inputRef.current,placeholder]),e.useLayoutEffect(()=>{var newValue;newValue=innerValue,sizerRef.current&&(sizerRef.current.innerHTML=newValue||placeholder||""),(newValue=>{setPrefixSuffix(!!newValue)})(innerValue)},[innerValue,placeholder,inputRef.current,sizerRef.current]),e.useLayoutEffect(()=>updateInputLeftPadding(),[prefix,dimension,prefixRef.current,inputRef.current,showPrefixSuffix]),e.useLayoutEffect(()=>updateInputRightPadding(),[suffix,dimension,suffixRef.current,inputRef.current,showPrefixSuffix,align]),e.useLayoutEffect(()=>{if(prefixRef.current){const resizeObserver=new ResizeObserver(entries=>{entries.forEach(()=>{updateInputLeftPadding()})});return resizeObserver.observe(prefixRef.current),()=>{resizeObserver.disconnect()}}},[prefixRef.current,inputRef.current,showPrefixSuffix,prefix]),e.useLayoutEffect(()=>{if(suffixRef.current){const resizeObserver=new ResizeObserver(entries=>{entries.forEach(()=>{updateInputRightPadding()})});return resizeObserver.observe(suffixRef.current),()=>{resizeObserver.disconnect()}}},[suffixRef.current,inputRef.current,showPrefixSuffix,suffix,align]),jsxRuntime.jsxs(jsxRuntime.Fragment,{children:[jsxRuntime.jsxs(HiddenContent,{$iconCount:iconCount,$dimension:dimension,children:[prefix&&showPrefixSuffix&&jsxRuntime.jsxs(Prefix,{ref:prefixRef,disabled:props.disabled,$align:align,children:[prefix,!hideSpaceAfterPrefix&&jsxRuntime.jsx(jsxRuntime.Fragment,{children:" "})]}),jsxRuntime.jsx(Sizer,{ref:sizerRef,$hasPrefix:!!prefix,$align:align}),suffix&&showPrefixSuffix&&jsxRuntime.jsxs(Suffix,{ref:suffixRef,disabled:props.disabled,$align:align,children:[!hideSpaceBeforeSuffix&&jsxRuntime.jsx(jsxRuntime.Fragment,{children:" "}),suffix]})]}),jsxRuntime.jsx(Input,{...props,$dimension:dimension,ref:refSetter.refSetter(ref,inputRef),placeholder,type:"text","data-status":status,$align:align}),jsxRuntime.jsx(BorderedDiv,{$status:status})]})});exports.AutoSizeInput=AutoSizeInput,exports.BorderedDiv=BorderedDiv,exports.horizontalPaddingValue=horizontalPaddingValue,exports.iconSizeValue=iconSizeValue;
