// @admiral-ds/react-ui/dist/components/TabMenuComponent/containers/HorizontalCardTabMenuContainer.cjs.js
"use strict";function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(require("styled-components"));const HorizontalCardTabMenuContainer=styled__default.default.div`
  position: relative;
  display: flex;
  flex: 1 1 auto;
  flex-wrap: nowrap;
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  box-shadow: inset 0 -1px 0 0
    ${p=>p.$showUnderline?`var(--admiral-color-Neutral_Neutral20, ${p.theme.color["Neutral/Neutral 20"]})`:"transparent"};
  overflow: hidden;
`;HorizontalCardTabMenuContainer.displayName="HorizontalCardTabMenuContainer";const CardTabMenuHorizontalLeadingGapWrapper=styled__default.default.div`
  padding: 0 ${p=>p.$horizontalPaddingSize}px;
  box-shadow: inset 0 -1px 0 0
    ${p=>`var(--admiral-color-Neutral_Neutral20, ${p.theme.color["Neutral/Neutral 20"]})`};
`;CardTabMenuHorizontalLeadingGapWrapper.displayName="CardTabMenuHorizontalLeadingGapWrapper",exports.CardTabMenuHorizontalLeadingGapWrapper=CardTabMenuHorizontalLeadingGapWrapper,exports.HorizontalCardTabMenuContainer=HorizontalCardTabMenuContainer;
