// @admiral-ds/react-ui/dist/components/Calendar/styled/CalendarComponent.cjs.js
"use strict";var styled=require("styled-components"),typography=require("../../Typography/typography.cjs.js");require("../../Typography/defaultTypographyMixin.cjs.js");var borderRadius=require("../../themes/borderRadius.cjs.js"),parseShadowFromTheme=require("../../common/utils/parseShadowFromTheme.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}const CalendarComponent=_interopDefault(styled).default.div`
  position: relative;
  box-sizing: border-box;
  text-align: center;
  border: 0 none;
  z-index: 0; /* to fix range rounded corners fill */

  padding: ${({$yearsView,$monthsView})=>$yearsView?"20px 12px 16px":$monthsView?"20px 16px 4px":"20px 12px 12px"};

  width: ${284}px;
  background: var(--admiral-color-Special_ElevatedBG, ${p=>p.theme.color["Special/Elevated BG"]});
  ${typography.typography["Body/Body 2 Long"]}
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});
  border-radius: var(--admiral-border-radius-Medium, ${p=>borderRadius.mediumGroupBorderRadius(p.theme.shape)});
  box-shadow: var(--admiral-box-shadow-Shadow08, ${p=>parseShadowFromTheme.parseShadow(p.theme.shadow["Shadow 08"])});
`;exports.CalendarComponent=CalendarComponent;
