// @admiral-ds/react-ui/dist/components/Calendar/styled/PanelDate.cjs.js
"use strict";var styled=require("styled-components"),typography=require("../../Typography/typography.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}require("../../Typography/defaultTypographyMixin.cjs.js");var styled__default=_interopDefault(styled);const panelItemStyle=styled.css`
  cursor: pointer;
  display: flex;
  align-items: center;
  box-sizing: border-box;
  height: ${"32px"};
  padding: ${"4px 8px"};
  margin: 0;
  border-radius: ${"16px"};
  ${typography.typography["Subtitle/Subtitle 2"]}
  color: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
  background: ${p=>p.$view?`var(--admiral-color-Opacity_Focus, ${p.theme.color["Opacity/Focus"]})`:"transparent"};
  &:hover {
    background: var(--admiral-color-Opacity_Hover, ${p=>p.theme.color["Opacity/Hover"]});
  }
`,Month=styled__default.default.h6`
  ${panelItemStyle}
`,Year=styled__default.default.h6`
  ${panelItemStyle};
  width: ${"54px"};
`,PanelDate=styled__default.default.div`
  display: flex;
  justify-content: center;
  width: 100%;
`;exports.Month=Month,exports.PanelDate=PanelDate,exports.Year=Year;
