// @admiral-ds/react-ui/dist/components/input/Select/styled.cjs.js
"use strict";var styled=require("styled-components"),typography=require("../../Typography/typography.cjs.js");require("../../Typography/defaultTypographyMixin.cjs.js");var constants=require("./constants.cjs.js"),borderRadius=require("../../themes/borderRadius.cjs.js"),parseShadowFromTheme=require("../../common/utils/parseShadowFromTheme.cjs.js"),Container=require("../Container.cjs.js"),index=require("../../Menu/index.cjs.js"),MenuItem=require("../../Menu/MenuItem.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const BorderedDiv=styled__default.default.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  margin: 0;
  pointer-events: none;
  overflow: hidden;
  min-width: 0;

  background: none;
  border-width: 1px;
  border-style: solid;
  border-radius: inherit;
`,rowHeightStyle=styled.css`
  min-height: ${({$multiple,$minRowCount,$dimension})=>$multiple?$minRowCount?24*$minRowCount+4*($minRowCount-1)+"px":"24px":("s"===$dimension?20:24)+"px"};

  max-height: ${({$multiple,$maxRowCount,$opened,$idleHeight})=>$multiple?$maxRowCount?24*$maxRowCount+4*($maxRowCount-1)+"px":$opened||"fixed"!==$idleHeight?"none":"24px":"none"};
`,chipsShiftStyle=styled.css`
  > * {
    margin-left: ${`-${constants.COUNTER_WIDTH+constants.CHIP_OFFSET}px`};
  }

  padding-left: ${`${constants.COUNTER_WIDTH+constants.CHIP_OFFSET}px`};
`,ValueWrapper=styled__default.default.div`
  flex: 1 1 auto;
  display: flex;
  overflow: hidden;

  ${({$multiple,$isEmpty})=>$multiple&&!$isEmpty&&chipsShiftStyle}

  gap: 4px;
  flex-wrap: ${({$multiple})=>$multiple?"wrap":"unset"};
  align-items: center;
  align-content: flex-start;

  ${props=>"s"===props.$dimension?typography.typography["Body/Body 2 Long"]:typography.typography["Body/Body 1 Long"]}
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});

  ${rowHeightStyle}
  [data-disabled='true'] &&& {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }
`,StringValueWrapper=styled__default.default.div`
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`,ieFixes=styled.css`
  ::-ms-clear,
  ::-ms-reveal {
    display: none;
  }
`,Input=styled__default.default.input`
  outline: none;
  appearance: none;

  box-sizing: border-box;
  flex: 1 1 auto;
  width: 1px;
  border: none;
  background: transparent;
  text-overflow: ellipsis;
  padding: 0;

  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});

  ${({$dimension})=>"s"===$dimension?typography.typography["Body/Body 2 Long"]:typography.typography["Body/Body 1 Long"]}
  &::placeholder {
    color: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }

  &:read-only {
    cursor: inherit;
  }

  &:disabled,
  &:disabled::placeholder {
    cursor: inherit;
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }

  [data-disable-copying] & {
    user-select: none;
    pointer-events: none;
  }

  height: ${({$dimension,$isMultiple})=>{return(multiple=$isMultiple,"s"!==$dimension||multiple?24:20)+"px";var multiple}};

  ${ieFixes};
`,disableEventMixin=styled.css`
  pointer-events: none;
  cursor: default;
`,disabledStyle=styled.css`
  && > *:not(${ValueWrapper}) {
    pointer-events: none;
  }

  & ${BorderedDiv} {
    border-color: transparent;
  }
`,focusedStyle=styled.css`
  ${BorderedDiv} {
    border-width: 2px;
  }
`,IconPanel=styled__default.default.div`
  flex: 0 0 auto;

  display: flex;
  align-items: center;

  padding: ${({$dimension,$multiple})=>"s"===$dimension&&$multiple?"2px 0":"0"};

  & > * {
    margin-left: 8px;
    display: block;
    width: ${({$dimension})=>"s"===$dimension?20:24}px;
    height: ${({$dimension})=>"s"===$dimension?20:24}px;
  }

  [data-disabled='true'] &&& {
    & *[fill^='#'] {
      fill: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
    }
  }
`,SelectWrapper=styled__default.default.div`
  position: relative;
  box-sizing: border-box;
  display: flex;
  align-items: ${p=>p.$multiple?"flex-start":"center"};
  cursor: ${({disabled,$readonly,$isLoading})=>disabled?"not-allowed":$readonly||$isLoading?"default":"pointer"};

  padding: ${({$dimension,$multiple})=>{switch($dimension){case"xl":return"16px 16px";case"s":return $multiple?"4px 12px":"6px 12px";default:return"8px 16px"}}};

  background: ${({theme,disabled,$readonly})=>disabled||$readonly?`var(--admiral-color-Neutral_Neutral10, ${theme.color["Neutral/Neutral 10"]})`:`var(--admiral-color-Neutral_Neutral00, ${theme.color["Neutral/Neutral 00"]})`};

  ${({disabled,$readonly})=>$readonly||disabled?disabledStyle:""};
  ${({$focused,$readonly,$isLoading})=>!$focused||$readonly||$isLoading?"":focusedStyle};

  & ${BorderedDiv} {
    border-color: ${p=>p.disabled||p.$readonly?"transparent":p.$focused&&!p.$isLoading?`var(--admiral-color-Primary_Primary60Main, ${p.theme.color["Primary/Primary 60 Main"]})`:`var(--admiral-color-Neutral_Neutral40, ${p.theme.color["Neutral/Neutral 40"]})`};
  }

  &:hover ${BorderedDiv} {
    ${p=>!p.disabled&&!p.$readonly&&!p.$focused&&!p.$isLoading&&`\n      border-color: var(--admiral-color-Neutral_Neutral60, ${p.theme.color["Neutral/Neutral 60"]});\n    `};
  }

  &[data-status='success'] {
    ${p=>!p.disabled&&!p.$readonly&&!p.$isLoading&&`\n      ${BorderedDiv} {\n      border-color: var(--admiral-color-Success_Success50Main, ${p.theme.color["Success/Success 50 Main"]});\n      }\n      &:hover ${BorderedDiv} {\n        border-color: var(--admiral-color-Success_Success60, ${p.theme.color["Success/Success 60"]});\n      }\n    `}
  }

  &[data-status='error'],
  &:user-invalid {
    ${p=>!p.disabled&&!p.$readonly&&!p.$isLoading&&`\n      ${BorderedDiv} {\n        border-color: var(--admiral-color-Error_Error60Main, ${p.theme.color["Error/Error 60 Main"]});\n      }\n  \n      &:hover ${BorderedDiv} {\n        border-color: var(--admiral-color-Error_Error70, ${p.theme.color["Error/Error 70"]});\n      }\n    `}
  }

  border-radius: ${p=>p.$skeleton?0:`var(--admiral-border-radius-Medium, ${borderRadius.mediumGroupBorderRadius(p.theme.shape)})`};

  ${({$skeleton})=>$skeleton&&Container.skeletonMixin};
  ${({$skeleton})=>$skeleton&&disableEventMixin};
`,OptionWrapper=styled__default.default.div`
  flex: 0 0 auto;

  word-break: break-word;
  display: flex;
  align-items: center;
  padding: ${({$dimension})=>{switch($dimension){case"xl":return"12px 16px";case"s":return"6px 12px";default:return"8px 16px"}}};

  background-color: transparent;

  ${({$dimension})=>{switch($dimension){case"xl":return typography.typography["Body/Body 1 Long"];case"s":return typography.typography["Body/Body 2 Long"];default:return typography.typography["Body/Body 1 Short"]}}}
`,ShadowContainerMixin=styled.css`
  ${typography.typography["Body/Body 1 Short"]}
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});
  background-color: var(--admiral-color-Special_ElevatedBG, ${p=>p.theme.color["Special/Elevated BG"]});
  border-radius: var(--admiral-border-radius-Medium, ${p=>borderRadius.mediumGroupBorderRadius(p.theme.shape)});
  box-shadow: var(--admiral-box-shadow-Shadow08, ${p=>parseShadowFromTheme.parseShadow(p.theme.shadow["Shadow 08"])});
  overflow: auto;
`,StyledMenu=styled__default.default(index.Menu)`
  ${ShadowContainerMixin}
`,EmptyMessageWrapper=styled__default.default.div`
  user-select: none;
`,CustomOptionWrapper=styled__default.default(MenuItem.MenuItem)`
  justify-content: flex-start;
  flex-wrap: nowrap;
  white-space: pre-wrap;
  ${props=>props.$selected&&!props.$hovered&&!props.$preselected&&props.$multiple&&"background-color: transparent;"}
`;exports.BorderedDiv=BorderedDiv,exports.CustomOptionWrapper=CustomOptionWrapper,exports.EmptyMessageWrapper=EmptyMessageWrapper,exports.IconPanel=IconPanel,exports.Input=Input,exports.OptionWrapper=OptionWrapper,exports.SelectWrapper=SelectWrapper,exports.StringValueWrapper=StringValueWrapper,exports.StyledMenu=StyledMenu,exports.ValueWrapper=ValueWrapper;
