// @admiral-ds/react-ui/dist/components/input/TreeSelect/styled.cjs.js
"use strict";var index$2=require("../../Chips/index.cjs.js"),parseShadowFromTheme=require("../../common/utils/parseShadowFromTheme.cjs.js"),index$1=require("../MultiInput/index.cjs.js"),index=require("../../Menu/index.cjs.js");require("../../themes/common/locales.cjs.js");var typography=require("../../Typography/typography.cjs.js");require("../../Typography/defaultTypographyMixin.cjs.js");var borderRadius=require("../../themes/borderRadius.cjs.js"),styled=require("styled-components");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const ShadowContainerMixin=styled.css`
  ${typography.typography["Body/Body 1 Short"]}
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});
  background-color: var(--admiral-color-Special_ElevatedBG, ${p=>p.theme.color["Special/Elevated BG"]});
  border-radius: var(--admiral-border-radius-Medium, ${p=>borderRadius.mediumGroupBorderRadius(p.theme.shape)});
  box-shadow: var(--admiral-box-shadow-Shadow08, ${p=>parseShadowFromTheme.parseShadow(p.theme.shadow["Shadow 08"])});
  overflow: auto;
`,StyledMenu=styled__default.default(index.Menu)`
  ${ShadowContainerMixin}
`,StyledMultiInput=styled__default.default(index$1.MultiInput)`
  cursor: pointer;
  & .wrapper-options input {
    display: none;
  }
`,disabledChipStyle=styled.css`
  pointer-events: auto;
  &:hover {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }
`,hoverChipStyle=styled.css`
  &:hover {
    background-color: var(--admiral-color-Neutral_Neutral10, ${p=>p.theme.color["Neutral/Neutral 10"]});
  }
`,StyledChip=styled__default.default(index$2.Chips)`
  ${({disabled,readOnly})=>disabled?disabledChipStyle:readOnly?null:hoverChipStyle}
`;exports.StyledChip=StyledChip,exports.StyledMenu=StyledMenu,exports.StyledMultiInput=StyledMultiInput;
