// @admiral-ds/react-ui/dist/components/input/TextArea/index.cjs.js
"use strict";var e=require("react"),jsxRuntime=require("react/jsx-runtime"),styled=require("styled-components"),index$2=require("../../themes/light/index.cjs.js");require("../../themes/common/locales.cjs.js");var typography=require("../../Typography/typography.cjs.js");require("../../Typography/defaultTypographyMixin.cjs.js");var changeInputData=require("../../common/dom/changeInputData.cjs.js"),refSetter=require("../../common/utils/refSetter.cjs.js"),CloseOutline=require("../../../node_modules/@admiral-ds/icons/build/service/CloseOutline.svg.cjs.js"),CopyOutline=require("../../../node_modules/@admiral-ds/icons/build/documents/CopyOutline.svg.cjs.js"),InputIconButton=require("../../InputIconButton.cjs.js"),Container=require("../Container.cjs.js"),index=require("../../Scrollbar/index.cjs.js"),index$1=require("../../TooltipHOC/index.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const iconSizeValue=props=>{switch(props.$dimension){case"xl":default:return 24;case"s":return 20}},verticalPaddingValue=props=>{switch(props.$dimension){case"xl":return 16;case"m":default:return 8;case"s":return 6}},horizontalPaddingValue=props=>{switch(props.$dimension){case"xl":case"m":default:return 16;case"s":return 12}},extraPadding=styled.css`
  padding-right: ${props=>horizontalPaddingValue(props)+(props.$iconsAfterCount?iconSizeValue(props)+8:0)}px;
`,disabledColors=styled.css`
  background-color: var(--admiral-color-Neutral_Neutral10, ${p=>p.theme.color["Neutral/Neutral 10"]});
  border-color: transparent;
`,BorderedDiv=styled__default.default.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  margin: 0;
  pointer-events: none;
  overflow: hidden;
  min-width: 0;

  background: none;
  border: 1px solid var(--admiral-color-Neutral_Neutral40, ${p=>p.theme.color["Neutral/Neutral 40"]});
  border-radius: inherit;

  [data-status='error'] &&& {
    border: 1px solid var(--admiral-color-Error_Error60Main, ${p=>p.theme.color["Error/Error 60 Main"]});
  }

  [data-status='success'] &&& {
    border: 1px solid var(--admiral-color-Success_Success50Main, ${p=>p.theme.color["Success/Success 50 Main"]});
  }

  [data-read-only] &&& {
    border-color: transparent;
  }
`,colorsBorderAndBackground=styled.css`
  background-color: var(--admiral-color-Neutral_Neutral00, ${p=>p.theme.color["Neutral/Neutral 00"]});

  &:focus + ${BorderedDiv} {
    border: 2px solid var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
  }

  &&&:disabled + ${BorderedDiv}, [data-read-only] &&& + ${BorderedDiv} {
    border-color: transparent;
  }

  &:hover:not(:focus) + ${BorderedDiv} {
    border-color: ${p=>p.disabled?"transparent":`var(--admiral-color-Neutral_Neutral60, ${p.theme.color["Neutral/Neutral 60"]})`};
  }

  &:user-invalid + ${BorderedDiv}, &:user-invalid:hover + ${BorderedDiv} {
    border: 1px solid var(--admiral-color-Error_Error60Main, ${p=>p.theme.color["Error/Error 60 Main"]});
  }

  [data-status='error']
    &&&:hover:not(:disabled)
    + ${BorderedDiv},
    &:user-invalid:hover:not(:disabled)
    + ${BorderedDiv} {
    border: 1px solid var(--admiral-color-Error_Error70, ${p=>p.theme.color["Error/Error 70"]});
  }

  [data-status='success'] &&&:hover:not(:disabled) + ${BorderedDiv} {
    border: 1px solid var(--admiral-color-Success_Success60, ${p=>p.theme.color["Success/Success 60"]});
  }

  [data-status='error']
    &&&:focus:not(:disabled)
    + ${BorderedDiv},
    &:user-invalid:focus:not(:disabled)
    + ${BorderedDiv} {
    border: 2px solid var(--admiral-color-Error_Error60Main, ${p=>p.theme.color["Error/Error 60 Main"]});
  }

  [data-status='success'] &&&:focus:not(:disabled) + ${BorderedDiv} {
    border: 2px solid var(--admiral-color-Success_Success50Main, ${p=>p.theme.color["Success/Success 50 Main"]});
  }

  [data-read-only] &&&,
  &&&:disabled {
    ${disabledColors}
  }

  [data-read-only] &&&:hover + ${BorderedDiv}, [data-read-only] &&&:focus + ${BorderedDiv} {
    border-color: transparent;
  }
`,ieFixes=styled.css`
  ::-ms-clear,
  ::-ms-reveal {
    display: none;
  }
`,textBlockStyleMixin=styled.css`
  outline: none;
  appearance: none;
  border-radius: inherit;
  border: none;
  box-sizing: border-box;
  margin: 0;
  padding: ${verticalPaddingValue}px ${horizontalPaddingValue}px;
  overflow-wrap: break-word;

  ${props=>"s"===props.$dimension?typography.typography["Body/Body 2 Long"]:typography.typography["Body/Body 1 Long"]}
  ${colorsBorderAndBackground}
  ${extraPadding}
`,HiddenSpanContainer=styled__default.default.div`
  ${index.hideNativeScrollbarsCss}
  ${textBlockStyleMixin}

  [data-disable-copying] & {
    cursor: default;
  }
`,Text=styled__default.default.textarea`
  ${index.hideNativeScrollbarsCss}
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  resize: none;

  flex: 1 1 auto;
  min-width: 10px;
  background: transparent;

  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});

  ${props=>"s"===props.$dimension?typography.typography["Body/Body 2 Long"]:typography.typography["Body/Body 1 Long"]}
  &::placeholder {
    color: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }

  &:disabled::placeholder,
  &:disabled {
    cursor: not-allowed;
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }

  [data-disable-copying] & {
    user-select: none;
    pointer-events: none;
  }

  ${textBlockStyleMixin}
  ${ieFixes}
`,IconPanel=styled__default.default.div`
  position: absolute;
  top: 0;
  bottom: 0;
  right: 0;

  display: flex;
  flex-direction: column;
  align-items: center;

  margin-right: ${horizontalPaddingValue}px;
  padding-top: ${verticalPaddingValue}px;

  & > * {
    display: block;
    width: ${iconSizeValue}px;
    height: ${iconSizeValue}px;
  }
`;function defaultHandleInput(newInputData){return newInputData}const stopEvent=e=>e.preventDefault(),textAreaHeight=(rows,$dimension)=>rows*("s"===$dimension?20:24)+2*verticalPaddingValue({$dimension}),StyledContainer=styled__default.default(Container.Container)`
  min-height: ${p=>textAreaHeight(p.$rows,p.$dimension)}px;
  ${p=>p.$maxRows?`max-height: ${textAreaHeight(p.$maxRows,p.$dimension)}px;`:""}
  ${p=>p.$autoHeight?"":`height: ${textAreaHeight(p.$rows,p.$dimension)}px;`}
  ${p=>p.disabled?"cursor: not-allowed;":""}
`,CopyIconButton=e.forwardRef((props,ref)=>jsxRuntime.jsx("div",{ref,children:jsxRuntime.jsx(InputIconButton.InputIconButton,{...props})})),TooltipedInputIconButton=index$1.TooltipHoc(CopyIconButton);function toHtmlString(value){return String(value||"").replace(/&/g,"&amp;").replace(/</g,"&lt;").replace(/>/g,"&gt;").replace(/"/g,"&quot;").replace(/'/g,"&#39;").replace(/(\r?\n)$/g,"<br /><br />").replace(/\r?\n/g,"<br /> ")}const nothing=()=>{},TextArea=e.forwardRef(({rows=3,maxRows,value,displayClearIcon,displayCopyIcon,status,handleInput=defaultHandleInput,containerRef,icons,iconsAfter,children,className,autoHeight,skeleton=!1,dimension="m",disableCopying,locale,clearIconPropsConfig=nothing,copyIconPropsConfig=nothing,...props},ref)=>{const theme=styled.useTheme()||index$2.LIGHT_THEME,inputRef=e.useRef(null),[contentNode,setContentNode]=e.useState(null),hiddenDivRef=e.useRef(null),iconArray=e.Children.toArray(iconsAfter||icons),copyText=locale?.copyTextMessage||theme.locales[theme.currentLocale]?.textArea?.copyTextMessage,copiedText=locale?.copiedMessage||theme.locales[theme.currentLocale]?.textArea?.copiedMessage,[tooltipText,setTooltipText]=e.useState(copyText),handleIconMouseDown=e=>{e.preventDefault()},clearIconProps={icon:CloseOutline.ReactComponent,onMouseDown:handleIconMouseDown,onClick:()=>{inputRef.current&&changeInputData.changeInputData(inputRef.current,{value:""})},"aria-hidden":!0},copyIconProps={tooltipDimension:"s",renderContent:()=>tooltipText,icon:CopyOutline.ReactComponent,onMouseDown:handleIconMouseDown,onClick:()=>{document.hasFocus()||window.focus(),inputRef.current&&(navigator.clipboard.writeText(inputRef.current.value),setTooltipText(copiedText),setTimeout(()=>setTooltipText(copyText),2e3))},"aria-hidden":!0};!props.readOnly&&inputRef?.current?.value&&(displayClearIcon?iconArray.unshift(e.createElement(InputIconButton.InputIconButton,{...clearIconProps,...clearIconPropsConfig(clearIconProps),key:"clear-icon"})):displayCopyIcon&&iconArray.unshift(e.createElement(TooltipedInputIconButton,{...copyIconProps,...copyIconPropsConfig(copyIconProps),key:"copy-icon"})));const iconCount=iconArray.length,inputData=null!=value?handleInput({value:String(value)}):{};return e.useLayoutEffect(()=>{function oninput(){const{value,selectionStart,selectionEnd}=this,inputData=handleInput({value,selectionStart,selectionEnd});changeInputData.changeInputData(this,inputData)}if(inputRef.current){const node=inputRef.current;node.addEventListener("input",oninput);const{value,selectionStart,selectionEnd}=node,inputData=handleInput({value,selectionStart,selectionEnd});return changeInputData.changeInputData(node,inputData),()=>{node.removeEventListener("input",oninput)}}},[handleInput]),e.useEffect(()=>{function oninput(){const{value}=this,hiddenDiv=hiddenDivRef.current;hiddenDiv&&(hiddenDiv.innerHTML=toHtmlString(value),this.style.overflowY=hiddenDiv.clientHeight<hiddenDiv.scrollHeight?"":"hidden")}if(autoHeight&&inputRef.current&&hiddenDivRef.current){const node=inputRef.current;return hiddenDivRef.current.innerHTML=toHtmlString(node.value),node.addEventListener("input",oninput),node.style.overflowY=hiddenDivRef.current.clientHeight<hiddenDivRef.current.scrollHeight?"":"hidden",()=>{node.removeEventListener("input",oninput),node.style.overflowY=""}}},[autoHeight,inputData.value,props.defaultValue]),jsxRuntime.jsxs(StyledContainer,{className,ref:containerRef,"data-read-only":!!props.readOnly||void 0,"data-status":status,$skeleton:skeleton,disabled:props.disabled,"data-disable-copying":!!disableCopying||void 0,$autoHeight:!!autoHeight,$rows:rows,$maxRows:maxRows,$dimension:dimension,...disableCopying&&{onMouseDown:stopEvent},children:[jsxRuntime.jsx(HiddenSpanContainer,{ref:hiddenDivRef,$dimension:dimension,disabled:props.disabled,$iconsAfterCount:iconCount}),jsxRuntime.jsx(Text,{ref:refSetter.refSetter(ref,inputRef,node=>setContentNode(node)),...props,$dimension:dimension,$iconsAfterCount:iconCount,$autoHeight:autoHeight,value:inputData.value}),jsxRuntime.jsx(BorderedDiv,{}),jsxRuntime.jsx(index.Scrollbars,{contentNode}),iconCount>0&&jsxRuntime.jsx(IconPanel,{disabled:props.disabled,$dimension:dimension,children:iconArray}),children]})});TextArea.displayName="TextArea",exports.TextArea=TextArea;
