// @admiral-ds/react-ui/dist/components/input/FileInput/FileItem.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),styled=require("styled-components"),borderRadius=require("../../themes/borderRadius.cjs.js"),typography=require("../../Typography/typography.cjs.js");require("../../Typography/defaultTypographyMixin.cjs.js");var utils=require("./utils.cjs.js"),index=require("../../Spinner/index.cjs.js"),EyeOutline=require("../../../node_modules/@admiral-ds/icons/build/service/EyeOutline.svg.cjs.js"),style=require("./style.cjs.js"),index$2=require("../../Tooltip/index.cjs.js"),checkOverflow=require("../../common/utils/checkOverflow.cjs.js"),index$1=require("../../IconPlacement/index.cjs.js"),keyboardKey=require("../../common/keyboardKey.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const disabledStyles=styled.css`
  pointer-events: none;
`,Container=styled__default.default.div`
  display: flex;
  flex-direction: column;
  overflow: hidden;
  cursor: ${({disabled})=>disabled?"not-allowed":"default"};
  ${p=>p.$filesLayoutCssMixin}
`,statusMixin=styled.css`
  border-color: ${p=>"Error"===p.$status?`var(--admiral-color-Error_Error60Main, ${p.theme.color["Error/Error 60 Main"]})`:"Queue"===p.$status?`var(--admiral-color-Neutral_Neutral30, ${p.theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral40, ${p.theme.color["Neutral/Neutral 40"]})`};
  color: ${p=>"Queue"===p.$status?`var(--admiral-color-Neutral_Neutral30, ${p.theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral90, ${p.theme.color["Neutral/Neutral 90"]})`};
`,PreviewWrapper=styled__default.default.div`
  padding: 0;
  box-sizing: border-box;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  border-radius: var(--admiral-border-radius-Medium, ${p=>borderRadius.mediumGroupBorderRadius(p.theme.shape)});
  border-width: 1px;
  border-style: solid;
  ${typography.typography["Body/Body 2 Long"]};
  ${statusMixin};
`,sizeMixin=styled.css`
  align-items: center;
  justify-content: space-between;
`,hoveredFileTypeIconCss=styled.css`
  &:hover {
    &:not(:disabled) {
      &::before {
        content: '';
        position: absolute;
        border-radius: var(--admiral-border-radius-Medium, ${p=>borderRadius.mediumGroupBorderRadius(p.theme.shape)});
        left: 50%;
        top: 50%;
        transform: translate(-50%, -50%);
        width: 100%;
        height: 100%;
        opacity: 0.6;
        background-color: var(
          --admiral-color-Special_DarkStaticNeutral00,
          ${p=>p.theme.color["Special/Dark Static Neutral 00"]}
        );
      }
      & svg {
        visibility: visible;
      }
      cursor: pointer;
    }
  }
`,getColor=styled.css`
  ${({theme,$status})=>"Queue"===$status?`var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral50, ${theme.color["Neutral/Neutral 50"]})`}
`,ImagePreview=styled__default.default.div`
  min-width: 40px;
  width: 40px;
  height: 40px;
  border-radius: var(--admiral-border-radius-Medium, ${p=>borderRadius.mediumGroupBorderRadius(p.theme.shape)});
  margin-right: 8px;
  overflow: hidden;

  > img {
    object-fit: cover;
    height: 100%;
    width: 100%;
  }
`,IconWrapper=styled__default.default.div`
  position: relative;
  margin-right: 8px;
  border-radius: var(--admiral-border-radius-Medium, ${p=>borderRadius.mediumGroupBorderRadius(p.theme.shape)});
  width: ${style.FILE_ITEM_PREVIEW_ICON_SIZE_XL};
  height: ${style.FILE_ITEM_PREVIEW_ICON_SIZE_XL};

  & *[fill^='#'] {
    fill: ${getColor};
  }

  ${p=>"Queue"!==p.$status&&p.$showHover?hoveredFileTypeIconCss:disabledStyles};

  &:focus-visible {
    outline-offset: 2px;
    outline: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]}) solid 2px;
  }
`,StyledEyeOutline=styled__default.default(EyeOutline.ReactComponent)`
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  width: ${style.FILE_ITEM_FUNCTIONAL_ICON_SIZE_XL};
  height: ${style.FILE_ITEM_FUNCTIONAL_ICON_SIZE_XL};
  visibility: hidden;

  & *[fill^='#'] {
    fill: var(--admiral-color-Special_StaticWhite, ${p=>p.theme.color["Special/Static White"]});
  }
`,Content=styled__default.default.div`
  ${p=>"m"===p.$dimension&&sizeMixin};
  display: flex;
  flex-direction: ${p=>"m"===p.$dimension?"row":"column"};
  min-width: 0;
  overflow: hidden;
`,FileInfoBlock=styled__default.default.div`
  display: ${p=>"m"===p.$dimension?"block":"flex"};
  box-sizing: border-box;
  align-items: center;
  overflow: hidden;
  height: ${p=>"xl"===p.$dimension?"40px":"20px"};
  padding: ${p=>"xl"===p.$dimension?style.FILE_ITEM_WRAPPER_PADDING_XL:style.FILE_ITEM_WRAPPER_PADDING_M};
  height: ${p=>"xl"===p.$dimension?style.FILE_ITEM_WRAPPER_HEIGHT_XL:style.FILE_ITEM_WRAPPER_HEIGHT_M};
`,FileName=styled__default.default.div`
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
`,FileInfo=styled__default.default.span`
  color: ${getColor};
  margin-left: ${p=>"xl"===p.$dimension?"0":"4px"};
  white-space: nowrap;
`,FunctionalBlock=styled__default.default.div`
  display: flex;
  flex: 0 0 auto;
  margin-left: 8px;
  align-items: center;
`,StyledSpinner=styled__default.default(index.Spinner)`
  margin-right: 8px;
`,ErrorBlock=styled__default.default.div`
  margin-top: 8px;
  color: var(--admiral-color-Error_Error60Main, ${p=>p.theme.color["Error/Error 60 Main"]});
  ${typography.typography["Body/Body 2 Short"]};
  height: ${p=>"xl"===p.$dimension?style.ERROR_BLOCK_HEIGHT_XL:style.ERROR_BLOCK_HEIGHT_M};
`,CloseButton=styled__default.default(index$1.CloseIconPlacementButton)`
  margin-right: ${p=>"lSmall"===p.dimension?"8px":"16px"};
`,FileItem=e.forwardRef(({fileId,fileName,fileType,fileSize,status,errorMessage,previewImageURL,onCloseIconClick,onPreviewIconClick,dimension,filesLayoutCssMixin,formatFileSizeInfo=utils.formatFileSize,formatFileTypeInfo=utils.formatFileType,formatFileTypeIcon=utils.getFileTypeIcon,...props},ref)=>{const PreviewIcon=formatFileTypeIcon(fileType),fileInfo=`${formatFileTypeInfo(fileType)}・${formatFileSizeInfo(fileSize)}`,previewWrapperRef=e.useRef(null),titleRef=e.useRef(null),[tooltipVisible,setTooltipVisible]=e.useState(!1),[fileNameOverflow,setFileNameOverflow]=e.useState(!1);e.useEffect(()=>{const element=titleRef.current;element&&checkOverflow.checkOverflow(element)!==fileNameOverflow&&setFileNameOverflow(checkOverflow.checkOverflow(element))},[tooltipVisible,setFileNameOverflow]),e.useEffect(()=>{function show(){setTooltipVisible(!0)}function hide(){setTooltipVisible(!1)}const title=titleRef.current;if(title)return title.addEventListener("mouseenter",show),title.addEventListener("mouseleave",hide),()=>{title.removeEventListener("mouseenter",show),title.removeEventListener("mouseleave",hide)}},[setTooltipVisible]);const disabled="Queue"===status;return jsxRuntime.jsxs(Container,{ref,disabled,$dimension:dimension,$filesLayoutCssMixin:filesLayoutCssMixin,children:[jsxRuntime.jsxs(PreviewWrapper,{...props,ref:previewWrapperRef,$status:status,$dimension:dimension,children:[jsxRuntime.jsxs(FileInfoBlock,{$dimension:dimension,children:["xl"===dimension&&jsxRuntime.jsxs(IconWrapper,{$status:status,$showHover:!!onPreviewIconClick,onClick:()=>{onPreviewIconClick?.(fileId)},onKeyDown:e=>{const code=keyboardKey.keyboardKey.getCode(e);code!==keyboardKey.keyboardKey.Enter&&code!==keyboardKey.keyboardKey[" "]||onPreviewIconClick?.(fileId)},tabIndex:onPreviewIconClick?0:-1,children:[previewImageURL?jsxRuntime.jsx(ImagePreview,{children:jsxRuntime.jsx("img",{src:previewImageURL,alt:""})}):jsxRuntime.jsx(PreviewIcon,{}),jsxRuntime.jsx(StyledEyeOutline,{})]}),jsxRuntime.jsxs(Content,{$dimension:dimension,children:[jsxRuntime.jsx(FileName,{ref:titleRef,children:fileName}),tooltipVisible&&fileNameOverflow&&jsxRuntime.jsx(index$2.Tooltip,{targetElement:previewWrapperRef.current,renderContent:()=>`${fileName}`}),jsxRuntime.jsx(FileInfo,{$dimension:dimension,$status:status,children:fileInfo})]})]}),jsxRuntime.jsxs(FunctionalBlock,{children:["Loading"===status&&jsxRuntime.jsx(StyledSpinner,{dimension:"m"===dimension?"ms":"m"}),onCloseIconClick&&jsxRuntime.jsx(CloseButton,{dimension:"xl"===dimension?"lSmall":"mSmall",disabled,onClick:()=>{onCloseIconClick?.(fileId)}})]})]}),errorMessage&&"Error"===status&&jsxRuntime.jsx(ErrorBlock,{$status:status,children:errorMessage})]})});exports.ErrorBlock=ErrorBlock,exports.FileItem=FileItem;
