// @admiral-ds/react-ui/dist/components/input/FileInput/index.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),styled=require("styled-components"),AttachFileOutline=require("../../../node_modules/@admiral-ds/icons/build/system/AttachFileOutline.svg.cjs.js"),borderRadius=require("../../themes/borderRadius.cjs.js"),typography=require("../../Typography/typography.cjs.js");require("../../Typography/defaultTypographyMixin.cjs.js");var refSetter=require("../../common/utils/refSetter.cjs.js"),style=require("./style.cjs.js"),index=require("../../Field/index.cjs.js"),index$1=require("../../Label/index.cjs.js"),utils=require("./utils.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const ExtraText=styled__default.default(index.ExtraTextContainer)`
  padding-top: 20px;
`,Icon=styled__default.default(AttachFileOutline.ReactComponent)`
  height: ${p=>"xl"===p.$dimension?style.FILE_INPUT_ICON_SIZE_XL:style.FILE_INPUT_ICON_SIZE_M};
  width: ${p=>"xl"===p.$dimension?style.FILE_INPUT_ICON_SIZE_XL:style.FILE_INPUT_ICON_SIZE_M};
  margin-right: ${p=>"m"===p.$dimension?style.FILE_INPUT_ICON_MARGIN:""};
  margin-bottom: ${p=>"xl"===p.$dimension?style.FILE_INPUT_ICON_MARGIN:""};
  flex-shrink: 0;

  > * {
    fill: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
  }
`,LabelM=styled__default.default(index$1.Label)`
  display: flex;
`,Description=styled__default.default.div`
  color: ${p=>p.disabled?`var(--admiral-color-Neutral_Neutral30, ${p.theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral90, ${p.theme.color["Neutral/Neutral 90"]})`};
  ${typography.typography["Body/Body 1 Long"]};
`,TitleText=styled__default.default.div`
  color: ${p=>p.disabled?`var(--admiral-color-Neutral_Neutral30, ${p.theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral90, ${p.theme.color["Neutral/Neutral 90"]})`};
  ${p=>"xl"===p.$dimension?typography.typography["Body/Body 2 Long"]:typography.typography["Body/Body 1 Long"]}
  ${p=>"xl"===p.$dimension&&style.titleXL}
`,FocusBorder=styled__default.default.div`
  position: absolute;
  top: -3px;
  bottom: -3px;
  left: -3px;
  right: -3px;
  pointer-events: none;
  overflow: visible;
  border-radius: var(--admiral-border-radius-Medium, ${p=>borderRadius.mediumGroupBorderRadius(p.theme.shape)});
`,StyledInput=styled__default.default.input`
  display: block;
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  opacity: 0;
  cursor: pointer;
  &:disabled {
    cursor: not-allowed;
  }
  &:focus:focus-visible ~ ${FocusBorder} {
    outline: 2px solid var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
  }
`,CustomInput=styled__default.default.input`
  display: block;
  width: 0;
  height: 0;
  opacity: 0;
`,InputWrapper=styled__default.default.div`
  position: relative;
  display: flex;
  align-items: center;
  border: 1px dashed var(--admiral-color-Neutral_Neutral40, ${p=>p.theme.color["Neutral/Neutral 40"]});
  border-radius: var(--admiral-border-radius-Medium, ${p=>borderRadius.mediumGroupBorderRadius(p.theme.shape)});
  pointer-events: all;
  ${p=>p.disabled?style.disabledStyles:style.hoverStyles};
  ${p=>"m"===p.$dimension?style.dimensionMStyles:style.dimensionXLStyles};
  box-sizing: border-box;
  overflow: visible;
`,Wrapper=styled__default.default.div`
  min-width: ${p=>"m"===p.$dimension?style.FILE_INPUT_MIN_WIDTH_M:style.FILE_INPUT_MIN_WIDTH_XL};
  ${p=>p.$width?`width: ${"number"==typeof p.$width?p.$width+"px":p.$width};`:""}
  box-sizing: border-box;
  ${typography.typography["Body/Body 2 Long"]};
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  gap: 16px;
`,FileInputWrapper=styled__default.default.div`
  position: relative;
  width: 100%;
  overflow: visible;
`,FileInput=e.forwardRef(({dimension="xl",width,title,description,renderCustomFileInput,disabled,multiple=!0,children,files,extraText,status,...props},ref)=>{const inputRef=e.useRef(null),titleWithDescription="m"===dimension&&title&&description,titleWithoutDescription=title&&!description||title&&description&&"xl"===dimension;e.useEffect(()=>{const inputNode=inputRef.current;if(inputNode&&files){const dt=new DataTransfer;files.forEach(file=>dt.items.add(file)),inputNode.files=dt.files}},[files]);const[isDragOver,setDragOver]=e.useState(!1),handleDragLeave=()=>{setDragOver(!1)};return e.useEffect(()=>{const inputNode=inputRef.current;inputNode&&inputNode.addEventListener("drop",function(e){const files=Array.from(e.dataTransfer?.files||[]);if(files.length>0){e.preventDefault();const filteredFiles=files.filter(file=>utils.acceptFile(file,props.accept)),dt=new DataTransfer;let event;filteredFiles.forEach(file=>dt.items.add(file)),this.files=dt.files,"function"==typeof Event?event=new Event("input",{bubbles:!0}):(event=document.createEvent("Event"),event.initEvent("input",!0,!0)),this.dispatchEvent(event)}})},[props.accept]),jsxRuntime.jsxs(Wrapper,{$dimension:dimension,$width:width,"data-status":status,children:[jsxRuntime.jsx(FileInputWrapper,{children:renderCustomFileInput?jsxRuntime.jsxs(jsxRuntime.Fragment,{children:[renderCustomFileInput({onQueryUpload:()=>{inputRef.current?.click()}}),jsxRuntime.jsx(CustomInput,{...props,ref:refSetter.refSetter(ref,inputRef),type:"file",multiple,disabled,tabIndex:-1}),jsxRuntime.jsx(FocusBorder,{className:"focus-block"})]}):jsxRuntime.jsxs(jsxRuntime.Fragment,{children:[titleWithDescription&&jsxRuntime.jsx(LabelM,{disabled,children:title}),jsxRuntime.jsxs(InputWrapper,{$dimension:dimension,disabled,"data-isdragover":isDragOver,onDragEnter:disabled?void 0:()=>{setDragOver(!0)},onDragLeave:disabled?void 0:handleDragLeave,onDrop:disabled?void 0:handleDragLeave,children:[jsxRuntime.jsx(Icon,{$dimension:dimension}),titleWithoutDescription&&jsxRuntime.jsx(TitleText,{$dimension:dimension,disabled,children:title}),"m"===dimension&&description&&jsxRuntime.jsx(Description,{disabled,children:description}),jsxRuntime.jsx(StyledInput,{...props,ref:refSetter.refSetter(ref,inputRef),type:"file",disabled,multiple}),jsxRuntime.jsx(FocusBorder,{className:"focus-block"})]})]})}),children,extraText&&"error"===status&&jsxRuntime.jsx(ExtraText,{children:extraText})]})});FileInput.displayName="FileInput",exports.fullWidthPositionMixin=style.fullWidthPositionMixin,exports.halfWidthPositionMixin=style.halfWidthPositionMixin,exports.FileInput=FileInput;
