// @admiral-ds/react-ui/dist/components/input/FileInput/style.cjs.js
"use strict";var styled=require("styled-components");const disabledStyles=styled.css`
  border: 1px dashed var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  cursor: not-allowed;
  & > * {
    pointer-events: none;
  }

  & svg {
    > * {
      fill: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
    }
  }
`,hoverStyles=styled.css`
  &[data-isdragover='true'],
  &:hover {
    cursor: pointer;
    background: var(--admiral-color-Opacity_Hover, ${p=>p.theme.color["Opacity/Hover"]});
  }
  &:active {
    background: var(--admiral-color-Opacity_Press, ${p=>p.theme.color["Opacity/Press"]});
  }
`,dimensionMStyles=styled.css`
  flex-direction: row;
  padding: ${"8px 20px"};
  justify-content: center;
`,dimensionXLStyles=styled.css`
  flex-direction: column;
  padding: ${"24px"};
`,fullWidthPositionMixin=styled.css`
  flex: 1 1 100%;
`,halfWidthPositionMixin=styled.css`
  flex: 1 1 auto;
  max-width: calc(50% - 8px);
`,titleXL=styled.css`
  max-width: 100%;
  text-align: center;
  margin: 0 24px;
`;exports.ERROR_BLOCK_HEIGHT_M="20px",exports.ERROR_BLOCK_HEIGHT_XL="16px",exports.FILE_INPUT_ICON_MARGIN="16px",exports.FILE_INPUT_ICON_SIZE_M="24px",exports.FILE_INPUT_ICON_SIZE_XL="40px",exports.FILE_INPUT_MIN_WIDTH_M="288px",exports.FILE_INPUT_MIN_WIDTH_XL="320px",exports.FILE_ITEM_FUNCTIONAL_ICON_SIZE_XL="24px",exports.FILE_ITEM_PREVIEW_ICON_SIZE_XL="40px",exports.FILE_ITEM_WRAPPER_HEIGHT_M="40px",exports.FILE_ITEM_WRAPPER_HEIGHT_XL="56px",exports.FILE_ITEM_WRAPPER_PADDING_M="10px 16px",exports.FILE_ITEM_WRAPPER_PADDING_XL="8px",exports.dimensionMStyles=dimensionMStyles,exports.dimensionXLStyles=dimensionXLStyles,exports.disabledStyles=disabledStyles,exports.fullWidthPositionMixin=fullWidthPositionMixin,exports.halfWidthPositionMixin=halfWidthPositionMixin,exports.hoverStyles=hoverStyles,exports.titleXL=titleXL;
