// @admiral-ds/react-ui/dist/components/Flex/Flex.styled.cjs.js
"use strict";var styled=require("styled-components"),Flex_utils=require("./Flex.utils.cjs.js"),Flex_types=require("./Flex.types.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const withBoxSize=styled.css`
  box-sizing: border-box;
`,ContainerDiv=styled__default.default.div`
  ${withBoxSize};

  display: flex;
  flex-wrap: wrap;
  width: 100%;
  position: relative;
  row-gap: ${({$rowGap})=>$rowGap?`${$rowGap}px`:"none"};
  column-gap: ${({$columnGap})=>$columnGap?`${$columnGap}px`:"none"};
`,RowDiv=styled__default.default.div`
  ${withBoxSize};

  display: flex;
  width: 100%;
  flex-wrap: ${({$wrap="wrap"})=>$wrap};
  flex-direction: ${({$direction="row"})=>$direction};
  row-gap: ${({$rowGap})=>$rowGap?`${$rowGap}px`:"none"};
  column-gap: ${({$columnGap})=>$columnGap?`${$columnGap}px`:"none"};
  align-items: ${({$alignItems="normal"})=>$alignItems};
`,CellDiv=styled__default.default.div`
  ${withBoxSize};

  width: ${({$col=Flex_types.FLEX_CELL_MAX_SIZE,$columnGap=0})=>Flex_utils.calcCellWidth({column:$col,columnGap:$columnGap})};
  overflow: hidden;
`,GrowCellDiv=styled__default.default.div`
  ${withBoxSize};
  flex-grow: ${({$grow})=>$grow||1};
  overflow: hidden;
`;exports.CellDiv=CellDiv,exports.ContainerDiv=ContainerDiv,exports.GrowCellDiv=GrowCellDiv,exports.RowDiv=RowDiv,exports.withBoxSize=withBoxSize;
