// @admiral-ds/react-ui/dist/components/TabMenuComponent/tabs/BaseTab.cjs.js
"use strict";var styled=require("styled-components"),typography=require("../../Typography/typography.cjs.js");require("../../Typography/defaultTypographyMixin.cjs.js");var constants=require("../constants.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}const BaseTab=_interopDefault(styled).default.button`
  box-sizing: border-box;
  appearance: none;
  border: none;
  position: relative;
  display: inline-flex;
  gap: 8px;
  align-items: center;
  width: ${p=>p.$width};
  height: ${({$dimension})=>"m"===$dimension?constants.BASE_TAB_HEIGHT_M:constants.BASE_TAB_HEIGHT_L}px;
  padding: ${({$dimension})=>"m"===$dimension?constants.HORIZONTAL_TAB_PADDING_M:constants.HORIZONTAL_TAB_PADDING_L};
  background: transparent;
  color: ${p=>p.$selected?`var(--admiral-color-Neutral_Neutral90, ${p.theme.color["Neutral/Neutral 90"]})`:`var(--admiral-color-Neutral_Neutral50, ${p.theme.color["Neutral/Neutral 50"]})`};
  ${({$dimension})=>"m"===$dimension?typography.typography["Body/Body 2 Long"]:typography.typography["Body/Body 1 Long"]}
  user-select: none;
  cursor: pointer;
  -webkit-tap-highlight-color: transparent;

  &:hover:not(:disabled) {
    background: var(--admiral-color-Opacity_Hover, ${p=>p.theme.color["Opacity/Hover"]});
  }
  &:active:not(:disabled) {
    background: var(--admiral-color-Opacity_Press, ${p=>p.theme.color["Opacity/Press"]});
  }
  &:disabled {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
    cursor: not-allowed;
  }

  &:focus-visible {
    border-radius: 0;
    outline: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]}) solid 2px;
    outline-offset: -2px;
  }
`;BaseTab.displayName="BaseTab",exports.BaseTab=BaseTab;
