// @admiral-ds/react-ui/dist/components/Stepper/style.cjs.js
"use strict";var styled=require("styled-components"),typography=require("../Typography/typography.cjs.js");require("../Typography/defaultTypographyMixin.cjs.js");var Step=require("./svg/Step.svg.cjs.js"),borderRadius=require("../themes/borderRadius.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const StepTrack=styled__default.default.div`
  box-sizing: border-box;
  display: flex;
  align-items: center;
  align-self: stretch;
  flex-direction: row;
  width: 100%;
  margin: 0 0 8px 0;
  [data-orientation='vertical'] & {
    flex-direction: column;
    width: 20px;
    margin: 0 8px 0 0;
  }
`,StepIcon=styled__default.default(Step.ReactComponent)`
  margin: 0 4px 0 0;
  [data-orientation='vertical'] & {
    margin: 0 0 4px 0;
  }
  [data-active='true'] & {
    path {
      fill: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
    }
  }
  [data-error='true'] & {
    path {
      fill: var(--admiral-color-Error_Error60Main, ${p=>p.theme.color["Error/Error 60 Main"]});
    }
  }
  [data-warning='true'] & {
    path {
      fill: var(--admiral-color-Warning_Warning50Main, ${p=>p.theme.color["Warning/Warning 50 Main"]});
    }
  }
  [data-disabled='true'] & {
    path {
      fill: var(--admiral-color-Primary_Primary30, ${p=>p.theme.color["Primary/Primary 30"]});
    }
  }
  flex-shrink: 0;
`,StepRail=styled__default.default.div`
  height: 2px;
  width: 100%;
  [data-orientation='vertical'] & {
    height: 100%;
    width: 2px;
  }
  background-color: var(--admiral-color-Neutral_Neutral20, ${p=>p.theme.color["Neutral/Neutral 20"]});
  [data-completed='true'] & {
    background-color: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
  }
  [data-error='true'] & {
    background-color: var(--admiral-color-Error_Error60Main, ${p=>p.theme.color["Error/Error 60 Main"]});
  }
  [data-warning='true'] & {
    background-color: var(--admiral-color-Warning_Warning50Main, ${p=>p.theme.color["Warning/Warning 50 Main"]});
  }
  [data-disabled='true'] & {
    background-color: var(--admiral-color-Primary_Primary30, ${p=>p.theme.color["Primary/Primary 30"]});
  }
  border-radius: 1px;
`,StepContentWrapper=styled__default.default.span`
  position: relative;
  display: flex;
  width: 100%;
  height: 100%;
  flex-shrink: 0;
  flex-direction: column;
  [data-orientation='vertical'] & {
    flex-direction: row;
  }
`,clickableCss=styled.css`
  &:hover svg {
    path {
      fill: ${({$error,$warning,theme})=>$error?`var(--admiral-color-Error_Error70, ${theme.color["Error/Error 70"]})`:$warning?`var(--admiral-color-Warning_Warning70, ${theme.color["Warning/Warning 70"]})`:`var(--admiral-color-Primary_Primary70, ${theme.color["Primary/Primary 70"]})`};
    }
  }
`,clickableNotActiveCss=styled.css`
  &:hover ${StepRail} {
    background-color: ${({$error,$warning,theme})=>$error?`var(--admiral-color-Error_Error70, ${theme.color["Error/Error 70"]})`:$warning?`var(--admiral-color-Warning_Warning70, ${theme.color["Warning/Warning 70"]})`:`var(--admiral-color-Primary_Primary70, ${theme.color["Primary/Primary 70"]})`};
  }
`,StepWrapper=styled__default.default.li`
  display: flex;
  position: relative;
  box-sizing: border-box;
  width: ${({$orientation,$stepWidth,$stepsAmount,$mobile})=>$stepWidth?`${"number"==typeof $stepWidth?`${$stepWidth}px`:$stepWidth}`:$mobile?"horizontal"===$orientation?"188px":"224px":"horizontal"===$orientation?100/$stepsAmount+"%":"100%"};
  ${({$mobile})=>$mobile&&"flex-shrink: 0;"}
  background: none;
  padding: 0;
  margin: 0;
  text-align: left;
  border: none;
  text-decoration: none;
  ${({$clickable})=>$clickable&&"cursor: pointer;"}
  ${({$clickable})=>$clickable&&clickableCss}
  ${({$clickable,$active})=>$clickable&&!$active&&clickableNotActiveCss}

    &:focus {
    outline: none;
    &:before {
      position: absolute;
      content: '';
      border: 2px solid var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
      border-radius: var(--admiral-border-radius-Medium, ${p=>borderRadius.mediumGroupBorderRadius(p.theme.shape)});
      top: 0px;
      left: 0px;
      bottom: 0px;
      right: 0px;
    }
  }
  &:focus,
  & > ${StepContentWrapper} {
    outline: none;
  }
`,ContentWrapper=styled__default.default.div`
  display: flex;
  width: 100%;
  ${({$withTooltip})=>$withTooltip&&"cursor: pointer;"}
  [data-orientation='vertical'] & {
    margin: 0 0 20px 0;
  }
`,Content=styled__default.default.div`
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: ${({$lineClamp})=>1===$lineClamp?1:2===$lineClamp?2:3};
  overflow: hidden;
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});
  [data-disabled='true'] & {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }
  ${typography.typography["Body/Body 2 Long"]}
  margin: 0 12px 0 0;
`,List=styled__default.default.ul`
  position: relative;
  display: flex;
  box-sizing: border-box;
  scroll-behavior: smooth;
  padding: 0;
  ${({$orientation})=>"horizontal"===$orientation?"\nflex-direction: row;\noverflow-x: scroll;\n":"\nflex-direction: column;\n"}

  & > * {
    padding: 2px;
  }

  &::-webkit-scrollbar {
    width: 0 !important;
    height: 0 !important;
  }

  overflow: -moz-scrollbars-none;
  -ms-overflow-style: none;
  scrollbar-width: none;
`;exports.Content=Content,exports.ContentWrapper=ContentWrapper,exports.List=List,exports.StepContentWrapper=StepContentWrapper,exports.StepIcon=StepIcon,exports.StepRail=StepRail,exports.StepTrack=StepTrack,exports.StepWrapper=StepWrapper;
