// @admiral-ds/react-ui/dist/components/PaginationTwo/Items.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),styled=require("styled-components"),typography=require("../Typography/typography.cjs.js");require("../Typography/defaultTypographyMixin.cjs.js");var ChevronLeftOutline=require("../../node_modules/@admiral-ds/icons/build/system/ChevronLeftOutline.svg.cjs.js"),ChevronRightOutline=require("../../node_modules/@admiral-ds/icons/build/system/ChevronRightOutline.svg.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const selectedMixin=styled.css`
  background: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
  color: var(--admiral-color-Special_StaticWhite, ${p=>p.theme.color["Special/Static White"]});
`,hoverMixin=styled.css`
  background: var(--admiral-color-Opacity_Hover, ${p=>p.theme.color["Opacity/Hover"]});
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});
  & *[fill^='#'] {
    fill: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }
`,pressMixin=styled.css`
  background: var(--admiral-color-Opacity_Press, ${p=>p.theme.color["Opacity/Press"]});
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});
  & *[fill^='#'] {
    fill: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }
`,ButtonContent=styled__default.default.span`
  position: relative;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
`,SideButtonsMixin=styled.css`
  ${p=>p.$isPreviousButton&&"margin-right: 8px;"}
  ${p=>p.$isNextButton&&"margin-left: 8px;"}
`,Button=styled__default.default.button`
  box-sizing: border-box;
  appearance: none;
  outline: none;
  margin: 0;
  ${SideButtonsMixin};
  user-select: none;
  display: flex;
  flex: 1 1 auto;
  position: relative;
  min-width: ${p=>"m"===p.$dimension?40:32}px;
  height: ${p=>"m"===p.$dimension?40:32}px;
  padding: 0 6px;
  align-items: center;
  justify-content: center;
  border: none;
  border-radius: 20px;
  ${p=>"m"===p.$dimension?typography.typography["Body/Body 1 Short"]:typography.typography["Body/Body 2 Short"]}
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});
  background: transparent;
  -webkit-tap-highlight-color: transparent;
  cursor: pointer;

  & *[fill^='#'] {
    fill: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }

  &:disabled {
    cursor: not-allowed;
    & *[fill^='#'] {
      fill: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
    }
    ${({selected,theme})=>!selected&&`color: var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]});`}
  }

  ${({selected})=>selected&&selectedMixin}
  &:hover:not(:disabled) {
    ${({selected})=>!selected&&hoverMixin}
  }
  &:active:not(:disabled) {
    ${({selected})=>!selected&&pressMixin}
  }

  &:focus:not(:active) {
    &:before {
      position: absolute;
      content: '';
      border: 2px solid
        ${({theme,$current})=>$current?`var(--admiral-color-Neutral_Neutral00, ${theme.color["Neutral/Neutral 00"]})`:`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`};
      top: ${({$current})=>$current?"2px":"0px"};
      bottom: ${({$current})=>$current?"2px":"0px"};
      left: ${({$current})=>$current?"2px":"0px"};
      right: ${({$current})=>$current?"2px":"0px"};
      border-radius: 20px;
    }
  }
`,EllipsisWrapper=styled__default.default.div`
  display: flex;
  min-width: ${p=>"m"===p.$dimension?40:32}px;
  height: ${p=>"m"===p.$dimension?40:32}px;
  align-items: center;
  justify-content: center;
  ${p=>"m"===p.$dimension?typography.typography["Body/Body 1 Short"]:typography.typography["Body/Body 2 Short"]}
  background: transparent;
`;exports.Ellipsis=({dimension})=>jsxRuntime.jsx(EllipsisWrapper,{$dimension:dimension,children:"..."}),exports.PaginationItem=({onClick,page,selected,disabled,type,dimension="m"})=>{let label="";return"page"===type&&(label=`Страница ${page}, выбрать`),"next"===type&&(label="Следующая страница, выбрать"),"previous"===type&&(label="Предыдущая страница, выбрать"),jsxRuntime.jsx(Button,{type:"button","aria-current":selected,"aria-label":label,tabIndex:disabled?-1:0,$current:selected,$dimension:dimension,$isPreviousButton:"previous"===type,$isNextButton:"next"===type,disabled,selected,onClick,children:jsxRuntime.jsxs(ButtonContent,{tabIndex:-1,children:["page"===type&&page,"next"===type&&jsxRuntime.jsx(ChevronRightOutline.ReactComponent,{width:"s"===dimension?20:24,height:"s"===dimension?20:24}),"previous"===type&&jsxRuntime.jsx(ChevronLeftOutline.ReactComponent,{width:"s"===dimension?20:24,height:"s"===dimension?20:24})]})})};
