// @admiral-ds/react-ui/dist/components/ActionBar/ActionBarItems.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),styled=require("styled-components"),constants=require("./constants.cjs.js"),index=require("../IconButton/index.cjs.js"),index$1=require("../TooltipHOC/index.cjs.js"),MenuItem=require("../Menu/MenuItem.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const Divider=styled__default.default.div`
  box-sizing: border-box;
  width: 1px;
  height: ${({$dimension})=>"s"===$dimension?constants.DIVIDER_SIZE_S:constants.DIVIDER_SIZE_M}px;
  background-color: var(--admiral-color-Neutral_Neutral20, ${p=>p.theme.color["Neutral/Neutral 20"]});
`,Wrapper=styled__default.default.div`
  box-sizing: border-box;
  width: ${({$dimension})=>"s"===$dimension?constants.DIVIDER_SIZE_S:constants.DIVIDER_SIZE_M}px;
  height: ${({$dimension})=>"s"===$dimension?constants.DIVIDER_SIZE_S:constants.DIVIDER_SIZE_M}px;
  display: flex;
  align-items: center;
  justify-content: center;
`,ActionBarItem=e.forwardRef((props,ref)=>jsxRuntime.jsx(index.IconButton,{...props,ref})),ActionBarItemWithTooltip=index$1.TooltipHoc(ActionBarItem),actionBarDropMenuItemIconSize=styled.css`
  width: ${p=>"s"===p.$dimension?constants.ACTION_BAR_DROP_MENU_ITEM_ICON_SIZE_S:constants.ACTION_BAR_DROP_MENU_ITEM_ICON_SIZE_M}px;
  height: ${p=>"s"===p.$dimension?constants.ACTION_BAR_DROP_MENU_ITEM_ICON_SIZE_S:constants.ACTION_BAR_DROP_MENU_ITEM_ICON_SIZE_M}px;
`,ActionBarDropMenuItemWrapper=styled__default.default.div`
  display: flex;
  overflow: hidden;
  text-overflow: ellipsis;
  align-items: center;
  gap: 8px;

  & > svg {
    ${actionBarDropMenuItemIconSize}
  }
`,ActionBarDropMenuItem=e.forwardRef(({dimension="l",children,...props},ref)=>jsxRuntime.jsx(MenuItem.MenuItem,{...props,ref,dimension:"xl"===dimension?"l":dimension,children:jsxRuntime.jsx(ActionBarDropMenuItemWrapper,{$dimension:dimension,children})}));exports.ActionBarDivider=({dimension,...props})=>jsxRuntime.jsx(Wrapper,{...props,$dimension:dimension,children:jsxRuntime.jsx(Divider,{$dimension:dimension})}),exports.ActionBarDropMenuItem=ActionBarDropMenuItem,exports.ActionBarDropMenuItemWrapper=ActionBarDropMenuItemWrapper,exports.ActionBarItem=ActionBarItem,exports.ActionBarItemWithTooltip=ActionBarItemWithTooltip;
