// @admiral-ds/react-ui/dist/components/form/CheckboxField/style.cjs.js
"use strict";var styled=require("styled-components"),typography=require("../../Typography/typography.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}require("../../Typography/defaultTypographyMixin.cjs.js");var styled__default=_interopDefault(styled);const CheckboxComponentWrapper=styled__default.default.label`
  display: inline-flex;
  align-items: flex-start;
  position: relative;
  box-sizing: content-box;
  padding: 0;
  user-select: none;

  cursor: ${p=>p.disabled?"not-allowed":p.readOnly?"default":"pointer"};

  ${p=>"s"===p.$dimension?typography.typography["Body/Body 2 Short"]:typography.typography["Body/Body 1 Short"]}
  color: ${p=>p.disabled?`var(--admiral-color-Neutral_Neutral30, ${p.theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral90, ${p.theme.color["Neutral/Neutral 90"]})`};

  fieldset[data-dimension='s'] && {
    ${typography.typography["Body/Body 2 Short"]};
  }
  fieldset:disabled && {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
    cursor: not-allowed;
  }

  ${p=>p.readOnly?"pointer-events: none":""};
`,CheckboxComponentLabel=styled__default.default.div`
  display: flex;
  flex-direction: column;
  align-content: center;
  margin-top: 2px;
  margin-bottom: 2px;
  margin-left: ${"8px"};
  ${p=>"s"===p.$dimension?typography.typography["Body/Body 2 Short"]:typography.typography["Body/Body 1 Short"]}
  color: ${p=>p.disabled?`var(--admiral-color-Neutral_Neutral30, ${p.theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral90, ${p.theme.color["Neutral/Neutral 90"]})`};
  fieldset:disabled && {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }
  fieldset[data-dimension='s'] && {
    ${typography.typography["Body/Body 2 Short"]}
  }
`,CheckboxComponentHint=styled__default.default.div`
  margin-top: 4px;
  ${p=>"s"===p.$dimension?typography.typography["Caption/Caption 1"]:typography.typography["Body/Body 2 Short"]}
  color: ${p=>p.disabled?`var(--admiral-color-Neutral_Neutral30, ${p.theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral50, ${p.theme.color["Neutral/Neutral 50"]})`};

  fieldset[data-dimension='s'] && {
    ${typography.typography["Caption/Caption 1"]}
  }
`;exports.CheckboxComponentHint=CheckboxComponentHint,exports.CheckboxComponentLabel=CheckboxComponentLabel,exports.CheckboxComponentWrapper=CheckboxComponentWrapper;
