// @admiral-ds/react-ui/dist/components/TextButton/appearanceMixin.cjs.js
"use strict";var styled=require("styled-components"),commonMixin=require("./commonMixin.cjs.js"),borderRadius=require("../themes/borderRadius.cjs.js");const defaultAppearanceMixin=styled.css`
  ${commonMixin.IconContainer} {
    ${({$skeleton})=>$skeleton&&commonMixin.skeletonMixin};
    border-radius: ${({$skeleton})=>$skeleton?"50%":""};
    svg {
      visibility: ${p=>p.$skeleton||p.$loading?"hidden":"visible"};
    }
  }

  border-radius: ${p=>p.$skeleton?0:`var(--admiral-border-radius-Medium, ${borderRadius.mediumGroupBorderRadius(p.theme.shape)})`};

  &:focus-visible {
    outline-offset: 2px;
    outline: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]}) solid 2px;
  }

  &:hover {
    cursor: pointer;
    color: var(--admiral-color-Primary_Primary70, ${p=>p.theme.color["Primary/Primary 70"]});
    ${commonMixin.IconContainer} {
      svg {
        path {
          fill: var(--admiral-color-Primary_Primary70, ${p=>p.theme.color["Primary/Primary 70"]});
        }
      }
    }
  }

  &:active {
    color: var(--admiral-color-Primary_Primary80, ${p=>p.theme.color["Primary/Primary 80"]});
    ${commonMixin.IconContainer} {
      svg {
        path {
          fill: var(--admiral-color-Primary_Primary80, ${p=>p.theme.color["Primary/Primary 80"]});
        }
      }
    }
  }

  &:disabled {
    cursor: ${({$skeleton,$loading})=>$skeleton||$loading?"default":"not-allowed"};
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
    ${commonMixin.IconContainer} {
      svg {
        path {
          fill: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
        }
      }
    }
  }
`,primaryAppearanceMixin=styled.css`
  color: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});

  ${commonMixin.IconContainer} {
    svg {
      path {
        fill: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
      }
    }
  }
`,secondaryAppearanceMixin=styled.css`
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});

  ${commonMixin.IconContainer} {
    svg {
      path {
        fill: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
      }
    }
  }
`,appearanceMixin=styled.css`
  &[data-appearance='primary'] {
    ${primaryAppearanceMixin}
  }
  &[data-appearance='secondary'] {
    ${secondaryAppearanceMixin}
  }

  ${defaultAppearanceMixin}
`;exports.appearanceMixin=appearanceMixin;
