// @admiral-ds/react-ui/dist/components/ContentSwitcher/ContentSwitcherComponent.cjs.js
"use strict";var styled=require("styled-components"),ContentSwitcherItem=require("./ContentSwitcherItem.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const SIZE={l:40,m:32,s:24},BORDER_RADIUS={l:24,m:20,s:16},padding={l:20,m:16,s:12},adaptiveMixin=styled.css`
  width: 100%;
  overflow: hidden;
  min-width: fit-content;
`,ContentSwitcherComponent=styled__default.default.div`
  background: var(--admiral-color-Opacity_Neutral8, ${p=>p.theme.color["Opacity/Neutral 8"]});
  border: 2px solid transparent;
  border-radius: ${({$dimension})=>BORDER_RADIUS[$dimension]}px;
  display: inline-flex;
  height: ${({$dimension})=>SIZE[$dimension]}px;
  padding: 2px;

  ${({$adaptive})=>"fill"===$adaptive?adaptiveMixin:""}

  &[data-appearance~='primary'] {
    border: 2px solid var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
    background-color: transparent;
  }

  ${ContentSwitcherItem.ContentSwitcherItemButton} {
    height: 100%;
    padding: ${({$dimension})=>`0 ${padding[$dimension]}px`};
    font-size: ${({$dimension})=>"l"===$dimension?16:14}px;
    line-height: ${({$dimension})=>"l"===$dimension?24:20}px;
  }

  > *:not(:first-child) {
    margin-left: 8px;
  }
`;exports.ContentSwitcherComponent=ContentSwitcherComponent,exports.SIZE=SIZE;
