// @admiral-ds/react-ui/dist/components/ContentSwitcher/ContentSwitcherItem.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),styled=require("styled-components"),typography=require("../Typography/typography.cjs.js");require("../Typography/defaultTypographyMixin.cjs.js");var index=require("../Badge/index.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const colorMixin=styled.css`
  color: ${({$active,disabled,theme})=>$active?`var(--admiral-color-Neutral_Neutral90, ${theme.color["Neutral/Neutral 90"]})`:disabled?`var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral50, ${theme.color["Neutral/Neutral 50"]})`};
  [data-appearance~='primary'] && {
    color: ${({$active,disabled,theme})=>$active?`var(--admiral-color-Special_StaticWhite, ${theme.color["Special/Static White"]})`:disabled?`var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`};
  }
`,adaptiveMixin=styled.css`
  width: 100%;
  display: flex;
  justify-content: center;
`,ContentSwitcherItemButton=styled__default.default.button`
  border: none;
  border-radius: ${22}px;
  cursor: ${({disabled})=>disabled?"not-allowed":"pointer"};
  display: flex;
  align-items: center;
  position: relative;
  outline: 0;
  background-color: ${({theme,disabled,$active})=>$active&&!disabled?`var(--admiral-color-Neutral_Neutral00, ${theme.color["Neutral/Neutral 00"]})`:"transparent"};

  ${typography.typography["Button/Button 1"]}
  white-space: nowrap;

  ${colorMixin};
  ${({$adaptive})=>"fill"===$adaptive?adaptiveMixin:""}

  &:hover:not(:disabled) {
    ${({$active,disabled,theme})=>$active||disabled?"":`background: var(--admiral-color-Opacity_Hover, ${theme.color["Opacity/Hover"]});`};
  }
  &:active:not(:disabled) {
    ${({$active,disabled,theme})=>$active||disabled?"":`background: var(--admiral-color-Opacity_Press, ${theme.color["Opacity/Press"]});`};
  }

  & *[fill^='#'] {
    fill: ${({disabled,theme})=>disabled?`var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral50, ${theme.color["Neutral/Neutral 50"]})`};
  }

  [data-appearance~='primary'] && *[fill^='#'] {
    fill: ${({$active,disabled,theme})=>$active?`var(--admiral-color-Special_StaticWhite, ${theme.color["Special/Static White"]})`:disabled?`var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`};
  }

  & ${index.BadgeComponent} {
    background-color: ${p=>p.$active?`var(--admiral-color-Neutral_Neutral50, ${p.theme.color["Neutral/Neutral 50"]})`:`var(--admiral-color-Neutral_Neutral00, ${p.theme.color["Neutral/Neutral 00"]})`};
    color: ${({theme,disabled,$active})=>$active?`var(--admiral-color-Special_StaticWhite, ${theme.color["Special/Static White"]})`:disabled?`var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral50, ${theme.color["Neutral/Neutral 50"]})`};
  }

  [data-appearance~='primary'] && ${index.BadgeComponent} {
    background-color: ${({$active,disabled,theme})=>$active?`var(--admiral-color-Special_StaticWhite, ${theme.color["Special/Static White"]})`:disabled?`var(--admiral-color-Neutral_Neutral10, ${theme.color["Neutral/Neutral 10"]})`:`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`};
    color: ${({theme,disabled,$active})=>$active?`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`:disabled?`var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Special_StaticWhite, ${theme.color["Special/Static White"]})`};
  }

  [data-appearance~='primary'] && {
    background-color: ${({theme,disabled,$active})=>$active&&!disabled?`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`:"transparent"};
    &:hover:not(:disabled) {
      ${({$active,disabled,theme})=>$active||disabled?"":`background: var(--admiral-color-Opacity_Hover, ${theme.color["Opacity/Hover"]});`};
    }
    &:active:not(:disabled) {
      ${({$active,disabled,theme})=>$active||disabled?"":`background: var(--admiral-color-Opacity_Press, ${theme.color["Opacity/Press"]});`};
    }
  }

  ${p=>p.$cssMixin&&p.$cssMixin}
`,ContentSwitcherItem=e.forwardRef(({active,cssMixin,type="button",adaptive,...props},ref)=>jsxRuntime.jsx(ContentSwitcherItemButton,{ref,type,...props,$adaptive:adaptive,$active:active,$cssMixin:cssMixin}));exports.ContentSwitcherItem=ContentSwitcherItem,exports.ContentSwitcherItemButton=ContentSwitcherItemButton;
