// @admiral-ds/react-ui/dist/components/input/EditMode/style.cjs.js
"use strict";var styled=require("styled-components"),index$2=require("../../Button/index.cjs.js"),index=require("../TextInput/index.cjs.js"),index$1=require("../TextArea/index.cjs.js"),typography=require("../../Typography/typography.cjs.js");require("../../Typography/defaultTypographyMixin.cjs.js");var EditSolid=require("../../../node_modules/@admiral-ds/icons/build/system/EditSolid.svg.cjs.js"),SentOutline=require("../../../node_modules/@admiral-ds/icons/build/service/SentOutline.svg.cjs.js"),CloseOutline=require("../../../node_modules/@admiral-ds/icons/build/service/CloseOutline.svg.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const TypographyMixin=styled.css`
  [data-dimension='s'] & {
    ${typography.typography["Body/Body 2 Long"]}
  }

  [data-dimension='s-bold'] & {
    ${typography.typography["Subtitle/Subtitle 3"]}
  }

  [data-dimension='m'] & {
    ${typography.typography["Body/Body 1 Long"]}
  }

  [data-dimension='m-bold'] & {
    ${typography.typography["Subtitle/Subtitle 2"]}
  }

  [data-dimension='xl'] & {
    ${typography.typography["Header/H5"]}
  }

  [data-dimension='xxl'] & {
    ${typography.typography["Header/H4"]}
  }
`,EditInput=styled__default.default(index.TextInput)`
  flex: 1 1 auto;
  & input {
    ${TypographyMixin}
  }
`,EditArea=styled__default.default(index$1.TextArea)`
  flex: 1 1 auto;
  & textarea,
  & :first-child {
    ${TypographyMixin}
  }
`,EditButton=styled__default.default(index$2.Button)`
  margin-left: 4px;
  flex-shrink: 0;
  ${p=>p.$multiline&&"align-self: flex-start;"}
`,multilineIcon=styled.css`
  align-self: flex-start;

  [data-dimension='s-bold'] &&&,
  [data-dimension='s'] &&& {
    margin-top: 6px;
  }

  [data-dimension='m'] &&&,
  [data-dimension='m-bold'] &&& {
    margin-top: 8px;
  }

  [data-dimension='xl'] &&& {
    margin-top: 14px;
  }

  [data-dimension='xxl'] &&& {
    margin-top: 16px;
  }
`,iconStyle=styled.css`
  display: flex;
  flex-shrink: 0;
  user-select: none;
`,EditIcon=styled__default.default(EditSolid.ReactComponent)`
  ${iconStyle}
  & *[fill^='#'] {
    fill: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }
  cursor: pointer;

  [data-disabled='true'] & {
    cursor: not-allowed;
    & *[fill^='#'] {
      fill: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
    }
  }
  [data-disabled='false'] & {
    &:hover {
      & *[fill^='#'] {
        fill: var(--admiral-color-Primary_Primary70, ${p=>p.theme.color["Primary/Primary 70"]});
      }
    }
  }

  ${p=>p.$multiline&&multilineIcon}
`,ConfirmIcon=styled__default.default(SentOutline.ReactComponent)`
  ${iconStyle}
`,CancelIcon=styled__default.default(CloseOutline.ReactComponent)`
  ${iconStyle}
`,Wrapper=styled__default.default.div`
  display: flex;
  align-items: center;
  ${({$cssMixin})=>$cssMixin||""};
`,MultilineMixin=styled.css`
  overflow-wrap: anywhere;
  height: auto;
  white-space: normal;

  [data-dimension='s-bold'] &&&,
  [data-dimension='s'] &&& {
    padding-top: 6px;
    padding-bottom: 6px;
  }

  [data-dimension='m'] &&&,
  [data-dimension='m-bold'] &&& {
    padding-top: 8px;
    padding-bottom: 8px;
  }

  [data-dimension='xl'] &&& {
    padding-top: 14px;
    padding-bottom: 14px;
  }

  [data-dimension='xxl'] &&& {
    padding-top: 12px;
    padding-bottom: 12px;
  }
`,SinglelineMixin=styled.css`
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;

  [data-dimension='s'] &&& {
    height: 32px;
    line-height: 32px;
  }

  [data-dimension='s-bold'] &&& {
    height: 32px;
    line-height: 32px;
  }

  [data-dimension='m'] &&& {
    height: 40px;
    line-height: 40px;
  }

  [data-dimension='m-bold'] &&& {
    height: 40px;
    line-height: 40px;
  }

  [data-dimension='xl'] &&& {
    height: 56px;
    line-height: 56px;
  }

  [data-dimension='xxl'] &&& {
    height: 56px;
    line-height: 56px;
  }
`,Text=styled__default.default.div`
  display: block;

  align-items: center;
  cursor: text;
  margin-right: 12px;
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});

  ${TypographyMixin};
  ${p=>p.$multiline?MultilineMixin:SinglelineMixin}

  [data-disabled='true'] & {
    cursor: default;
    pointer-events: none;
  }

  [data-disabled='false'] & {
    &:hover {
      & + ${EditIcon} *[fill^='#'] {
        fill: var(--admiral-color-Primary_Primary70, ${p=>p.theme.color["Primary/Primary 70"]});
      }
    }
  }

  [data-disable-copying] & {
    user-select: none;
    pointer-events: none;
  }
`;exports.CancelIcon=CancelIcon,exports.ConfirmIcon=ConfirmIcon,exports.EditArea=EditArea,exports.EditButton=EditButton,exports.EditIcon=EditIcon,exports.EditInput=EditInput,exports.Text=Text,exports.Wrapper=Wrapper,exports.multilineIcon=multilineIcon;
