// @admiral-ds/react-ui/dist/components/CarouselSlider/index.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),styled=require("styled-components");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const primaryAppearanceMixin=styled.css`
  background-color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  border: none;
`,primaryAppearanceSelectedMixin=styled.css`
  background-color: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
  border: none;
`,defaultAppearanceMixin=styled.css`
  background-color: var(--admiral-color-Special_StaticWhite, ${p=>p.theme.color["Special/Static White"]});
  border: 0.5px solid
    var(--admiral-color-Special_DarkStaticNeutral00, ${p=>p.theme.color["Special/Dark Static Neutral 80"]});
`,ItemContent=styled__default.default.div`
  box-sizing: border-box;
  width: ${p=>p.$isCurrent?28:12}px;
  height: 4px;
  border-radius: 2px;
  ${p=>"default"===p.$appearance?defaultAppearanceMixin:p.$isCurrent?primaryAppearanceSelectedMixin:primaryAppearanceMixin};
`,Item=styled__default.default.button`
  all: unset;
  display: block;
  appearance: none;
  cursor: pointer;
  -webkit-tap-highlight-color: transparent;

  box-sizing: border-box;
  width: ${p=>p.$isCurrent?34:18}px;
  height: 16px;
  padding: 7px 3px 6px 3px;

  &:focus-visible > ${ItemContent} {
    outline-offset: 2px;
    outline: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]}) solid 2px;
  }
`,CarouselSliderItem=({appearance="default",isCurrent=!1,...props})=>jsxRuntime.jsx(Item,{...props,$isCurrent:isCurrent,children:jsxRuntime.jsx(ItemContent,{$appearance:appearance,$isCurrent:isCurrent})});CarouselSliderItem.displayName="CarouselSliderItem";const Wrapper=styled__default.default.div`
  position: relative;
  display: flex;
  overflow-x: scroll;
  width: fit-content;

  &::-webkit-scrollbar {
    width: 0 !important;
    height: 0 !important;
  }

  overflow: -moz-scrollbars-none;
  -ms-overflow-style: none;
  scrollbar-width: none;

  & ${Item}, ${ItemContent} {
    transition: width 0.3s ease-in-out;
  }
`,CarouselSlider=({children,...props})=>jsxRuntime.jsx(Wrapper,{...props,children});CarouselSlider.displayName="CarouselSlider",exports.CarouselSlider=CarouselSlider,exports.CarouselSliderItem=CarouselSliderItem;
