// @admiral-ds/react-ui/dist/components/Table/filter/Button.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),styled=require("styled-components"),MoreHorizontalOutline=require("../../../node_modules/@admiral-ds/icons/build/system/MoreHorizontalOutline.svg.cjs.js"),refSetter=require("../../common/utils/refSetter.cjs.js"),e=require("react");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const activeFilter=styled.css`
  & *[fill^='#'] {
    fill: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
  }
`,ButtonComponent=styled__default.default.button`
  display: flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
  border: 0;
  outline: none;
  background: transparent;
  -webkit-tap-highlight-color: transparent;
  padding: 0;
  margin: 2px 0;
`,ButtonContent=styled__default.default.span`
  position: relative;
  display: inline-flex;
  align-items: center;
  width: 100%;
  height: 100%;

  & svg {
    flex-shrink: 0;
    width: ${({$iconSize})=>$iconSize};
    height: ${({$iconSize})=>$iconSize};
  }

  & *[fill^='#'] {
    fill: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }

  &:hover {
    *[fill^='#'] {
      fill: var(--admiral-color-Primary_Primary70, ${p=>p.theme.color["Primary/Primary 70"]});
    }
  }

  ${({$isFilterActive})=>$isFilterActive&&activeFilter}
`,Button=e.forwardRef(({dimension,renderFilterIcon,isFilterActive,...props},ref)=>{const iconSize="s"===dimension||"m"===dimension?"16px":"20px",btnRef=e.useRef(null);return e.useEffect(()=>{const button=btnRef.current;function handleMove(e){e.stopPropagation()}if(button)return button.addEventListener("mousemove",handleMove),button.addEventListener("touchmove",handleMove),()=>{button.removeEventListener("mousemove",handleMove),button.removeEventListener("touchmove",handleMove)}},[]),jsxRuntime.jsx(ButtonComponent,{ref:refSetter.refSetter(ref,btnRef),type:"button",...props,children:jsxRuntime.jsx(ButtonContent,{$isFilterActive:isFilterActive,$iconSize:iconSize,children:renderFilterIcon?renderFilterIcon():jsxRuntime.jsx(MoreHorizontalOutline.ReactComponent,{"aria-hidden":!0})})})});exports.Button=Button,exports.ButtonContent=ButtonContent;
