// @admiral-ds/react-ui/dist/components/RadioButton/style.cjs.js
"use strict";var styled=require("styled-components"),typography=require("../Typography/typography.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}require("../Typography/defaultTypographyMixin.cjs.js");var styled__default=_interopDefault(styled);const RadioButtonHover=styled__default.default.div`
  visibility: hidden;
  pointer-events: none;
  position: absolute;
  background: var(--admiral-color-Opacity_Hover, ${p=>p.theme.color["Opacity/Hover"]});
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  border-radius: 50%;
  ${p=>`\n        width: calc(100% + ${"s"===p.$dimension?14:16}px);\n        height: calc(100% + ${"s"===p.$dimension?14:16}px);\n      `}
  background-color: var(--admiral-color-Opacity_Hover, ${p=>p.theme.color["Opacity/Hover"]});
`,RadioButtonLabel=styled__default.default.div`
  display: flex;
  flex-direction: column;
  align-content: center;
  margin-left: ${p=>"s"===p.$dimension?"9px":"10px"};
  ${p=>"s"===p.$dimension?"margin-top: 1px;":""}
  ${p=>"s"===p.$dimension?typography.typography["Body/Body 2 Short"]:typography.typography["Body/Body 1 Short"]}
  color: ${p=>p.disabled?`var(--admiral-color-Neutral_Neutral30, ${p.theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral90, ${p.theme.color["Neutral/Neutral 90"]})`};
  fieldset:disabled && {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }
  fieldset[data-dimension='s'] && {
    margin-left: ${"9px"};
    margin-top: 1px;
    ${typography.typography["Body/Body 2 Short"]}
  }
`,disabledCss=styled.css`
  border-color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  background-color: var(--admiral-color-Neutral_Neutral10, ${p=>p.theme.color["Neutral/Neutral 10"]});
`,Span=styled__default.default.span`
  display: inline-block;
  position: absolute;
  flex-shrink: 0;
  margin: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  content: '';
  box-sizing: border-box;
  pointer-events: none;
  border-radius: 50%;
  transition: all 0.25s ease-in-out;

  width: ${p=>"s"===p.$dimension?18:20}px;
  height: ${p=>"s"===p.$dimension?18:20}px;
  background-color: var(--admiral-color-Neutral_Neutral00, ${p=>p.theme.color["Neutral/Neutral 00"]});
  border: ${1}px solid
    var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});

  fieldset[data-dimension='s'] & {
    width: ${18}px;
    height: ${18}px;
  }
  ${p=>p.disabled?disabledCss:""}
  ${p=>p.$error?`border-color: var(--admiral-color-Error_Error60Main, ${p.theme.color["Error/Error 60 Main"]});`:""}

  fieldset:disabled & {
    border: ${1}px solid
      var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }
`,InputContainer=styled__default.default.div`
  pointer-events: none;
  position: relative;
  flex-shrink: 0;
  border-radius: 50%;
  ${p=>`\n    width: ${"s"===p.$dimension?18:20}px;\n    height: ${"s"===p.$dimension?18:20}px;\n  `}
  fieldset[data-dimension='s'] & {
    width: ${18}px;
    height: ${18}px;
  }
`,readOnlyCss=styled.css`
  pointer-events: none;

  &:not(:checked) {
    & + div > span {
      background-color: var(--admiral-color-Neutral_Neutral10, ${p=>p.theme.color["Neutral/Neutral 10"]});
    }
  }

  &:checked {
    & + div > span {
      background-color: var(--admiral-color-Neutral_Neutral00, ${p=>p.theme.color["Neutral/Neutral 00"]});
    }
  }
`,Input=styled__default.default.input`
  position: absolute;
  cursor: pointer;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  padding: 0;
  margin: 0;
  opacity: 0;

  box-sizing: border-box;

  ${p=>p.readOnly&&readOnlyCss};

  &:disabled {
    cursor: not-allowed;
  }

  &:not(:checked):disabled {
    & + div > span {
      background-color: var(--admiral-color-Neutral_Neutral10, ${p=>p.theme.color["Neutral/Neutral 10"]});
    }
  }

  &:checked:disabled {
    & + div > span {
      border: ${5}px solid
        var(--admiral-color-Primary_Primary30, ${p=>p.theme.color["Primary/Primary 30"]});
      background-color: var(--admiral-color-Neutral_Neutral00, ${p=>p.theme.color["Neutral/Neutral 00"]});
    }
  }

  &:checked:not(:disabled) {
    & + div > span {
      border: ${p=>p.readOnly?`5px solid var(--admiral-color_Primary_Primary30, ${p.theme.color["Primary/Primary 30"]})`:`5px solid var(--admiral-color_Primary_Primary60Main, ${p.theme.color["Primary/Primary 60 Main"]})`};
    }
  }

  &:not(:disabled) {
    &:focus-visible + div {
      outline-offset: ${2}px;
      outline: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]}) solid
        ${2}px;
    }

    &:hover {
      & + div > div {
        ${p=>!p.readOnly&&"visibility: visible"};
      }
      &:focus-visible + div {
        outline: none;
      }
    }
    &:active {
      & + div > div {
        ${p=>!p.readOnly&&"visibility: visible"};
        background: var(--admiral-color-Opacity_Press, ${p=>p.theme.color["Opacity/Press"]});
      }
      &:focus-visible + div {
        outline: none;
      }
    }
  }
`,Hint=styled__default.default.div`
  margin-top: 6px;
  ${p=>"s"===p.$dimension?typography.typography["Caption/Caption 1"]:typography.typography["Body/Body 2 Short"]}
  color: ${p=>p.disabled?`var(--admiral-color-Neutral_Neutral30, ${p.theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral50, ${p.theme.color["Neutral/Neutral 50"]})`};

  fieldset[data-dimension='s'] && {
    ${typography.typography["Caption/Caption 1"]}
  }
`,RadioButtonComponent=styled__default.default.label`
  display: flex;
  align-items: flex-start;
  position: relative;
  box-sizing: content-box;
  padding: ${p=>"s"===p.$dimension?"1px 0 1px 1px":"2px 0 2px 2px"};
  user-select: none;

  cursor: ${p=>p.disabled?"not-allowed":p.readOnly?"default":"pointer"};

  ${p=>"s"===p.$dimension?typography.typography["Body/Body 2 Short"]:typography.typography["Body/Body 1 Short"]}
  color: ${p=>p.disabled?`var(--admiral-color-Neutral_Neutral30, ${p.theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral90, ${p.theme.color["Neutral/Neutral 90"]})`};

  fieldset[data-dimension='s'] && {
    padding: 1px 0 1px 1px;
    ${typography.typography["Body/Body 2 Short"]};
  }
  fieldset:disabled && {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
    cursor: not-allowed;
  }

  ${p=>p.readOnly&&styled.css`
      & > * {
        pointer-events: none;
      }
    `};
`;exports.Hint=Hint,exports.Input=Input,exports.InputContainer=InputContainer,exports.RadioButtonComponent=RadioButtonComponent,exports.RadioButtonHover=RadioButtonHover,exports.RadioButtonLabel=RadioButtonLabel,exports.Span=Span;
