// @admiral-ds/react-ui/dist/components/Drawer/components.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),styled=require("styled-components"),typography=require("../Typography/typography.cjs.js");require("../Typography/defaultTypographyMixin.cjs.js");var debounce=require("../common/utils/debounce.cjs.js"),resizePaddings=require("../common/dom/resizePaddings.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const Title=styled__default.default.h5`
  ${({$mobile})=>$mobile?typography.typography["Header/H6"]:typography.typography["Header/H5"]};
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});
  margin: 0;
  padding: ${({$mobile,$displayCloseIcon})=>$mobile?$displayCloseIcon?"0 48px 8px 16px":"0 16px 8px":$displayCloseIcon?"0 56px 8px 24px":"0 24px 8px"};
`,Content=styled__default.default.div`
  overflow-y: auto;
  outline: none;
  scrollbar-gutter: stable;
  padding-block: 8px;
  padding-inline: ${({$mobile})=>($mobile?16:24)+"px"};
  height: 100%;
`,ButtonPanel=styled__default.default.div`
  display: flex;
  flex-direction: ${({$mobile})=>$mobile?"column-reverse":"row-reverse"};
  padding: ${({$mobile})=>$mobile?"16px 16px 0":"16px 24px 0"};

  & > button {
    margin: ${({$mobile})=>$mobile?"0 0 16px 0":"0 16px 0 0"};
    ${({$mobile})=>$mobile&&"width: 100%;"}
  }

  & > button:first-child {
    margin: 0;
  }
`,DrawerContext=e.createContext({mobile:!1,displayCloseIcon:!0});exports.DrawerButtonPanel=({children,...props})=>{const mobile=e.useContext(DrawerContext).mobile;return jsxRuntime.jsx(ButtonPanel,{$mobile:mobile,...props,children})},exports.DrawerContent=({children,...props})=>{const contentRef=e.useRef(null),mobile=e.useContext(DrawerContext).mobile;return e.useLayoutEffect(()=>{const node=contentRef.current;if(node){resizePaddings.resizePaddings(node);const resizeObserver=new ResizeObserver(debounce.debounce(()=>resizePaddings.resizePaddings(node),250));return resizeObserver.observe(node),()=>{resizeObserver.unobserve(node)}}},[]),jsxRuntime.jsx(Content,{tabIndex:-1,ref:contentRef,$mobile:mobile,...props,children})},exports.DrawerContext=DrawerContext,exports.DrawerTitle=({children,...props})=>{const{mobile,displayCloseIcon}=e.useContext(DrawerContext),asProp=mobile?"h6":"h5";return jsxRuntime.jsx(Title,{$mobile:mobile,$displayCloseIcon:displayCloseIcon,as:asProp,...props,children})};
