// @admiral-ds/react-ui/dist/components/input/DateTime/index.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),styled=require("styled-components"),index$1=require("../DateInput/index.cjs.js"),index$2=require("../TimeInput/index.cjs.js"),index=require("../TextInput/index.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const getFocusBorderColor=styled.css`
  ${({status,theme})=>{if(!status)return`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`;switch(status){case"error":return`var(--admiral-color-Error_Error60Main, ${theme.color["Error/Error 60 Main"]})`;case"success":return`var(--admiral-color-Success_Success50Main, ${theme.color["Success/Success 50 Main"]})`}}}
`,DateTimeDateInput=styled__default.default(index$1.DateInput)`
  flex: 1 1 100%;
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;

  & ${index.InputBorderedDiv}${index.InputBorderedDiv}${index.InputBorderedDiv} {
    border-right: none;
  }

  &:focus-within ${index.InputBorderedDiv}${index.InputBorderedDiv}${index.InputBorderedDiv} {
    ${p=>!p.disabled&&!p.readOnly&&styled.css`
        border-right: 1px solid ${getFocusBorderColor};
      `};
  }
`;DateTimeDateInput.displayName="DateTimeDateInput";const DateTimeTimeInput=styled__default.default(index$2.TimeInput)`
  flex: 0 0 0;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;

  & ${index.InputBorderedDiv}${index.InputBorderedDiv}${index.InputBorderedDiv} {
    border-left: none;
  }

  &:focus-within ${index.InputBorderedDiv}${index.InputBorderedDiv}${index.InputBorderedDiv} {
    ${p=>!p.disabled&&!p.readOnly&&styled.css`
        border-left: 1px solid ${getFocusBorderColor};
      `};
  }
`;DateTimeTimeInput.displayName="DateTimeTimeInput";const getSeparatorColor=styled.css`
  ${({disabled,readOnly,$status,theme})=>{if(disabled)return`var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]})`;if(readOnly||!$status)return`var(--admiral-color-Neutral_Neutral40, ${theme.color["Neutral/Neutral 40"]})`;switch($status){case"error":return`var(--admiral-color-Error_Error60Main, ${theme.color["Error/Error 60 Main"]})`;case"success":return`var(--admiral-color-Success_Success50Main, ${theme.color["Success/Success 50 Main"]})`}return`var(--admiral-color-Neutral_Neutral40, ${theme.color["Neutral/Neutral 40"]})`}}
`,getHoverSeparatorColor=styled.css`
  border-left-color: ${({$status,theme})=>{if(!$status)return`var(--admiral-color-Neutral_Neutral60, ${theme.color["Neutral/Neutral 60"]})`;switch($status){case"error":return`var(--admiral-color-Error_Error70, ${theme.color["Error/Error 70"]})`;case"success":return`var(--admiral-color-Success_Success60, ${theme.color["Success/Success 60"]})`}}};
`,getFocusSeparatorColor=styled.css`
  border-left-color: ${({$status,theme})=>{if(!$status)return`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`;switch($status){case"error":return`var(--admiral-color-Error_Error60Main, ${theme.color["Error/Error 60 Main"]})`;case"success":return`var(--admiral-color-Success_Success50Main, ${theme.color["Success/Success 50 Main"]})`}}};
`,DateTimeSeparatorDiv=styled__default.default.div`
  border-left: 1px solid ${getSeparatorColor};
`,DateTimeSeparator=e.forwardRef(({status,...props},ref)=>jsxRuntime.jsx(DateTimeSeparatorDiv,{ref,$status:status,...props}));DateTimeSeparator.displayName="DateTimeSeparator";const DateTimeContainerDiv=styled__default.default.div`
  display: flex;
  min-width: 288px;

  &:hover ${DateTimeSeparatorDiv} {
    ${p=>!p.disabled&&!p.readOnly&&getHoverSeparatorColor};
  }
  &:focus-within ${DateTimeSeparatorDiv} {
    ${p=>!p.disabled&&!p.readOnly&&getFocusSeparatorColor};
  }
`,DateTimeContainer=e.forwardRef(({status,...props},ref)=>jsxRuntime.jsx(DateTimeContainerDiv,{ref,$status:status,...props}));DateTimeContainer.displayName="DateTimeContainer",exports.DateTimeContainer=DateTimeContainer,exports.DateTimeDateInput=DateTimeDateInput,exports.DateTimeSeparator=DateTimeSeparator,exports.DateTimeTimeInput=DateTimeTimeInput;
