// @admiral-ds/react-ui/dist/components/Accordion/style.cjs.js
"use strict";var styled=require("styled-components"),ChevronDownOutline=require("../../node_modules/@admiral-ds/icons/build/system/ChevronDownOutline.svg.cjs.js"),typography=require("../Typography/typography.cjs.js");require("../Typography/defaultTypographyMixin.cjs.js");var index=require("../themes/common/index.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const Chevron=styled__default.default(ChevronDownOutline.ReactComponent)`
  transition: all 0.3s;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  margin: 0 0 0 8px;
  [data-icon='left'] & {
    margin: 0 8px 0 0;
  }
  & *[fill^='#'] {
    fill: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }
  [data-disabled='true'] & {
    & *[fill^='#'] {
      fill: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
    }
  }
  [data-dimension='m'] & {
    width: 20px;
    height: 20px;
  }
`,ItemTitleContent=styled__default.default.span`
  position: relative;
  width: 100%;
  height: 100%;
  display: flex;
  align-items: flex-start;

  padding: 16px 16px 15px 16px;
  [data-dimension='m'] && {
    padding: 10px 16px 9px 16px;
  }

  [data-icon='left'] & {
    flex-direction: row-reverse;
  }
`,TitleContent=styled__default.default.span`
  display: inline-flex;
  flex: 1 1 auto;
`,eventsMixin=styled.css`
  &:hover {
    cursor: pointer;
    background: var(--admiral-color-Opacity_Hover, ${p=>p.theme.color["Opacity/Hover"]});
  }
  &:active {
    background: var(--admiral-color-Opacity_Press, ${p=>p.theme.color["Opacity/Press"]});
  }
  &:focus-visible {
    outline: none;
    &:before {
      position: absolute;
      content: '';
      border: 2px solid var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
      top: -1px;
      left: 0;
      bottom: -1px;
      right: 0;
    }
  }
`,ItemTitle=styled__default.default.div`
  position: relative;
  display: flex;
  justify-content: space-between;
  align-items: center;
  min-height: 54px;
  width: 100%;
  background: transparent;
  -webkit-tap-highlight-color: transparent;
  text-align: left;
  border: none;
  background: none;
  margin: 0;
  padding: 0;
  overflow: visible;
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color[index.DefaultFontColorName]});
  ${typography.typography["Subtitle/Subtitle 2"]}

  [data-disabled='false'] & {
    ${eventsMixin}
  }

  [data-disabled='true'] & {
    cursor: not-allowed;
    user-select: none;
  }

  [data-dimension='m'] & {
    min-height: 38px;
    ${typography.typography["Subtitle/Subtitle 3"]}
  }
`,ItemWrapper=styled__default.default.div`
  border-bottom: 1px solid var(--admiral-color-Neutral_Neutral20, ${p=>p.theme.color["Neutral/Neutral 20"]});
  & > ${ItemTitle} ${Chevron} {
    transform: ${p=>p.$opened?"rotate(180deg)":"rotate(0deg)"};
  }
`,ItemContent=styled__default.default.div`
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color[index.DefaultFontColorName]});
  overflow-y: auto;
  max-height: ${p=>p.$contentMaxHeight};
  padding: 4px 16px 16px 16px;
  ${typography.typography["Body/Body 1 Long"]}
  [data-dimension='m'] & {
    padding: 4px 16px 10px 16px;
    ${typography.typography["Body/Body 2 Long"]}
  }
`,AccordionWrapper=styled__default.default.div`
  position: relative;

  & > ${ItemWrapper}:first-child {
    & ${ItemTitleContent} {
      padding: ${({$dimension})=>"l"===$dimension?"15px 16px":"9px 16px"};
    }
    border-top: 1px solid
      ${({theme,$hideTopDivider})=>$hideTopDivider?"transparent":`var(--admiral-color-Neutral_Neutral20, ${theme.color["Neutral/Neutral 20"]})`};
  }
  & > ${ItemWrapper}:last-child {
    border-bottom: 1px solid
      ${({theme,$hideBottomDivider})=>$hideBottomDivider?"transparent":`var(--admiral-color-Neutral_Neutral20, ${theme.color["Neutral/Neutral 20"]})`};
  }
`;exports.AccordionWrapper=AccordionWrapper,exports.Chevron=Chevron,exports.ItemContent=ItemContent,exports.ItemTitle=ItemTitle,exports.ItemTitleContent=ItemTitleContent,exports.ItemWrapper=ItemWrapper,exports.TitleContent=TitleContent;
