// @admiral-ds/react-ui/dist/components/DropDownItem/index.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),styled=require("styled-components"),mixins=require("./mixins.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}function _interopNamespace(e){if(e&&e.__esModule)return e;var n=Object.create(null);return e&&Object.keys(e).forEach(function(k){if("default"!==k){var d=Object.getOwnPropertyDescriptor(e,k);Object.defineProperty(n,k,d.get?d:{enumerable:!0,get:function(){return e[k]}})}}),n.default=e,Object.freeze(n)}var e__namespace=_interopNamespace(e),styled__default=_interopDefault(styled);const hoverStyle=styled.css`
  background-color: var(--admiral-color-Opacity_Hover, ${p=>p.theme.color["Opacity/Hover"]});
  cursor: pointer;
`,Item=styled__default.default.li`
  display: flex;
  align-items: center;
  user-select: none;
  flex-flow: wrap;
  position: relative;
  justify-content: space-between;
  outline: none;
  white-space: pre;
  margin: 0;
  ${mixins.paddings}
  ${mixins.styleTextMixin}
  ${mixins.colorTextMixin}
  ${mixins.backgroundColor}
  ${p=>p.$hovered&&hoverStyle};

  &:hover {
    ${hoverStyle}
  }

  &&[data-disabled='true'] {
    cursor: not-allowed;
    background-color: var(--admiral-color-Special_ElevatedBG, ${p=>p.theme.color["Special/Elevated BG"]});
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
    && *[fill^='#'] {
      fill: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
    }
  }
`,DropDownItem=e__namespace.forwardRef(({children,disabled,hovered,dimension="l",selected=!1,id,...props},ref)=>jsxRuntime.jsx(Item,{ref,$dimension:dimension,tabIndex:disabled?void 0:props.tabIndex||0,selected,$hovered:hovered,id,"data-disabled":disabled,...props,children}));DropDownItem.displayName="DropDownItem",exports.DropDownItem=DropDownItem;
