// @admiral-ds/react-ui/dist/components/DropDownItem/mixins.cjs.js
"use strict";var styled=require("styled-components"),typography=require("../Typography/typography.cjs.js");require("../Typography/defaultTypographyMixin.cjs.js");const styleTextMixin=styled.css`
  ${({$dimension})=>"s"===$dimension?typography.typography["Body/Body 2 Long"]:typography.typography["Body/Body 1 Long"]}
  ul[data-dimension='s'] & {
    ${typography.typography["Body/Body 2 Long"]}
  }
  text-overflow: ellipsis;
`,colorTextMixin=styled.css`
  color: ${({theme,disabled})=>disabled?`var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral90, ${theme.color["Neutral/Neutral 90"]})`};
`,backgroundColor=styled.css`
  background: ${({theme,selected})=>selected?`var(--admiral-color-Opacity_Focus, ${theme.color["Opacity/Focus"]})`:`var(--admiral-color-Special_ElevatedBG, ${theme.color["Special/Elevated BG"]})`};
  ${({id,theme})=>id&&`ul[data-selectedid="${id}"] && {background: var(--admiral-color-Opacity_Focus, ${theme.color["Opacity/Focus"]});}`}
`,paddings=styled.css`
  padding: ${({$dimension})=>{switch($dimension){case"l":default:return"12px 16px";case"m":return"8px 16px";case"s":return"6px 12px"}}};
  ul[data-dimension='m'] && {
    padding: 8px 16px;
  }
  ul[data-dimension='s'] && {
    padding: 6px 12px;
  }
`;exports.backgroundColor=backgroundColor,exports.colorTextMixin=colorTextMixin,exports.paddings=paddings,exports.styleTextMixin=styleTextMixin;
