// @admiral-ds/react-ui/dist/components/Slider/style.cjs.js
"use strict";var styled=require("styled-components"),animation=require("../skeleton/animation.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const Wrapper=styled__default.default.div`
  position: relative;
  -webkit-tap-highlight-color: transparent;
  cursor: ${p=>p.$disabled?"not-allowed":p.$skeleton?"default":"pointer"};
`,skeletonMixin=styled.css`
  ${animation.skeletonAnimationMixin};
  pointer-events: none;
  & > * {
    visibility: hidden;
  }
`,TrackWrapper=styled__default.default.div`
  position: relative;
  box-sizing: border-box;
  width: 100%;
  display: flex;
  align-items: center;
  height: ${({$dimension})=>"m"===$dimension?36:40}px;
  padding-bottom: 20px;
  [data-disabled='true'] && {
    pointer-events: none;
  }

  ${({$skeleton})=>$skeleton&&skeletonMixin};
`,Track=styled__default.default.div`
  height: 2px;
  background: var(--admiral-color-Neutral_Neutral20, ${p=>p.theme.color["Neutral/Neutral 20"]});
  [data-disabled='true'] && {
    background: var(--admiral-color-Neutral_Neutral20, ${p=>p.theme.color["Neutral/Neutral 20"]});
  }
  width: 100%;
`,DefaultTrack=styled__default.default.div`
  height: 2px;
  transform: translateY(-50%);
`,FilledTrack=styled__default.default.div`
  display: block;
  [data-disabled='true'] && {
    background: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }
  position: absolute;
  height: 2px;
  width: 0%;
  background-color: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
  transition: ${({$animation})=>$animation?"all  ease-in 0.3s":"none"};
`,Thumb=styled__default.default.div`
  position: absolute;
  width: ${({$dimension})=>"m"===$dimension?16:20}px;
  height: ${({$dimension})=>"m"===$dimension?16:20}px;
  align-items: center;
  justify-content: center;
  transform: translateX(-50%) translateY(-50%);
  border-radius: 50%;
  top: 2px;
  left: 0;
  transition: ${({$animation})=>$animation?"all  ease-in 0.3s":"none"};
`,ThumbCircle=styled__default.default.div`
  position: relative;
  height: ${({$dimension})=>"m"===$dimension?16:20}px;
  width: ${({$dimension})=>"m"===$dimension?16:20}px;
  background: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
  border-radius: 50%;
  &:before {
    position: absolute;
    content: '';
    height: ${({$dimension})=>"m"===$dimension?6:8}px;
    width: ${({$dimension})=>"m"===$dimension?6:8}px;
    background: var(--admiral-color-Special_StaticWhite, ${p=>p.theme.color["Special/Static White"]});
    top: ${({$dimension})=>"m"===$dimension?5:6}px;
    bottom: ${({$dimension})=>"m"===$dimension?5:6}px;
    left: ${({$dimension})=>"m"===$dimension?5:6}px;
    right: ${({$dimension})=>"m"===$dimension?5:6}px;
    border-radius: 50%;
  }
  &:hover {
    background: var(--admiral-color-Primary_Primary70, ${p=>p.theme.color["Primary/Primary 70"]});
  }
  &:active {
    background: var(--admiral-color-Primary_Primary70, ${p=>p.theme.color["Primary/Primary 70"]});
  }

  [data-disabled='true'] && {
    background: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
    pointer-events: none;
    &:hover,
    &:active {
      background: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
    }
  }
`;exports.DefaultTrack=DefaultTrack,exports.FilledTrack=FilledTrack,exports.Thumb=Thumb,exports.ThumbCircle=ThumbCircle,exports.Track=Track,exports.TrackWrapper=TrackWrapper,exports.Wrapper=Wrapper;
