// @admiral-ds/react-ui/dist/components/Slider/TickMarks.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),styled=require("styled-components"),index=require("../themes/light/index.cjs.js");require("../themes/common/locales.cjs.js");var typography=require("../Typography/typography.cjs.js");require("../Typography/defaultTypographyMixin.cjs.js");var utils=require("../input/NumberInput/utils.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const Wrapper=styled__default.default.div`
  position: absolute;
  left: ${({$position})=>$position}%;
  display: flex;
  align-items: center;
  justify-content: center;
  user-select: none;
`,Content=styled__default.default.div`
  display: flex;
  align-items: center;
  position: absolute;
  top: 16px;
  [data-dimension='m'] && {
    top: 14px;
  }
  white-space: nowrap;
  ${typography.typography["Body/Body 2 Short"]}
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});
  &:hover {
    color: var(--admiral-color-Primary_Primary70, ${p=>p.theme.color["Primary/Primary 70"]});
  }
  &:active {
    color: var(--admiral-color-Primary_Primary70, ${p=>p.theme.color["Primary/Primary 70"]});
  }
  [data-disabled='true'] && {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }
`;exports.Content=Content,exports.TickMarks=({minValue,maxValue,decimal:userDecimal,thousand:userThousand,precision=2,tickMarks,onPointClick,renderTickMark})=>{const theme=styled.useTheme()||index.LIGHT_THEME,thousand=void 0!==userThousand&&utils.validateThousand(userThousand)?userThousand.slice(0,1):utils.getThousandSeparator(theme.currentLocale),decimal=userDecimal?.slice(0,1)??utils.getDecimalSeparator(theme.currentLocale),items=tickMarks.filter(d=>d>=minValue&&d<=maxValue),range=maxValue-minValue;return jsxRuntime.jsx(jsxRuntime.Fragment,{children:items.map((value,index)=>{const key=value.toString()+index,position=(value-minValue)/range*100;let style={};return style=0===index?{left:0}:style,style=index===tickMarks.length-1?{right:0}:style,jsxRuntime.jsx(Wrapper,{$position:position,onMouseDown:e=>{onPointClick(e,value)},children:jsxRuntime.jsx(Content,{style,children:renderTickMark?renderTickMark(utils.fitToCurrency(value,precision,decimal,thousand)):utils.fitToCurrency(value,precision,decimal,thousand)})},key)})})};
