// @admiral-ds/react-ui/dist/components/List/style.cjs.js
"use strict";var styled=require("styled-components"),typography=require("../Typography/typography.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}require("../Typography/defaultTypographyMixin.cjs.js");var styled__default=_interopDefault(styled);const cyrillicStyle=styled.css`
  @counter-style lower-cyrillic {
    system: fixed;
    symbols: а б в г д е ж з и к л м н о п р с т у ф х ц ч ш щ ы э ю я;
  }
  @counter-style upper-cyrillic {
    system: fixed;
    symbols: А Б В Г Д Е Ж З И К Л М Н О П Р С Т У Ф Х Ц Ч Ш Щ Ы Э Ю Я;
  }
`;const listMixin=styled.css`
  display: flex;
  flex-direction: column;
  gap: ${p=>p.$gap};
  padding: 0;
  margin: 0;
  list-style: none;
  counter-reset: admiral-list-counter 0;

  & > li ul,
  & > li ol {
    margin-top: ${p=>p.$gap};
  }
`,listMarkerMixin=styled.css`
  display: inline-flex;
  flex-shrink: 0;
  margin-inline-end: 8px;
  ${p=>function($styleType){switch($styleType){case"lower-letters":return styled.css`
        ${cyrillicStyle}
        content: counter(admiral-list-counter, lower-cyrillic) ')';
      `;case"upper-letters":return styled.css`
        ${cyrillicStyle}
        content: counter(admiral-list-counter, upper-cyrillic) ')';
      `;case"virgule":return styled.css`
        content: '—';
      `;case"numbers":return styled.css`
        content: counters(admiral-list-counter, '.') '.';
      `;case"bullet":return styled.css`
        content: '•';
        /** Размер шрифта, при котором достигается необходимый размер точки */
        font-size: 18px;
        color: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
      `;default:return"content: none;"}}(p.$styleType)}
  height: ${p=>"m"==p.$dimension?24:20}px;
`,OrderedListComponent=styled__default.default.ol`
  ${listMixin}
  & > li::before {
    ${listMarkerMixin}
    justify-content: flex-start;
    min-width: ${p=>"numbers"==p.$styleType?"auto":("m"==p.$dimension?24:20)+"px"};
    ${p=>p.$markerCssMixin}
  }
`,UnorderedListComponent=styled__default.default.ul`
  ${listMixin}
  & > li::before {
    ${listMarkerMixin}
    justify-content: center;
    width: ${p=>"m"==p.$dimension?24:20}px;
    ${p=>p.$markerCssMixin}
  }
`,ListItemComponent=styled__default.default.li`
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});
  counter-increment: admiral-list-counter 1;
  display: inline-flex;

  ol[data-dimension='m'] &,
  ul[data-dimension='m'] & {
    ${typography.typography["Body/Body 1 Long"]}
  }
  ol[data-dimension='s'] &,
  ul[data-dimension='s'] & {
    ${typography.typography["Body/Body 2 Long"]}
  }
`,ListItemContent=styled__default.default.div`
  display: block;
`,Icon=styled__default.default.svg`
  ul[data-dimension='m'] & {
    width: 24px;
    height: 24px;
  }
  ul[data-dimension='s'] & {
    width: 20px;
    height: 20px;
  }
  vertical-align: bottom;
  margin-inline-end: 8px;
  *[fill^='#'] {
    fill: ${p=>p.color?p.color:`var(--admiral-color-Neutral_Neutral50, ${p.theme.color["Neutral/Neutral 50"]})`};
  }
`;exports.Icon=Icon,exports.ListItemComponent=ListItemComponent,exports.ListItemContent=ListItemContent,exports.OrderedListComponent=OrderedListComponent,exports.UnorderedListComponent=UnorderedListComponent;
