// @admiral-ds/react-ui/dist/components/input/Container.cjs.js
"use strict";var styled=require("styled-components"),borderRadius=require("../themes/borderRadius.cjs.js"),animation=require("../skeleton/animation.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const containerHeights=styled.css`
  height: ${({$dimension})=>{switch($dimension){case"xl":return"56px";case"s":return"32px";default:return"40px"}}};
`,skeletonMixin=styled.css`
  ${animation.skeletonAnimationMixin}
  & > * {
    visibility: hidden;
  }
`,Container=styled__default.default.div`
  position: relative;
  display: flex;
  align-items: stretch;
  border: none;
  border-radius: ${p=>p.$skeleton?0:`var(--admiral-border-radius-Medium, ${borderRadius.mediumGroupBorderRadius(p.theme.shape)})`};

  pointer-events: ${p=>p.$skeleton?"none":"all"};
  ${({$skeleton})=>$skeleton&&skeletonMixin};
`,HeightLimitedContainer=styled__default.default(Container)`
  ${containerHeights};
`;exports.Container=Container,exports.HeightLimitedContainer=HeightLimitedContainer,exports.containerHeights=containerHeights,exports.skeletonMixin=skeletonMixin;
