// @admiral-ds/react-ui/dist/components/Anchor/AnchorItem.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),styled=require("styled-components"),typography=require("../Typography/typography.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}require("../Typography/defaultTypographyMixin.cjs.js");var styled__default=_interopDefault(styled);const singleLineMixin=styled.css`
  height: ${({$dimension})=>"m"===$dimension?28:24}px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
`,AnchorItemLink=styled__default.default.a`
  box-sizing: border-box;
  text-decoration: none;
  color: ${p=>p.$active?`var(--admiral-color-Neutral_Neutral90, ${p.theme.color["Neutral/Neutral 90"]})`:`var(--admiral-color-Neutral_Neutral50, ${p.theme.color["Neutral/Neutral 50"]})`};

  padding: ${p=>"m"===p.$dimension?`6px 8px 6px ${18+16*p.$level}px`:`4px 8px 4px ${18+16*p.$level}px`};

  ${p=>"m"===p.$dimension?typography.typography["Body/Body 2 Short"]:typography.typography["Caption/Caption 1"]};

  ${p=>!p.$multilineView&&singleLineMixin}

  &:hover {
    background-color: var(--admiral-color-Opacity_Hover, ${p=>p.theme.color["Opacity/Hover"]});
  }

  &:active {
    background-color: var(--admiral-color-Opacity_Press, ${p=>p.theme.color["Opacity/Press"]});
  }
`;AnchorItemLink.displayName="AnchorItemLink";const AnchorItem=e.forwardRef(({active,dimension="m",level=0,multilineView=!1,children,...props},ref)=>jsxRuntime.jsx(AnchorItemLink,{...props,$dimension:dimension,$multilineView:multilineView,$active:active,$level:level,ref,children}));AnchorItem.displayName="AnchorItem",exports.ANCHOR_ITEM_HEIGHT_M=28,exports.ANCHOR_ITEM_HEIGHT_S=24,exports.AnchorItem=AnchorItem;
