// @admiral-ds/react-ui/dist/components/Menu/MenuItem.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),styled$1=require("styled-components"),menuItemMixins=require("./menuItemMixins.cjs.js"),styled=require("./styled.cjs.js"),refSetter=require("../common/utils/refSetter.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled$1);const stopEventHandler=e=>{e.preventDefault(),e.stopPropagation()},MenuItem=e.forwardRef(({children,expandIcon=jsxRuntime.jsx(styled.Chevron,{}),hasSubmenu,onHover,disabled,readOnly,hovered,dimension="l",selected=!1,preselected=!1,selfRef,onMouseDown,onLeave,containerRef,...props},ref)=>{const handleMouseDown=disabled?stopEventHandler:onMouseDown,resolvedRef=selfRef?refSetter.refSetter(ref,selfRef):ref;return jsxRuntime.jsxs(Item,{ref:resolvedRef,$dimension:dimension,$preselected:preselected,$selected:selected,$hovered:hovered,"data-preselected":preselected,"data-hovered":hovered,"data-disabled":disabled,"data-readonly":readOnly,"data-dimension":dimension,onMouseEnter:e=>{onHover?.(e),props.onMouseEnter?.(e)},onMouseLeave:e=>{onLeave?.(e),props.onMouseLeave?.(e)},onMouseMove:e=>{onHover?.(e),props.onMouseMove?.(e)},onMouseDown:handleMouseDown,...props,children:[e.Children.toArray(children).map((child,index)=>"string"==typeof child?jsxRuntime.jsx(TextWrapper,{children:child},child+index):child),hasSubmenu&&expandIcon]})});MenuItem.displayName="MenuItem";const Item=styled__default.default.div`
  display: flex;
  align-items: center;
  user-select: none;
  position: relative;
  justify-content: space-between;
  outline: none;
  white-space: pre;
  margin: 0;
  cursor: pointer;
  box-sizing: border-box;
  ${menuItemMixins.paddings}
  ${menuItemMixins.styleTextMixin}
  ${menuItemMixins.colorTextMixin}
  ${menuItemMixins.backgroundColor}

  &&[data-disabled='true'] {
    cursor: not-allowed;
    background-color: var(--admiral-color-Special_ElevatedBG, ${p=>p.theme.color["Special/Elevated BG"]});
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
    && *[fill^='#'] {
      fill: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
    }
    && input[type='checkbox'] + * [fill^='#'] {
      fill: var(--admiral-color-Neutral_Neutral00, ${p=>p.theme.color["Neutral/Neutral 00"]});
    }
  }

  &&[data-readonly='true'] {
    cursor: default;
  }
`,TextWrapper=styled__default.default.span`
  overflow: hidden;
  text-overflow: ellipsis;
`;exports.MenuItem=MenuItem;
