// @admiral-ds/react-ui/dist/components/Toast/index.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),styled=require("styled-components"),useToast=require("./useToast.cjs.js"),ToastProvider=require("./ToastProvider.cjs.js"),ToastItem=require("./ToastItem.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const Container=styled__default.default.div`
  position: fixed;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  width: 384px;
  z-index: var(--admiral-z-index-notification, ${({theme})=>theme.zIndex.notification});
  ${p=>"top-right"===p.$position&&"top: 16px"};
  ${p=>("bottom-right"===p.$position||"bottom-left"===p.$position)&&"bottom: 16px"};
  ${p=>("top-right"===p.$position||"bottom-right"===p.$position)&&"right: 16px"};
  ${p=>"bottom-left"===p.$position&&"left: 16px"};
  pointer-events: none;
  > * {
    pointer-events: initial;
  }
`,fadeInRight=styled.keyframes`
  from {
    transform: translateX(100%);

  }
  to {
    transform: translateX(0);
  }
`,fadeInLeft=styled.keyframes`
  from {
    transform: translateX(-100%);

  }
  to {
    transform: translateX(0);
  }
`,fadeMixin=styled.css`
  animation-name: ${({$position})=>"bottom-left"===$position?fadeInLeft:fadeInRight};
`,Transition=styled__default.default.div`
  margin-bottom: 16px;
  animation-duration: 1s;
  animation-timing-function: ease-out;
  ${fadeMixin}
`,Toast=({position="top-right",...props})=>{const{toastItemList}=useToast.useToast();return jsxRuntime.jsx(Container,{$position:position,...props,children:!!toastItemList?.length&&toastItemList.map(({renderToast,id})=>jsxRuntime.jsx(Transition,{$position:position,children:renderToast(id)},id))})};Toast.displayName="Toast",exports.useToast=useToast.useToast,exports.ToastContext=ToastProvider.ToastContext,exports.ToastProvider=ToastProvider.ToastProvider,exports.DefaultToastItem=ToastItem.DefaultToastItem,exports.ToastItemWithAutoDelete=ToastItem.ToastItemWithAutoDelete,exports.ToastItemWithProgress=ToastItem.ToastItemWithProgress,exports.Toast=Toast;
