// @admiral-ds/react-ui/dist/components/Field/index.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),styled=require("styled-components"),refSetter=require("../common/utils/refSetter.cjs.js"),Container$1=require("../input/Container.cjs.js"),CharacterCounter=require("../input/TextArea/CharacterCounter.cjs.js"),index=require("../Label/index.cjs.js"),uid=require("../common/uid.cjs.js"),typography=require("../Typography/typography.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}require("../Typography/defaultTypographyMixin.cjs.js");var styled__default=_interopDefault(styled);const ContentWrapper=styled__default.default.div``,inlineMixin=styled.css`
  flex-direction: row;
  align-items: center;

  ${index.Label} {
    margin: 0 8px 0 0;
  }

  ${ContentWrapper} {
    flex: 1 1 auto;
  }
`,Container=styled__default.default.div`
  display: flex;
  overflow: hidden;
  ${props=>props.$displayInline?inlineMixin:"flex-direction: column;"}
`,SkeletonLabel=styled__default.default.div`
  position: absolute;
  top: 0;
  left: 0;
  height: 16px;
  width: 50%;
  margin-bottom: 8px;
  ${Container$1.skeletonMixin}
`,LabelContainer=styled__default.default.div`
  position: relative;
  text-align: left;
`,textSkeletonMixin=styled.css`
  color: transparent;
`,StyledLabel=styled__default.default(index.Label)`
  ${p=>p.$skeleton&&textSkeletonMixin};
`,containerSkeletonMixin=styled.css`
  visibility: hidden;
`,ExtrasContainer=styled__default.default.div`
  display: flex;
  justify-content: space-between;
  ${p=>p.$skeleton&&containerSkeletonMixin};
`,ExtraTextContainer=styled__default.default.div`
  flex: 1 1 auto;

  padding-top: 8px;

  text-align: left;

  ${typography.typography["Body/Body 2 Short"]}

  color: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});

  [data-disabled] & {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }

  [data-status='error'] & {
    color: var(--admiral-color-Error_Error60Main, ${p=>p.theme.color["Error/Error 60 Main"]});
  }

  [data-status='success'] & {
    color: var(--admiral-color-Success_Success50Main, ${p=>p.theme.color["Success/Success 50 Main"]});
  }
`,PositionedCharacterCounter=styled__default.default(CharacterCounter.CharacterCounter)`
  flex: 0 0 auto;
  padding: 8px 0 0 8px;
`,Field=e.forwardRef(({children,maxLength,inputRef,displayCharacterCounter=!0,characterCounterVisibilityThreshold=.8,...props},ref)=>{const containerRef=e.useRef(null),[hasFocus,setFocus]=e.useState(!1),{className,style,displayInline,status,extraText,label,required,disabled,id,skeleton=!1,...restFieldProps}=props,[defaultID]=e.useState(uid.uid()),fieldContainerProps={className,style,...restFieldProps},labelProps={htmlFor:id??defaultID,children:label,required,disabled};return e.useEffect(()=>{const onFocusIn=()=>{setFocus(!0)},onFocusOut=()=>{setFocus(!1)},containerNode=containerRef.current;return containerNode&&(containerNode.addEventListener("focusin",onFocusIn),containerNode.addEventListener("focusout",onFocusOut)),()=>{containerNode&&(containerNode.removeEventListener("focusin",onFocusIn),containerNode.removeEventListener("focusout",onFocusOut))}},[]),jsxRuntime.jsxs(Container,{...fieldContainerProps,$displayInline:displayInline,"data-status":status,"data-focus-within":hasFocus?"":void 0,"data-required-within":required?"":void 0,"data-disabled":disabled?"":void 0,"data-read-only":!!props.readOnly||void 0,ref:refSetter.refSetter(containerRef,ref),children:[labelProps.children&&jsxRuntime.jsxs(LabelContainer,{children:[skeleton&&jsxRuntime.jsx(SkeletonLabel,{}),jsxRuntime.jsx(StyledLabel,{$skeleton:skeleton,...labelProps})]}),jsxRuntime.jsxs(ContentWrapper,{children:[children,jsxRuntime.jsxs(ExtrasContainer,{$skeleton:skeleton,children:[extraText&&jsxRuntime.jsx(ExtraTextContainer,{children:extraText}),displayCharacterCounter&&inputRef&&void 0!==maxLength&&jsxRuntime.jsxs(jsxRuntime.Fragment,{children:[jsxRuntime.jsx("div",{}),jsxRuntime.jsx(PositionedCharacterCounter,{maxLength,visibilityThreshold:characterCounterVisibilityThreshold,inputRef})]})]})]})]})});Field.displayName="Field",exports.ExtraTextContainer=ExtraTextContainer,exports.ExtrasContainer=ExtrasContainer,exports.Field=Field;
