// @admiral-ds/react-ui/dist/components/Carousel/CarouselButton.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),styled=require("styled-components"),ChevronLeftOutline=require("../../node_modules/@admiral-ds/icons/build/system/ChevronLeftOutline.svg.cjs.js"),ChevronRightOutline=require("../../node_modules/@admiral-ds/icons/build/system/ChevronRightOutline.svg.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const IconWrapper=styled__default.default.div`
  display: flex;
  align-items: center;
  justify-content: center;

  box-sizing: border-box;
  width: ${40}px;
  height: ${40}px;
  border-radius: 50%;

  background-color: transparent;

  & > svg {
    width: ${24}px;
    height: ${24}px;
  }
  & *[fill^='#'] {
    fill: ${p=>"default"===p.$appearance?`var(--admiral-color-Special_StaticWhite, ${p.theme.color["Special/Static White"]})`:`var(--admiral-color-Neutral_Neutral50, ${p.theme.color["Neutral/Neutral 50"]})`};
  }
`,StyledButton=styled__default.default.button`
  all: unset;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;

  position: absolute;
  ${p=>"left"===p.$direction?"left: 0;":"right: 0;"}
  width: ${56}px;
  height: 100%;

  background-color: transparent;

  &:hover,
  &:active {
    & > ${IconWrapper} {
      background-color: ${p=>"primary"===p.$appearance?`var(--admiral-color-Opacity_Hover, ${p.theme.color["Opacity/Hover"]})`:`var(--admiral-color-Opacity_Modal, ${p.theme.color["Opacity/Modal"]})`};
    }
  }
`;exports.CarouselButton=({appearance,direction,...props})=>jsxRuntime.jsx(StyledButton,{...props,$direction:direction,$appearance:appearance,children:jsxRuntime.jsx(IconWrapper,{$appearance:appearance,children:"left"===direction?jsxRuntime.jsx(ChevronLeftOutline.ReactComponent,{}):jsxRuntime.jsx(ChevronRightOutline.ReactComponent,{})})});
