// @admiral-ds/react-ui/dist/components/SelectTree/SelectTreeNode.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),styled=require("styled-components"),typography=require("../Typography/typography.cjs.js");require("../Typography/defaultTypographyMixin.cjs.js"),require("../form/InputField/index.cjs.js");var index=require("../form/CheckboxField/index.cjs.js");require("../form/DateField/index.cjs.js"),require("../form/SelectField/index.cjs.js"),require("../form/FieldSet/index.cjs.js"),require("../form/SuggestField/index.cjs.js"),require("../form/TextField/index.cjs.js"),require("../form/SliderInputField/index.cjs.js"),require("../form/SliderRangeField/index.cjs.js"),require("../form/TimeField/index.cjs.js"),require("../form/NumberInputField/index.cjs.js"),require("../form/EditModeField/index.cjs.js"),require("../form/EditModeAreaField/index.cjs.js"),require("../form/PhoneInputField/index.cjs.js"),require("../form/InputExField/index.cjs.js"),require("../form/FileInputField/index.cjs.js"),require("../form/MultiInputField/index.cjs.js"),require("../form/TreeSelectField/index.cjs.js");var ChevronRightOutline=require("../../node_modules/@admiral-ds/icons/build/system/ChevronRightOutline.svg.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const Chevron=styled__default.default(ChevronRightOutline.ReactComponent)`
  transition: all 0.3s;
  & path {
    fill: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }
  width: 100%;
  height: 100%;
  transform: ${p=>p.$isOpened?"rotate(90deg)":"rotate(0deg)"};
`,ChevronWrapper=styled__default.default.div`
  flex-shrink: 0;
  margin-right: 16px;
  box-sizing: border-box;
  cursor: pointer;
  position: relative;

  width: ${p=>"m"===p.$dimension?"24px":"20px"};
  height: ${p=>"m"===p.$dimension?"24px":"20px"};

  &:hover {
    &::after {
      content: '';
      position: absolute;
      left: 50%;
      top: 50%;
      transform: translate(-50%, -50%);
      border-radius: 50%;
      width: calc(100% + 12px);
      height: calc(100% + 12px);
      background-color: var(--admiral-color-Opacity_Hover, ${p=>p.theme.color["Opacity/Hover"]});
    }
  }
`,TreeItem=styled__default.default.ul`
  list-style: none;
  margin: 0;
  padding: 0;
  padding-left: ${p=>"m"===p.$dimension?"40px":"36px"};
`,Wrapper=styled__default.default.li`
  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});
  ${p=>"m"===p.$dimension?typography.typography["Body/Body 1 Short"]:typography.typography["Body/Body 2 Short"]};
  display: flex;
  align-items: flex-start;
  padding: ${p=>"m"===p.$dimension?`8px 16px 8px ${40*p.$level}px`:`6px 16px 6px ${36*p.$level}px`};
`,IconWrapper=styled__default.default.div`
  margin-right: 8px;
  flex-shrink: 0;
  width: ${p=>"m"===p.$dimension?"24px":"20px"};
  height: ${p=>"m"===p.$dimension?"24px":"20px"};
  > svg {
    width: 100%;
    height: 100%;
  }
`,StyledCheckbox=styled__default.default(index.CheckboxField)`
  margin-right: 8px;
  flex-shrink: 0;
`,TitleContent=styled__default.default.div`
  padding-top: 2px;
`,SelectTreeNode=({dimension,node,level,onChange,onButtonClick,onKeyDown,expandAll})=>jsxRuntime.jsxs(jsxRuntime.Fragment,{children:[jsxRuntime.jsxs(Wrapper,{$dimension:dimension,$level:level,onKeyDown,children:[node.children&&jsxRuntime.jsx(ChevronWrapper,{"data-key":node.id,$dimension:dimension,onClick:onButtonClick,children:jsxRuntime.jsx(Chevron,{$isOpened:node.expanded})}),"status"in node&&jsxRuntime.jsx(StyledCheckbox,{id:node.id,"data-key":node.id,dimension,indeterminate:"indeterminate"===node.status,checked:"unchecked"!==node.status,onChange}),node.icon&&jsxRuntime.jsx(IconWrapper,{$dimension:dimension,children:node.icon}),jsxRuntime.jsx(TitleContent,{children:node.label})]}),node.expanded&&node.children&&node.children.map((child,i)=>jsxRuntime.jsx(TreeItem,{$dimension:dimension,children:jsxRuntime.jsx(SelectTreeNode,{dimension,node:child,level:level+1,expandAll,onKeyDown,onButtonClick,onChange},node.id)},[node.id,i].join("/")))]});exports.SelectTreeNode=SelectTreeNode;
