// @admiral-ds/react-ui/dist/components/TabMenuComponent/containerElements/SlideArrowButton.cjs.js
"use strict";var styled=require("styled-components"),parseShadowFromTheme=require("../../common/utils/parseShadowFromTheme.cjs.js"),constants=require("../constants.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}const SlideArrowButton=_interopDefault(styled).default.button`
  appearance: none;
  border: none;
  border-radius: 50%;
  width: ${constants.SLIDE_ARROW_BUTTON_SIZE}px;
  height: ${constants.SLIDE_ARROW_BUTTON_SIZE}px;
  background: var(--admiral-color-Special_ElevatedBG, ${p=>p.theme.color["Special/Elevated BG"]});
  box-shadow: var(--admiral-box-shadow-Shadow04, ${p=>parseShadowFromTheme.parseShadow(p.theme.shadow["Shadow 04"])});
  cursor: pointer;

  padding: ${constants.SLIDE_ARROW_BUTTON_PADDING}px;

  & svg {
    display: inline-block;
    flex: 1 0 auto;
    & *[fill^='#'] {
      fill: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
    }
    width: ${constants.SLIDE_ARROW_ICON_SIZE}px;
    height: ${constants.SLIDE_ARROW_ICON_SIZE}px;
    margin: 0;
  }

  &:hover:not(:disabled) {
    background: var(--admiral-color-Neutral_Neutral05, ${p=>p.theme.color["Neutral/Neutral 05"]});
  }
`;SlideArrowButton.displayName="SlideArrowButton",exports.SlideArrowButton=SlideArrowButton;
