// @admiral-ds/react-ui/dist/components/TabMenuComponent/containerElements/VerticalTabOverflowMenu.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),styled=require("styled-components"),index=require("../../OverflowMenu/index.cjs.js"),constants=require("../constants.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const OverflowMenuContainer=styled__default.default.div`
  box-sizing: border-box;
  display: flex;
  justify-content: center;
  visibility: ${p=>p.$isHidden?"hidden":"visible"};
  padding: ${constants.VERTICAL_OVERFLOW_MENU_CONTAINER_PADDING};

  width: 100%;
  height: ${p=>"l"===p.$dimension?constants.VERTICAL_OVERFLOW_MENU_CONTAINER_SIZE_L:constants.VERTICAL_OVERFLOW_MENU_CONTAINER_SIZE_M}px;
`,StyledOverflowMenu=styled__default.default(index.OverflowMenu)`
  &:focus-visible:not(:disabled) {
    outline-offset: ${constants.FOCUS_VERTICAL_OVERFLOW_MENU_BORDER_OFFSET}px;
    border-radius: 0;
  }
`,VerticalTabOverflowMenu=({dimension="l",isHidden=!1,...props})=>jsxRuntime.jsx(OverflowMenuContainer,{$dimension:dimension,$isHidden:isHidden,children:jsxRuntime.jsx(StyledOverflowMenu,{...props,dimension})});VerticalTabOverflowMenu.displayName="VerticalTabOverflowMenu",exports.VerticalTabOverflowMenu=VerticalTabOverflowMenu;
