// @admiral-ds/react-ui/dist/components/GlobalSearch/index.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),styled=require("styled-components"),e=require("react"),borderRadius=require("../themes/borderRadius.cjs.js"),typography=require("../Typography/typography.cjs.js");require("../Typography/defaultTypographyMixin.cjs.js");var index$1=require("../input/TextInput/index.cjs.js"),index=require("../DropdownContainer/index.cjs.js"),index$2=require("../Spinner/index.cjs.js"),SearchOutline=require("../../node_modules/@admiral-ds/icons/build/system/SearchOutline.svg.cjs.js"),CloseOutline=require("../../node_modules/@admiral-ds/icons/build/service/CloseOutline.svg.cjs.js"),InputIconButton=require("../InputIconButton.cjs.js"),changeInputData=require("../common/dom/changeInputData.cjs.js"),index$3=require("../Menu/index.cjs.js"),SuffixSelect=require("../input/InputEx/SuffixSelect.cjs.js"),refSetter=require("../common/utils/refSetter.cjs.js"),parseShadowFromTheme=require("../common/utils/parseShadowFromTheme.cjs.js"),keyboardKey=require("../common/keyboardKey.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const Drop=styled__default.default(index.DropdownContainer)`
  padding: 8px 0px;
  box-shadow: var(--admiral-box-shadow-Shadow08, ${p=>parseShadowFromTheme.parseShadow(p.theme.shadow["Shadow 08"])});
  background-color: var(--admiral-color-Special_ElevatedBG, ${p=>p.theme.color["Special/Elevated BG"]});
  border-radius: var(--admiral-border-radius-Medium, ${p=>borderRadius.mediumGroupBorderRadius(p.theme.shape)});
`,IconPanel=styled__default.default.div`
  flex: 0 0 auto;
  gap: 8px;
  display: flex;
  align-items: center;

  & > svg {
    border-radius: var(--admiral-border-radius-Medium, ${p=>borderRadius.mediumGroupBorderRadius(p.theme.shape)});
    display: block;
    width: ${props=>{switch(props.$dimension){case"xl":default:return 24;case"s":return 20}}}px;

    &:focus {
      outline: none;
    }

    &:focus-visible {
      outline-offset: 2px;
      outline: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]}) solid 2px;
    }
  }
`,Container=styled__default.default.div`
  min-width: 280px;
  display: inline-flex;
  gap: 8px;
  align-items: stretch;
  flex: 1 0 0;
  border-radius: var(--admiral-border-radius-Medium, ${p=>borderRadius.mediumGroupBorderRadius(p.theme.shape)});
  border: 2px solid var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});
  background: var(--admiral-color-Neutral_Neutral00, ${p=>p.theme.color["Neutral/Neutral 00"]});

  height: 40px;
  padding-left: 16px;
  ${typography.typography["Button/Button 1"]}

  &[data-dimension='xl'] {
    height: 56px;
  }

  &[data-dimension='s'] {
    height: 32px;
    padding-left: 12px;
    ${typography.typography["Button/Button 2"]}
  }
`,ieFixes=styled.css`
  ::-ms-clear,
  ::-ms-reveal {
    display: none;
  }
`,Input=styled__default.default.input`
  outline: none;
  appearance: none;
  border-radius: inherit;

  box-sizing: border-box;
  flex: 1 1 auto;
  min-width: 10px;
  border: none;
  background: transparent;
  text-overflow: ellipsis;
  padding: 0;

  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});

  &&&:disabled {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }

  ${props=>"s"===props.$dimension?typography.typography["Body/Body 2 Long"]:typography.typography["Body/Body 1 Long"]}
  &::placeholder {
    color: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }

  &:disabled::placeholder {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }

  [data-disable-copying] & {
    user-select: none;
    pointer-events: none;
  }

  &&&:user-invalid + ${index$1.InputBorderedDiv} {
    border: 1px solid var(--admiral-color-Error_Error60Main, ${p=>p.theme.color["Error/Error 60 Main"]});
  }

  &&&:user-invalid:hover:not(:disabled) + ${index$1.InputBorderedDiv} {
    border: 1px solid var(--admiral-color-Error_Error70, ${p=>p.theme.color["Error/Error 70"]});
  }

  &&&:user-invalid:focus:not(:disabled) + ${index$1.InputBorderedDiv} {
    border: 2px solid var(--admiral-color-Error_Error60Main, ${p=>p.theme.color["Error/Error 60 Main"]});
  }

  ${ieFixes}
`,PrefixContainer=styled__default.default.div`
  align-self: center;
  border-right: solid 1px var(--admiral-color-Neutral_Neutral20, ${p=>p.theme.color["Neutral/Neutral 20"]});
  padding-right: 8px;
  margin-right: 8px;
  ${props=>"s"===props.$dimension?typography.typography["Body/Body 2 Long"]:typography.typography["Body/Body 1 Long"]};
  color: ${p=>p.disabled?`var(--admiral-color-Neutral_Neutral30, ${p.theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral50, ${p.theme.color["Neutral/Neutral 50"]})`};
  white-space: nowrap;
`,SubmitButton=styled__default.default.div`
  display: flex;
  user-select: none;
  justify-content: center;
  align-items: flex-start;
  gap: 8px;

  background: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]});

  color: var(--admiral-color-Special_StaticWhite, ${p=>p.theme.color["Special/Static White"]});
  &&& *[fill^='#'] {
    fill: var(--admiral-color-Special_StaticWhite, ${p=>p.theme.color["Special/Static White"]});
  }

  &:hover {
    cursor: pointer;
  }

  padding: 8px 16px;

  & > svg {
    width: 24px;
    height: 24px;
  }

  [data-dimension='xl'] & {
    padding: 16px 24px;
    & > svg {
      width: 24px;
      height: 24px;
    }
  }

  [data-dimension='s'] & {
    padding: 6px 12px;
    & > svg {
      width: 20px;
      height: 20px;
    }
  }
`;exports.GlobalSearch=({dimension,alignDropRef,targetElement,placeholder,id,value,defaultValue,onChange,submitButtonProps={},icons,displayClearIcon,isLoading,model,prefixValue,renderPrefixValue=({value})=>value?String(value):value,prefixValueList,onPrefixValueChange,renderPrefixOption,prefixDropContainerStyle,...props})=>{const[needSubmit,setNeedSubmit]=e.useState(0),[tempValue,setTempValue]=e.useState(""),[displayOptionsVisible,setDisplayOptionsVisible]=e.useState(!1),[inFocus,setInFocus]=e.useState(!1),inputRef=e.useRef(null),containerRef=e.useRef(null),submitButtonRef=e.useRef(null),inputProps={placeholder,id,value,defaultValue,onChange};e.useEffect(()=>{if(inFocus){const listener=event=>{"Enter"!==event.code&&"NumpadEnter"!==event.code||(event.preventDefault(),event.stopPropagation(),submitButtonRef.current&&submitButtonRef.current.click()),"ArrowDown"===event.code&&(displayOptionsVisible||setDisplayOptionsVisible(!0))};return document.addEventListener("keydown",listener),()=>{document.removeEventListener("keydown",listener)}}},[inFocus,displayOptionsVisible,tempValue,model]);const iconArray=e.Children.toArray(icons);displayClearIcon&&value&&iconArray.unshift(jsxRuntime.jsx(InputIconButton.InputIconButton,{icon:CloseOutline.ReactComponent,onMouseDown:e=>{e.stopPropagation(),e.preventDefault();const inputEl=inputRef.current;inputEl&&changeInputData.changeInputData(inputEl,{value:""})},"aria-hidden":!0},"clear-icon")),isLoading&&iconArray.unshift(jsxRuntime.jsx(index$2.Spinner,{dimension:"s"===dimension?"ms":"m"},"loading-icon"));const iconCount=iconArray.length;e.useEffect(()=>{needSubmit>0&&submitButtonRef.current&&submitButtonRef.current.click()},[needSubmit]);const innerContainerRef=e.useRef(null),targetNode=targetElement||alignDropRef?.current||innerContainerRef.current,menuDimension="xl"===dimension?"l":dimension,renderPrefix=prefixValueList?props=>jsxRuntime.jsx(SuffixSelect.SuffixSelect,{alignSelf:"flex-start",dimension:menuDimension,targetElement:targetNode,value:props.value||"",onChange:value=>onPrefixValueChange?.(value),options:prefixValueList,disabled:props.disabled,readOnly:props.readOnly,dropContainerCssMixin:prefixDropContainerStyle?.dropContainerCssMixin,dropContainerClassName:prefixDropContainerStyle?.dropContainerClassName,dropContainerStyle:prefixDropContainerStyle?.dropContainerStyle,renderValue:renderPrefixValue,renderOption:renderPrefixOption}):renderPrefixValue,prefix=renderPrefix({value:prefixValue,disabled:props.disabled,readOnly:props.readOnly});return jsxRuntime.jsxs(Container,{"data-dimension":dimension,...props,ref:refSetter.refSetter(innerContainerRef,containerRef),onFocus:()=>setInFocus(!0),onBlur:()=>{setInFocus(!1),setDisplayOptionsVisible(!1),setTempValue("")},children:[!!prefix&&jsxRuntime.jsx(PrefixContainer,{$dimension:dimension,disabled:props.disabled,children:prefix}),jsxRuntime.jsx(Input,{...inputProps,onChange:e=>{setDisplayOptionsVisible(!0),setTempValue(""),inputProps.onChange?.(e.target.value)},onKeyDown:e=>{keyboardKey.keyboardKey.getCode(e)===keyboardKey.keyboardKey[" "]&&e.stopPropagation()},ref:inputRef}),iconCount>0?jsxRuntime.jsx(IconPanel,{disabled:props.disabled,$dimension:dimension,children:iconArray}):null,jsxRuntime.jsx(SubmitButton,{...submitButtonProps,children:submitButtonProps.children??jsxRuntime.jsx(SearchOutline.ReactComponent,{}),onClick:e=>{submitButtonProps.onClick?.(e),setDisplayOptionsVisible(!1)},ref:submitButtonRef}),displayOptionsVisible&&jsxRuntime.jsx(Drop,{alignSelf:"stretch",targetElement:containerRef.current,children:jsxRuntime.jsx(index$3.Menu,{dimension:"xl"===dimension?"l":dimension,maxHeight:"496px",rowCount:10,model,onSelectItem:id=>{setTempValue(""),setNeedSubmit(val=>val+1),inputProps.onChange?.(id)},active:tempValue,onActivateItem:id=>{setTempValue(id??"")},onMouseLeave:()=>{setTempValue("")},defaultIsActive:!0,disableSelectedOptionHighlight:!0,preventFocusSteal:!0})})]})};
