// @admiral-ds/react-ui/dist/components/input/TextInput/index.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),styled=require("styled-components"),changeInputData=require("../../common/dom/changeInputData.cjs.js"),refSetter=require("../../common/utils/refSetter.cjs.js"),Container=require("../Container.cjs.js"),typography=require("../../Typography/typography.cjs.js");require("../../Typography/defaultTypographyMixin.cjs.js");var CloseOutline=require("../../../node_modules/@admiral-ds/icons/build/service/CloseOutline.svg.cjs.js"),EyeCloseOutline=require("../../../node_modules/@admiral-ds/icons/build/service/EyeCloseOutline.svg.cjs.js"),EyeOutline=require("../../../node_modules/@admiral-ds/icons/build/service/EyeOutline.svg.cjs.js"),borderRadius=require("../../themes/borderRadius.cjs.js"),InputIconButton=require("../../InputIconButton.cjs.js"),index=require("../../Spinner/index.cjs.js"),index$1=require("../../Tooltip/index.cjs.js"),checkOverflow=require("../../common/utils/checkOverflow.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const iconSizeValue=props=>{switch(props.$dimension){case"xl":default:return 24;case"s":return 20}},horizontalPaddingValue=props=>{switch(props.$dimension){case"xl":default:return 16;case"s":return 12}},extraPadding=styled.css`
  padding-right: ${props=>horizontalPaddingValue(props)+(iconSizeValue(props)+8)*(props.$iconsAfterCount??0)}px;
  padding-left: ${props=>horizontalPaddingValue(props)+(iconSizeValue(props)+8)*(props.$iconsBeforeCount??0)}px;
`,disabledColors=styled.css`
  background-color: var(--admiral-color-Neutral_Neutral10, ${p=>p.theme.color["Neutral/Neutral 10"]});
  border-color: transparent;
`,getBorderColor=styled.css`
  ${({$status,theme})=>{if(!$status)return`var(--admiral-color-Neutral_Neutral40, ${theme.color["Neutral/Neutral 40"]})`;switch($status){case"error":return`var(--admiral-color-Error_Error60Main, ${theme.color["Error/Error 60 Main"]})`;case"success":return`var(--admiral-color-Success_Success50Main, ${theme.color["Success/Success 50 Main"]})`}}}
`,InputBorderedDiv=styled__default.default.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  margin: 0;
  pointer-events: none;
  overflow: hidden;
  min-width: 0;

  background: none;
  border-radius: inherit;

  border: 1px solid ${getBorderColor};
  ${p=>p.disabled&&"border-color: transparent;"};
`,getHoverBorderColor=styled.css`
  border-color: ${({$status,theme})=>{if(!$status)return`var(--admiral-color-Neutral_Neutral60, ${theme.color["Neutral/Neutral 60"]})`;switch($status){case"error":return`var(--admiral-color-Error_Error70, ${theme.color["Error/Error 70"]})`;case"success":return`var(--admiral-color-Success_Success60, ${theme.color["Success/Success 60"]})`}}};
`,getFocusBorder=styled.css`
  border: 2px solid
    ${({$status,theme})=>{if(!$status)return`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`;switch($status){case"error":return`var(--admiral-color-Error_Error60Main, ${theme.color["Error/Error 60 Main"]})`;case"success":return`var(--admiral-color-Success_Success50Main, ${theme.color["Success/Success 50 Main"]})`}}};
`,BorderedDivStyles=styled.css`
  &:focus-within:not(:disabled) > ${InputBorderedDiv} {
    ${p=>p.disabled||p.readOnly?"border-color: transparent":p.$isLoading?"":getFocusBorder}
  }

  &:hover:not(:focus-within) > ${InputBorderedDiv} {
    ${p=>p.$isLoading?"":p.disabled||p.readOnly?"transparent":getHoverBorderColor};
  }
`,ieFixes=styled.css`
  ::-ms-clear,
  ::-ms-reveal {
    display: none;
  }
`,Input=styled__default.default.input`
  outline: none;
  appearance: none;
  border-radius: inherit;

  box-sizing: border-box;
  flex: 1 1 auto;
  min-width: 10px;
  border: none;
  text-overflow: ellipsis;
  padding: 0 ${horizontalPaddingValue}px;

  ${props=>"s"===props.$dimension?typography.typography["Body/Body 2 Long"]:typography.typography["Body/Body 1 Long"]}

  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});

  &&&:disabled {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }

  &::placeholder {
    color: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }

  &:disabled::placeholder {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }

  [data-disable-copying] & {
    user-select: none;
    &::selection {
      background-color: transparent;
    }
  }

  background-color: var(--admiral-color-Neutral_Neutral00, ${p=>p.theme.color["Neutral/Neutral 00"]});

  &&&:user-invalid + ${InputBorderedDiv} {
    border: 1px solid var(--admiral-color-Error_Error60Main, ${p=>p.theme.color["Error/Error 60 Main"]});
  }

  &&&:user-invalid:hover:not(:disabled) + ${InputBorderedDiv} {
    border: 1px solid var(--admiral-color-Error_Error70, ${p=>p.theme.color["Error/Error 70"]});
  }

  &&&:user-invalid:focus:not(:disabled) + ${InputBorderedDiv} {
    border: 2px solid var(--admiral-color-Error_Error60Main, ${p=>p.theme.color["Error/Error 60 Main"]});
  }

  [data-read-only] &&&,
  &&&:disabled {
    ${disabledColors}
  }

  [data-loading] &&&,
  &&&:disabled {
    cursor: not-allowed;
    pointer-events: none;
  }

  ${extraPadding}
  ${ieFixes}
`,IconPanel=styled__default.default.div`
  position: absolute;
  top: 0;
  bottom: 0;

  display: flex;
  align-items: center;

  & svg {
    border-radius: var(--admiral-border-radius-Medium, ${p=>borderRadius.mediumGroupBorderRadius(p.theme.shape)});
    display: block;
    width: ${iconSizeValue}px;

    &:focus {
      outline: none;
    }

    &:focus-visible {
      outline-offset: 2px;
      outline: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]}) solid 2px;
    }
  }
`,IconPanelBefore=styled__default.default(IconPanel)`
  left: 0;

  padding-left: ${horizontalPaddingValue}px;

  & > *:not(:first-child) {
    margin-right: 8px;
  }
`,IconPanelAfter=styled__default.default(IconPanel)`
  right: 0;

  padding-right: ${horizontalPaddingValue}px;

  & > *:not(:first-child) {
    margin-left: 8px;
  }
`,StyledContainer=styled__default.default(Container.HeightLimitedContainer)`
  ${BorderedDivStyles}
  ${p=>p.disabled?"cursor: not-allowed;":p.$isLoading?"cursor: default;":""}
`;function defaultHandleInput(newInputData){return newInputData||{}}const TextInput=e.forwardRef(({dimension="m",type,displayClearIcon,isLoading,status,handleInput=defaultHandleInput,containerRef,icons,iconsBefore,iconsAfter,children,className,style,placeholder,skeleton=!1,showTooltip=!0,disableCopying,...props},ref)=>{const inputRef=e.useRef(null),wrapperRef=containerRef||e.useRef(null),iconAfterArray=e.Children.toArray(iconsAfter||icons),iconBeforeArray=e.Children.toArray(iconsBefore),[overflowActive,setOverflowActive]=e.useState(!1),[tooltipVisible,setTooltipVisible]=e.useState(!1),[innerValueState,setInnerValueState]=e.useState(props.defaultValue??void 0),innerValue=props.value??innerValueState;e.useEffect(()=>{checkOverflow.checkOverflow(inputRef.current)?setOverflowActive(!0):setOverflowActive(!1)},[tooltipVisible,setOverflowActive]),e.useLayoutEffect(()=>{function show(){document.activeElement!==inputRef.current&&setTooltipVisible(!0)}function hide(){setTooltipVisible(!1)}const wrapper=wrapperRef.current;if(wrapper)return wrapper.addEventListener("mouseenter",show),wrapper.addEventListener("mouseleave",hide),wrapper.addEventListener("mousedown",hide),()=>{wrapper.removeEventListener("mouseenter",show),wrapper.removeEventListener("mouseleave",hide),wrapper.removeEventListener("mousedown",hide)}},[setTooltipVisible]);const[isPasswordVisible,setPasswordVisible]=e.useState(!1);if(!props.readOnly&&"password"===type){const Icon=isPasswordVisible?EyeOutline.ReactComponent:EyeCloseOutline.ReactComponent;iconAfterArray.push(jsxRuntime.jsx(InputIconButton.InputIconButton,{icon:Icon,onClick:()=>{setPasswordVisible(!isPasswordVisible)},"aria-hidden":!0},"eye-icon"))}!props.readOnly&&displayClearIcon&&innerValue&&iconAfterArray.unshift(jsxRuntime.jsx(InputIconButton.InputIconButton,{icon:CloseOutline.ReactComponent,onMouseDown:e=>{e.preventDefault()},onClick:()=>{inputRef.current&&changeInputData.changeInputData(inputRef.current,{value:""})},"aria-hidden":!0},"clear-icon")),isLoading&&iconAfterArray.unshift(jsxRuntime.jsx(index.Spinner,{dimension:"s"===dimension?"ms":"m"},"loading-icon"));const iconsBeforeCount=iconBeforeArray.length,iconsAfterCount=iconAfterArray.length;return e.useLayoutEffect(()=>{const nullHandledValue=handleInput(null);function oninput(e){const{value,selectionStart,selectionEnd}=this,inputData=handleInput({value,selectionStart,selectionEnd},e);placeholder&&!changeInputData.isInputDataDifferent(nullHandledValue,inputData)?changeInputData.changeInputData(this,{...inputData,value:""}):changeInputData.changeInputData(this,inputData)}if("date"!==type&&inputRef.current){const node=inputRef.current;node.addEventListener("input",oninput);const{value,selectionStart,selectionEnd}=node,inputData=handleInput({value,selectionStart,selectionEnd});return placeholder&&!changeInputData.isInputDataDifferent(nullHandledValue,inputData)?changeInputData.changeInputData(node,{...inputData,value:""}):changeInputData.changeInputData(node,inputData),()=>{node.removeEventListener("input",oninput)}}},[handleInput,placeholder]),e.useEffect(()=>{function select(){this.selectionEnd=this.selectionStart}if(disableCopying&&inputRef.current){const node=inputRef.current;return node.addEventListener("select",select,!0),()=>node.removeEventListener("select",select,!0)}},[disableCopying]),jsxRuntime.jsxs(jsxRuntime.Fragment,{children:[jsxRuntime.jsxs(StyledContainer,{className,style,$dimension:dimension,ref:wrapperRef,disabled:props.disabled,readOnly:props.readOnly,$isLoading:isLoading,$status:status,"data-disabled":!!props.disabled||void 0,"data-read-only":!!props.readOnly||void 0,"data-loading":!!isLoading||void 0,"data-status":status,$skeleton:skeleton,"data-disable-copying":!!disableCopying||void 0,children:[jsxRuntime.jsx(Input,{ref:refSetter.refSetter(ref,inputRef),...props,onChange:e=>{setInnerValueState(e.currentTarget.value),props.onChange?.(e)},placeholder,$dimension:dimension,$iconsAfterCount:iconsAfterCount,$iconsBeforeCount:iconsBeforeCount,type:"password"===type&&isPasswordVisible?"text":type}),jsxRuntime.jsx(InputBorderedDiv,{$status:status,disabled:props.disabled||props.readOnly}),iconsBeforeCount>0&&jsxRuntime.jsx(IconPanelBefore,{disabled:props.disabled,$dimension:dimension,children:iconBeforeArray}),iconsAfterCount>0&&jsxRuntime.jsx(IconPanelAfter,{disabled:props.disabled,$dimension:dimension,children:iconAfterArray}),children]}),showTooltip&&tooltipVisible&&overflowActive&&jsxRuntime.jsx(index$1.Tooltip,{renderContent:()=>inputRef?.current?.value||"",targetElement:wrapperRef.current})]})});TextInput.displayName="TextInput",exports.BorderedDivStyles=BorderedDivStyles,exports.InputBorderedDiv=InputBorderedDiv,exports.TextInput=TextInput;
