// @admiral-ds/react-ui/dist/components/Link/mixins.cjs.js
"use strict";var styled=require("styled-components"),typography=require("../Typography/typography.cjs.js");require("../Typography/defaultTypographyMixin.cjs.js"),require("../themes/common/locales.cjs.js");var borderRadius=require("../themes/borderRadius.cjs.js");const styleDisabledMixin=styled.css`
  color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  & *[fill^='#'] {
    fill: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }
  user-select: none;
`,styleColorMixin=styled.css`
  color: ${({theme,$appearance})=>"secondary"===$appearance?`var(--admiral-color-Neutral_Neutral90, ${theme.color["Neutral/Neutral 90"]})`:`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`};
  & *[fill^='#'] {
    fill: ${({theme,$appearance})=>"secondary"===$appearance?`var(--admiral-color-Neutral_Neutral90, ${theme.color["Neutral/Neutral 90"]})`:`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`};
  }
`,styleMixin=styled.css`
  border-radius: var(--admiral-border-radius-Medium, ${p=>borderRadius.mediumGroupBorderRadius(p.theme.shape)});
`,styleEventMixin=styled.css`
  &:hover {
    color: var(--admiral-color-Primary_Primary70, ${p=>p.theme.color["Primary/Primary 70"]});
    & *[fill^='#'] {
      fill: var(--admiral-color-Primary_Primary70, ${p=>p.theme.color["Primary/Primary 70"]});
    }
  }
  &:focus {
    color: var(--admiral-color-Primary_Primary70, ${p=>p.theme.color["Primary/Primary 70"]});
    & *[fill^='#'] {
      fill: var(--admiral-color-Primary_Primary70, ${p=>p.theme.color["Primary/Primary 70"]});
    }
    outline-color: var(--admiral-color-Primary_Primary70, ${p=>p.theme.color["Primary/Primary 70"]});
  }
  &:active {
    color: var(--admiral-color-Primary_Primary80, ${p=>p.theme.color["Primary/Primary 80"]});
    & *[fill^='#'] {
      fill: var(--admiral-color-Primary_Primary80, ${p=>p.theme.color["Primary/Primary 80"]});
    }
  }

  &:focus-visible {
    outline-offset: 2px;
    outline: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]}) solid 2px;
  }
`,styleTextMixin=styled.css`
  ${({$dimension})=>"s"===$dimension?typography.typography["Body/Body 2 Long"]:typography.typography["Body/Body 1 Long"]}
`;exports.styleColorMixin=styleColorMixin,exports.styleDisabledMixin=styleDisabledMixin,exports.styleEventMixin=styleEventMixin,exports.styleMixin=styleMixin,exports.styleTextMixin=styleTextMixin;
