// @admiral-ds/react-ui/dist/components/input/InputEx/index.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),styled=require("styled-components"),CloseOutline=require("../../../node_modules/@admiral-ds/icons/build/service/CloseOutline.svg.cjs.js"),Container$1=require("../Container.cjs.js"),typography=require("../../Typography/typography.cjs.js");require("../../Typography/defaultTypographyMixin.cjs.js");var changeInputData=require("../../common/dom/changeInputData.cjs.js"),InputIconButton=require("../../InputIconButton.cjs.js"),refSetter=require("../../common/utils/refSetter.cjs.js"),borderRadius=require("../../themes/borderRadius.cjs.js"),SuffixSelect=require("./SuffixSelect.cjs.js"),index$1=require("../../Tooltip/index.cjs.js"),checkOverflow=require("../../common/utils/checkOverflow.cjs.js"),index=require("../TextInput/index.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const ieFixes=styled.css`
  ::-ms-clear,
  ::-ms-reveal {
    display: none;
  }
`,Input=styled__default.default.input`
  outline: none;
  appearance: none;
  border-radius: inherit;

  box-sizing: border-box;
  flex: 1 1 auto;
  min-width: 10px;
  border: none;
  background: transparent;
  text-overflow: ellipsis;
  padding: 0;

  color: var(--admiral-color-Neutral_Neutral90, ${p=>p.theme.color["Neutral/Neutral 90"]});

  &&&:disabled {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }

  ${props=>"s"===props.$dimension?typography.typography["Body/Body 2 Long"]:typography.typography["Body/Body 1 Long"]}
  &::placeholder {
    color: var(--admiral-color-Neutral_Neutral50, ${p=>p.theme.color["Neutral/Neutral 50"]});
  }

  &:disabled::placeholder {
    color: var(--admiral-color-Neutral_Neutral30, ${p=>p.theme.color["Neutral/Neutral 30"]});
  }

  [data-disable-copying] & {
    user-select: none;
    pointer-events: none;
  }

  &&&:user-invalid + ${index.InputBorderedDiv} {
    border: 1px solid var(--admiral-color-Error_Error60Main, ${p=>p.theme.color["Error/Error 60 Main"]});
  }

  &&&:user-invalid:hover:not(:disabled) + ${index.InputBorderedDiv} {
    border: 1px solid var(--admiral-color-Error_Error70, ${p=>p.theme.color["Error/Error 70"]});
  }

  &&&:user-invalid:focus:not(:disabled) + ${index.InputBorderedDiv} {
    border: 2px solid var(--admiral-color-Error_Error60Main, ${p=>p.theme.color["Error/Error 60 Main"]});
  }

  ${ieFixes}

  &&&:disabled {
    cursor: not-allowed;
  }
`,prefixsuffixMixin=styled.css`
  ${props=>"s"===props.$dimension?typography.typography["Body/Body 2 Long"]:typography.typography["Body/Body 1 Long"]};
  color: ${({disabled,theme})=>disabled?`var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral50, ${theme.color["Neutral/Neutral 50"]})`};
  white-space: nowrap;

  [data-disable-copying] & {
    cursor: default;
  }

  [data-read-only] & {
    pointer-events: none;
    user-select: none;
  }
`,PrefixContainer=styled__default.default.div`
  align-self: center;
  border-right: solid 1px var(--admiral-color-Neutral_Neutral20, ${p=>p.theme.color["Neutral/Neutral 20"]});
  padding-right: 8px;
  margin-right: 8px;
  ${prefixsuffixMixin}
`,SuffixContainer=styled__default.default.div`
  align-self: center;
  border-left: solid 1px var(--admiral-color-Neutral_Neutral20, ${p=>p.theme.color["Neutral/Neutral 20"]});
  padding-left: 8px;
  margin-left: 8px;
  ${prefixsuffixMixin}
`,IconPanel=styled__default.default.div`
  position: relative;
  display: flex;
  align-items: center;

  & > svg {
    border-radius: var(--admiral-border-radius-Medium, ${p=>borderRadius.mediumGroupBorderRadius(p.theme.shape)});
    display: block;
    width: ${props=>{switch(props.$dimension){case"xl":default:return 24;case"s":return 20}}}px;

    &:focus {
      outline: none;
    }

    &:focus-visible {
      outline-offset: 2px;
      outline: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]}) solid 2px;
    }
  }
`,IconPanelBefore=styled__default.default(IconPanel)`
  margin-right: 8px;
  & > *:not(:first-child) {
    margin-right: 8px;
  }
`,IconPanelAfter=styled__default.default(IconPanel)`
  margin-left: 8px;
  & > *:not(:first-child) {
    margin-left: 8px;
  }
`,preventDefault=e=>e.preventDefault(),Container=styled__default.default.div`
  position: relative;
  display: flex;
  align-items: stretch;
  overflow: hidden;
  border: none;
  border-radius: ${p=>p.$skeleton?0:`var(--admiral-border-radius-Medium, ${borderRadius.mediumGroupBorderRadius(p.theme.shape)})`};
  padding: 0 ${props=>{switch(props.$dimension){case"xl":default:return 16;case"s":return 12}}}px;
  background-color: var(--admiral-color-Neutral_Neutral00, ${p=>p.theme.color["Neutral/Neutral 00"]});

  &[data-read-only],
  &[data-disabled] {
    background-color: var(--admiral-color-Neutral_Neutral10, ${p=>p.theme.color["Neutral/Neutral 10"]});
  }

  ${Container$1.containerHeights}
  ${({$skeleton})=>$skeleton&&Container$1.skeletonMixin};
  ${index.BorderedDivStyles}
  ${({disabled})=>disabled?"cursor: not-allowed;":""}
`,InputEx=e.forwardRef(({dimension="m",menuWidth,displayClearIcon,status,containerRef=()=>null,alignDropRef,targetElement,icons,iconsBefore,iconsAfter,children,className,style,placeholder,prefixValue,renderPrefixValue=({value})=>value?String(value):value,prefixValueList,onPrefixValueChange,renderPrefixOption,suffixValue,renderSuffixValue=({value})=>value?String(value):value,suffixValueList,onSuffixValueChange,renderSuffixOption,skeleton=!1,dropContainerCssMixin,prefixDropContainerStyle,suffixDropContainerStyle,showTooltip=!0,...props},ref)=>{const innerContainerRef=e.useRef(null),targetNode=targetElement||alignDropRef?.current||innerContainerRef.current,menuDimension="xl"===dimension?"l":dimension,renderPrefix=prefixValueList?props=>jsxRuntime.jsx(SuffixSelect.SuffixSelect,{alignSelf:"flex-start",dimension:menuDimension,menuWidth:menuWidth||prefixDropContainerStyle?.menuWidth,menuMaxHeight:prefixDropContainerStyle?.menuMaxHeight,targetElement:targetNode,value:props.value||"",onChange:value=>onPrefixValueChange?.(value),options:prefixValueList,disabled:props.disabled,readOnly:props.readOnly,dropContainerCssMixin:prefixDropContainerStyle?.dropContainerCssMixin||dropContainerCssMixin,dropContainerClassName:prefixDropContainerStyle?.dropContainerClassName,dropContainerStyle:prefixDropContainerStyle?.dropContainerStyle,renderValue:renderPrefixValue,renderOption:renderPrefixOption}):renderPrefixValue,prefix=renderPrefix({value:prefixValue,disabled:props.disabled,readOnly:props.readOnly}),renderSuffix=suffixValueList?props=>jsxRuntime.jsx(SuffixSelect.SuffixSelect,{alignSelf:"flex-end",dimension:menuDimension,menuWidth:menuWidth||suffixDropContainerStyle?.menuWidth,menuMaxHeight:suffixDropContainerStyle?.menuMaxHeight,targetElement:targetNode,value:props.value||"",onChange:value=>onSuffixValueChange?.(value),options:suffixValueList,disabled:props.disabled,readOnly:props.readOnly,dropContainerCssMixin:suffixDropContainerStyle?.dropContainerCssMixin||dropContainerCssMixin,dropContainerClassName:suffixDropContainerStyle?.dropContainerClassName,dropContainerStyle:suffixDropContainerStyle?.dropContainerStyle,renderValue:renderSuffixValue,renderOption:renderSuffixOption}):renderSuffixValue,suffix=renderSuffix({value:suffixValue,disabled:props.disabled,readOnly:props.readOnly}),inputRef=e.useRef(null),[overflowActive,setOverflowActive]=e.useState(!1),[tooltipVisible,setTooltipVisible]=e.useState(!1),[innerValueState,setInnerValueState]=e.useState(props.defaultValue??void 0),innerValue=props.value??innerValueState;e.useEffect(()=>{checkOverflow.checkOverflow(inputRef.current)?setOverflowActive(!0):setOverflowActive(!1)},[tooltipVisible,setOverflowActive]),e.useLayoutEffect(()=>{function show(){document.activeElement!==inputRef.current&&setTooltipVisible(!0)}function hide(){setTooltipVisible(!1)}const wrapper=innerContainerRef.current;if(wrapper)return wrapper.addEventListener("mouseenter",show),wrapper.addEventListener("mouseleave",hide),wrapper.addEventListener("mousedown",hide),()=>{wrapper.removeEventListener("mouseenter",show),wrapper.removeEventListener("mouseleave",hide),wrapper.removeEventListener("mousedown",hide)}},[setTooltipVisible]);const iconAfterArray=e.Children.toArray(iconsAfter||icons),iconBeforeArray=e.Children.toArray(iconsBefore);!props.readOnly&&displayClearIcon&&innerValue&&iconAfterArray.unshift(jsxRuntime.jsx(InputIconButton.InputIconButton,{icon:CloseOutline.ReactComponent,onClick:()=>{inputRef.current&&changeInputData.changeInputData(inputRef.current,{value:""})},"aria-hidden":!0},"clear-icon"));const iconsAfterCount=iconAfterArray.length,iconsBeforeCount=iconBeforeArray.length;return jsxRuntime.jsxs(jsxRuntime.Fragment,{children:[jsxRuntime.jsxs(Container,{className,style,"data-disabled":!!props.disabled||void 0,$dimension:dimension,ref:refSetter.refSetter(innerContainerRef,containerRef),"data-read-only":!!props.readOnly||void 0,"data-status":status,"data-disable-copying":!!props.disableCopying||void 0,onMouseDown:props.disableCopying?preventDefault:void 0,$skeleton:skeleton,$status:status,disabled:props.disabled,readOnly:props.readOnly,children:[!!prefix&&jsxRuntime.jsx(PrefixContainer,{$dimension:dimension,disabled:props.disabled,children:prefix}),iconsBeforeCount>0&&jsxRuntime.jsx(IconPanelBefore,{disabled:props.disabled,$dimension:dimension,children:iconBeforeArray}),jsxRuntime.jsx(Input,{ref:refSetter.refSetter(ref,inputRef),...props,onChange:e=>{setInnerValueState(e.currentTarget.value),props.onChange?.(e)},placeholder,$iconsAfterCount:iconsAfterCount,$iconsBeforeCount:iconsBeforeCount,$dimension:dimension}),jsxRuntime.jsx(index.InputBorderedDiv,{$status:status,disabled:props.disabled||props.readOnly}),iconsAfterCount>0&&jsxRuntime.jsx(IconPanelAfter,{disabled:props.disabled,$dimension:dimension,children:iconAfterArray}),!!suffix&&jsxRuntime.jsx(SuffixContainer,{$dimension:dimension,disabled:props.disabled,children:suffix}),children]}),showTooltip&&tooltipVisible&&overflowActive&&jsxRuntime.jsx(index$1.Tooltip,{renderContent:()=>inputRef?.current?.value||"",targetElement:innerContainerRef.current})]})});InputEx.displayName="InputEx",exports.InputEx=InputEx;
