// @admiral-ds/react-ui/dist/components/input/InputEx/SuffixSelect.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),styled=require("styled-components"),OpenStatusButton=require("../../OpenStatusButton.cjs.js"),index=require("../../DropdownContainer/index.cjs.js"),Menu=require("./Menu.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const StyledMenu=styled__default.default(Menu.Menu)`
  width: ${({$width})=>$width||"auto"};
`,preventDefault=e=>e.preventDefault(),Container=styled__default.default.div`
  display: flex;
  align-items: center;
  cursor: pointer;

  &[data-read-only] {
    cursor: default;
  }

  & > svg {
    display: block;
    width: ${p=>p.$iconSizeValue||"24px"};

    &:focus {
      outline: none;
    }

    &:focus-visible {
      outline-offset: 2px;
      outline: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]}) solid 2px;
    }
  }

  ${p=>p.disabled?"cursor: not-allowed;":styled.css`
          &:hover {
            & > ${OpenStatusButton.OpenStatusButton} *[fill^='#'] {
              fill: var(--admiral-color-Primary_Primary70, ${p=>p.theme.color["Primary/Primary 70"]});
            }
          }
        `}
`,ValueContainer=styled__default.default.div`
  margin-right: 4px;
`;exports.SuffixSelect=({alignRef,targetElement,alignSelf,dimension,menuWidth,menuMaxHeight,onChange,options,value,disabled,dropContainerCssMixin,dropContainerClassName,dropContainerStyle,renderValue,renderOption,...props})=>{const[isOpenState,setIsOpenState]=e.useState(!1),isOpen=void 0===props.isOpen?isOpenState:props.isOpen,containerRef=e.useRef(null),targetNode=targetElement||alignRef?.current||containerRef.current;return jsxRuntime.jsxs(jsxRuntime.Fragment,{children:[jsxRuntime.jsxs(Container,{ref:containerRef,disabled,"data-read-only":!!props.readOnly||void 0,onMouseDown:props.readOnly||disabled?preventDefault:e=>{e.preventDefault();const newOpenStatus=!isOpen;props.onOpenChange?.(newOpenStatus),setIsOpenState(newOpenStatus)},children:[renderValue?renderValue({value}):jsxRuntime.jsx(ValueContainer,{children:String(value)}),!props.readOnly&&jsxRuntime.jsx(OpenStatusButton.OpenStatusButton,{$isOpen:isOpen,"aria-hidden":!0,"data-disabled":disabled})]}),isOpen&&jsxRuntime.jsx(index.StyledDropdownContainer,{role:"listbox",alignSelf,targetElement:targetNode,onClickOutside:e=>{e.target&&containerRef.current?.contains(e.target)||setIsOpenState(!1)},dropContainerCssMixin,className:dropContainerClassName,style:dropContainerStyle,children:jsxRuntime.jsx(StyledMenu,{$width:menuWidth,maxHeight:menuMaxHeight,options,selected:value,onSelect:value=>{onChange(value),setIsOpenState(!1)},renderOption,dimension,preventFocusSteal:!0})})]})};
