// @admiral-ds/react-ui/dist/components/Toggle/index.cjs.js
"use strict";var jsxRuntime=require("react/jsx-runtime"),e=require("react"),styled=require("styled-components"),typography=require("../Typography/typography.cjs.js");require("../Typography/defaultTypographyMixin.cjs.js");var mixins=require("./mixins.cjs.js"),keyboardKey=require("../common/keyboardKey.cjs.js");function _interopDefault(e){return e&&e.__esModule?e:{default:e}}var styled__default=_interopDefault(styled);const hoverInputStyles=styled.css`
  & + div > div {
    ${({$dimension})=>`left: calc(${"m"===$dimension?mixins.SLIDER_SIZE_M:mixins.SLIDER_SIZE_S} + -12px + 4px);`}
  }
`,Input=styled__default.default.input`
  position: absolute;
  cursor: ${({readOnly})=>readOnly?"default":"pointer"};
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  padding: 0;
  margin: 0;
  opacity: 0;

  &:disabled {
    cursor: not-allowed;
  }

  &:checked {
    & + div > span {
      &:before {
        ${({$dimension})=>`left: calc(${"m"===$dimension?mixins.SLIDER_SIZE_M:mixins.SLIDER_SIZE_S} + 2px);`}
      }

      background: ${({theme,readOnly})=>readOnly?`var(--admiral-color-Primary_Primary30, ${theme.color["Primary/Primary 30"]})`:`var(--admiral-color-Primary_Primary60Main, ${theme.color["Primary/Primary 60 Main"]})`};
    }

    &:disabled + div > span {
      background: var(--admiral-color-Primary_Primary30, ${p=>p.theme.color["Primary/Primary 30"]});
    }

    ${hoverInputStyles};
  }

  &:not(:disabled) {
    &:focus-visible + div > span {
      outline-offset: 2px;
      outline: var(--admiral-color-Primary_Primary60Main, ${p=>p.theme.color["Primary/Primary 60 Main"]}) solid 2px;
    }

    &:hover {
      & + div > div {
        ${({readOnly})=>!readOnly&&"visibility: visible"};
      }
      &:focus-visible + div > span {
        outline: none;
      }
    }
    &:active {
      & + div > div {
        ${({readOnly})=>!readOnly&&"visibility: visible"};
        background: var(--admiral-color-Opacity_Press, ${p=>p.theme.color["Opacity/Press"]});
      }
    }
  }
`,Label=styled__default.default.div`
  display: flex;
  flex-direction: column;
  ${({$position})=>"right"===$position?"margin-left: 8px;":"margin-right: 8px;"}
  ${({$dimension})=>"s"===$dimension?typography.typography["Body/Body 2 Short"]:typography.typography["Body/Body 1 Short"]}
  color: ${({disabled,theme})=>disabled?`var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral90, ${theme.color["Neutral/Neutral 90"]})`};
`,Hint=styled__default.default.div`
  margin-top: 4px;
  ${({$dimension})=>"s"===$dimension?typography.typography["Caption/Caption 1"]:typography.typography["Body/Body 2 Short"]}
  color: ${({theme,disabled})=>disabled?`var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral50, ${theme.color["Neutral/Neutral 50"]})`};
`,Slider=styled__default.default.span`
  position: absolute;
  top: 0;
  left: 0;
  border-radius: ${"10px"};
  flex-shrink: 0;
  ${mixins.sizes}

  &:before {
    content: '';
    position: absolute;
    top: ${2}px;
    left: ${2}px;
    background: var(--admiral-color-Special_StaticWhite, ${p=>p.theme.color["Special/Static White"]});
    transition: all 0.3s;
    ${mixins.sliderSizes}
  }

  background: ${({theme,$faded})=>$faded?`var(--admiral-color-Neutral_Neutral30, ${theme.color["Neutral/Neutral 30"]})`:`var(--admiral-color-Neutral_Neutral50, ${theme.color["Neutral/Neutral 50"]})`};
`,Hover=styled__default.default.div`
  visibility: hidden;
  pointer-events: none;
  position: absolute;
  top: ${"-8px"};
  left: ${"-8px"};
  background: var(--admiral-color-Opacity_Hover, ${p=>p.theme.color["Opacity/Hover"]});
  ${mixins.hoverSizes}
`,SliderWrapper=styled__default.default.div`
  pointer-events: none;
  position: relative;
  flex-shrink: 0;
  ${mixins.sizes}
`,Wrapper=styled__default.default.label`
  display: flex;
  flex-direction: ${({$labelPosition})=>"right"===$labelPosition?"row":"row-reverse"};
  align-items: flex-start;
  justify-content: space-between;
  position: relative;
  width: ${({width})=>width?"number"==typeof width?`${width}px`:width:"fit-content"};
  cursor: ${({disabled,readOnly})=>disabled?"not-allowed":readOnly?"default":"pointer"};
  -webkit-tap-highlight-color: transparent;
  user-select: none;
  ${({readOnly})=>readOnly&&"pointer-events: none"};
`,Toggle=e.forwardRef(({dimension="m",labelPosition="right",disabled=!1,readOnly=!1,width,extraText,className,children,...props},ref)=>jsxRuntime.jsxs(Wrapper,{className,width:"left"===labelPosition?width:void 0,$labelPosition:labelPosition,disabled,readOnly,role:"switch","aria-checked":props.checked||props["aria-checked"],children:[jsxRuntime.jsx(Input,{ref,type:"checkbox",$dimension:dimension,disabled,readOnly,...props,onKeyDown:e=>{if(readOnly){keyboardKey.keyboardKey.getCode(e)===keyboardKey.keyboardKey[" "]&&e.preventDefault()}props.onKeyDown?.(e)}}),jsxRuntime.jsxs(SliderWrapper,{$dimension:dimension,children:[jsxRuntime.jsx(Hover,{$dimension:dimension}),jsxRuntime.jsx(Slider,{$dimension:dimension,$faded:disabled||readOnly,"aria-hidden":!0})]}),children&&jsxRuntime.jsxs(Label,{$dimension:dimension,disabled,$position:labelPosition,children:[children,extraText&&jsxRuntime.jsx(Hint,{$dimension:dimension,disabled,children:extraText})]})]}));Toggle.displayName="Toggle",exports.Toggle=Toggle;
